<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="wolfemd" />


<title>Review and QC of NRCRI training data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/darkly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">NRCRI_2020GS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Synopsis</a>
</li>
<li>
  <a href="01-cleanTPdata.html">Data Cleaning</a>
</li>
<li>
  <a href="02-curateByTrial.html">Curate Data</a>
</li>
<li>
  <a href="03-GetBLUPs.html">Get BLUPs</a>
</li>
<li>
  <a href="04-CrossValidation.html">Check prediction accuracy</a>
</li>
<li>
  <a href="05-GetGEBVs.html">Get GEBV for GS C2</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/wolfemd/NRCRI_2020GS">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Review and QC of NRCRI training data</h1>
<h4 class="author">wolfemd</h4>
<h4 class="date">2020-April-21</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-04-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>NRCRI_2020GS/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.1). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200421code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200421)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200421code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200421)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong8c45991"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> 8c45991 </a>
</p>
</div>
<div id="strongRepositoryversionstrong8c45991" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version 8c45991. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  NRCRI 2020 GS.gslides
    Untracked:  data/DatabaseDownload_42120/
    Untracked:  data/GBSdataMasterList_31818.csv
    Untracked:  data/IITA_GBStoPhenoMaster_33018.csv
    Untracked:  data/NRCRI_CleanedTrialData_2020April21.rds
    Untracked:  data/NRCRI_ExptDesignsDetected_2020April21.rds
    Untracked:  data/NRCRI_GBStoPhenoMaster_40318.csv
    Untracked:  data/iita_blupsForCrossVal_72619.rds
    Untracked:  data/iita_blupsForCrossVal_outliersRemoved_73019.rds
    Untracked:  output/DosageMatrix_NRCRI_SamplesForGP_2020April27.rds
    Untracked:  output/GEBV_NRCRI_IITAaugmented_ModelADE_2020April27.rds
    Untracked:  output/GEBV_NRCRI_IITAaugmented_ModelA_2020April27.rds
    Untracked:  output/GEBV_NRCRI_NRCRIalone_ModelADE_2020April27.rds
    Untracked:  output/GEBV_NRCRI_NRCRIalone_ModelA_2020April27.rds
    Untracked:  output/GETGV_NRCRI_IITAaugmented_ModelADE_2020April27.rds
    Untracked:  output/GETGV_NRCRI_NRCRIalone_ModelADE_2020April27.rds
    Untracked:  output/Kinship_AA_NRCRI_2020April27.rds
    Untracked:  output/Kinship_AD_NRCRI_2020April27.rds
    Untracked:  output/Kinship_A_NRCRI_2020April27.rds
    Untracked:  output/Kinship_DD_NRCRI_2020April27.rds
    Untracked:  output/Kinship_D_NRCRI_2020April27.rds
    Untracked:  output/NRCRI_CuratedTrials_2020April21.rds
    Untracked:  output/NRCRI_CuratedTrials_2020April27.rds
    Untracked:  output/cvresults_AD_nrOnly.rds
    Untracked:  output/cvresults_A_iitaAugmented.rds
    Untracked:  output/cvresults_A_nrOnly.rds
    Untracked:  output/genomicPredictions_ModelADE_NRCRI_2020April27.rds
    Untracked:  output/genomicPredictions_ModelA_NRCRI_2020April27.rds
    Untracked:  output/maxNOHAV_byStudy_NRCRI_2020April27.csv
    Untracked:  output/nrcri_blupsForModelTraining_2020April21.rds
    Untracked:  output/nrcri_blupsForModelTraining_2020April27.rds
    Untracked:  workflowr_log.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/01-cleanTPdata.Rmd</code>) and HTML (<code>docs/01-cleanTPdata.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
8c45991
</td>
<td>
wolfemd
</td>
<td>
2020-04-28
</td>
<td>
Publish the first set of analyses and files for NRCRI 2020 GS.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="objective" class="section level1">
<h1>Objective</h1>
<p>Follow outlined GenomicPredictionChecklist and previous pipeline to process cassavabase data for ultimate genomic prediction.</p>
<p>Below we will clean and format NRCRI training data, all years, all trials, downloaded from DB.</p>
<ul>
<li>Inputs: “Raw” field trial data</li>
<li>Expected outputs: “Cleaned” field trial data</li>
</ul>
</div>
<div id="user-input-cassavabase-download" class="section level1">
<h1>[User input] Cassavabase download</h1>
<p>Using the <a href="https://www.cassavabase.org/breeders/search">Cassavabase search wizard</a>:</p>
<ol style="list-style-type: decimal">
<li>Used the Wizard.</li>
<li>Selected all NRCRI trials till present. Make a list. Named it <strong>All_NRCRI_TRIALS_2020April21</strong>.</li>
<li>Go to <strong>Manage</strong> –&gt; <strong>Download</strong>. Download phenotypes and meta-data as CSV using the corresponding boxes / drop-downs.</li>
</ol>
<ul>
<li><strong>TRIED TO DOWNLOAD META-DATA, BUT DB IS GIVING “SERVER ERROR”</strong></li>
</ul>
</div>
<div id="read-in-trial-data" class="section level1">
<h1>Read-in trial data</h1>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="kw">library</span>(tidyverse); <span class="kw">library</span>(magrittr);</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a>readDBdata&lt;-<span class="cf">function</span>(phenotypeFile,<span class="dt">metadataFile=</span><span class="ot">NULL</span>){</span>
<span id="cb2-2"><a href="#cb2-2"></a>      indata&lt;-<span class="kw">read.csv</span>(phenotypeFile,</span>
<span id="cb2-3"><a href="#cb2-3"></a>                      <span class="dt">na.strings =</span> <span class="kw">c</span>(<span class="st">&quot;#VALUE!&quot;</span>,<span class="ot">NA</span>,<span class="st">&quot;.&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="st">&quot; &quot;</span>,<span class="st">&quot;-&quot;</span>,<span class="st">&quot;</span><span class="ch">\&quot;</span><span class="st">&quot;</span>),</span>
<span id="cb2-4"><a href="#cb2-4"></a>                      <span class="dt">stringsAsFactors =</span> F)</span>
<span id="cb2-5"><a href="#cb2-5"></a>      <span class="cf">if</span>(<span class="op">!</span><span class="kw">is.null</span>(metadataFile)){</span>
<span id="cb2-6"><a href="#cb2-6"></a>      meta&lt;-<span class="kw">read.csv</span>(metadataFile,</span>
<span id="cb2-7"><a href="#cb2-7"></a>                     <span class="dt">na.strings =</span> <span class="kw">c</span>(<span class="st">&quot;#VALUE!&quot;</span>,<span class="ot">NA</span>,<span class="st">&quot;.&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="st">&quot; &quot;</span>,<span class="st">&quot;-&quot;</span>,<span class="st">&quot;</span><span class="ch">\&quot;</span><span class="st">&quot;</span>),</span>
<span id="cb2-8"><a href="#cb2-8"></a>                     <span class="dt">stringsAsFactors =</span> F) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb2-9"><a href="#cb2-9"></a><span class="st">            </span><span class="kw">rename</span>(<span class="dt">programName=</span>breedingProgramName,</span>
<span id="cb2-10"><a href="#cb2-10"></a>                   <span class="dt">programDescription=</span>breedingProgramDescription,</span>
<span id="cb2-11"><a href="#cb2-11"></a>                   <span class="dt">programDbId=</span>breedingProgramDbId)</span>
<span id="cb2-12"><a href="#cb2-12"></a>      indata&lt;-<span class="kw">left_join</span>(indata,meta) }</span>
<span id="cb2-13"><a href="#cb2-13"></a>      indata <span class="op">%&lt;&gt;%</span><span class="st"> </span></span>
<span id="cb2-14"><a href="#cb2-14"></a><span class="st">        </span><span class="kw">filter</span>(observationLevel<span class="op">==</span><span class="st">&quot;plot&quot;</span>)</span>
<span id="cb2-15"><a href="#cb2-15"></a>      <span class="kw">return</span>(indata) }</span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>dbdata&lt;-<span class="kw">readDBdata</span>(<span class="dt">phenotypeFile =</span> here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data/DatabaseDownload_42120/&quot;</span>,<span class="st">&quot;2020-04-21T154040phenotype_download.csv&quot;</span>))</span></code></pre></div>
</div>
<div id="group-and-select-trials-to-analyze" class="section level1">
<h1>Group and select trials to analyze</h1>
<p>Make TrialType Variable</p>
<p>The function below requires an iterative, interactive process for me to make sure I include and correctly classify trials.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>makeTrialTypeVar&lt;-<span class="cf">function</span>(indata){</span>
<span id="cb4-2"><a href="#cb4-2"></a>  <span class="co"># So far, this function is not very general</span></span>
<span id="cb4-3"><a href="#cb4-3"></a>  <span class="co"># Handles IITA and NRCRI trial names as of April 2020.</span></span>
<span id="cb4-4"><a href="#cb4-4"></a>  <span class="co"># Can customize this or add lines to grab TrialTypes for each breeding program</span></span>
<span id="cb4-5"><a href="#cb4-5"></a>  <span class="cf">if</span>(indata<span class="op">$</span>programName<span class="op">==</span><span class="st">&quot;IITA&quot;</span>){</span>
<span id="cb4-6"><a href="#cb4-6"></a>  outdata&lt;-indata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb4-7"><a href="#cb4-7"></a><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">TrialType=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;CE|clonal|13NEXTgenC1&quot;</span>,studyName,<span class="dt">ignore.case =</span> T),<span class="st">&quot;CET&quot;</span>,<span class="ot">NA</span>),</span>
<span id="cb4-8"><a href="#cb4-8"></a>           <span class="dt">TrialType=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;EC&quot;</span>,studyName,<span class="dt">ignore.case =</span> T),<span class="st">&quot;ExpCET&quot;</span>,TrialType),</span>
<span id="cb4-9"><a href="#cb4-9"></a>           <span class="dt">TrialType=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;PYT&quot;</span>,studyName,<span class="dt">ignore.case =</span> T),<span class="st">&quot;PYT&quot;</span>,TrialType),</span>
<span id="cb4-10"><a href="#cb4-10"></a>           <span class="dt">TrialType=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;AYT&quot;</span>,studyName,<span class="dt">ignore.case =</span> T),<span class="st">&quot;AYT&quot;</span>,TrialType),</span>
<span id="cb4-11"><a href="#cb4-11"></a>           <span class="dt">TrialType=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;UYT&quot;</span>,studyName,<span class="dt">ignore.case =</span> T),<span class="st">&quot;UYT&quot;</span>,TrialType),</span>
<span id="cb4-12"><a href="#cb4-12"></a>           <span class="dt">TrialType=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;geneticgain|gg|genetic gain&quot;</span>,studyName,<span class="dt">ignore.case =</span> T),<span class="st">&quot;GeneticGain&quot;</span>,TrialType),</span>
<span id="cb4-13"><a href="#cb4-13"></a>           <span class="dt">TrialType=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;Cassava&quot;</span>,studyName,<span class="dt">ignore.case =</span> T) <span class="op">&amp;</span><span class="st"> </span><span class="kw">grepl</span>(<span class="st">&quot;/&quot;</span>,studyName),<span class="st">&quot;GeneticGain&quot;</span>,TrialType),</span>
<span id="cb4-14"><a href="#cb4-14"></a>           <span class="dt">TrialType=</span><span class="kw">ifelse</span>((<span class="kw">grepl</span>(<span class="st">&quot;clonal evaluation trial&quot;</span>,<span class="op">!</span><span class="kw">grepl</span>(<span class="st">&quot;genetic gain&quot;</span>,studyDescription,<span class="dt">ignore.case =</span> T), </span>
<span id="cb4-15"><a href="#cb4-15"></a>                                   <span class="dt">ignore.case =</span> T)),<span class="st">&quot;CET&quot;</span>,TrialType),</span>
<span id="cb4-16"><a href="#cb4-16"></a>           <span class="dt">TrialType=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;preliminary yield trial&quot;</span>,studyDescription,<span class="dt">ignore.case =</span> T),<span class="st">&quot;PYT&quot;</span>,TrialType),</span>
<span id="cb4-17"><a href="#cb4-17"></a>           <span class="dt">TrialType=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;Crossingblock|GS.C4.CB|cross&quot;</span>,studyName) <span class="op">&amp;</span><span class="st"> </span><span class="kw">is.na</span>(TrialType),<span class="st">&quot;CrossingBlock&quot;</span>,TrialType),</span>
<span id="cb4-18"><a href="#cb4-18"></a>           <span class="dt">TrialType=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;NCRP&quot;</span>,studyName) <span class="op">&amp;</span><span class="st"> </span><span class="kw">is.na</span>(TrialType),<span class="st">&quot;NCRP&quot;</span>,TrialType),</span>
<span id="cb4-19"><a href="#cb4-19"></a>           <span class="dt">TrialType=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;conservation&quot;</span>,studyName) <span class="op">&amp;</span><span class="st"> </span><span class="kw">is.na</span>(TrialType),<span class="st">&quot;Conservation&quot;</span>,TrialType)) }</span>
<span id="cb4-20"><a href="#cb4-20"></a>    <span class="cf">if</span>(indata<span class="op">$</span>programName<span class="op">==</span><span class="st">&quot;NRCRI&quot;</span>){</span>
<span id="cb4-21"><a href="#cb4-21"></a>        outdata&lt;-indata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb4-22"><a href="#cb4-22"></a><span class="st">          </span><span class="kw">mutate</span>(<span class="dt">TrialType=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;TP1&quot;</span>,studyName,<span class="dt">ignore.case =</span> T),<span class="st">&quot;TP1&quot;</span>,<span class="ot">NA</span>),</span>
<span id="cb4-23"><a href="#cb4-23"></a>                 <span class="dt">TrialType=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;TP2&quot;</span>,studyName,<span class="dt">ignore.case =</span> T),<span class="st">&quot;TP2&quot;</span>,TrialType),</span>
<span id="cb4-24"><a href="#cb4-24"></a>                 <span class="dt">TrialType=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;C1a&quot;</span>,studyName,<span class="dt">ignore.case =</span> T),<span class="st">&quot;C1a&quot;</span>,TrialType),</span>
<span id="cb4-25"><a href="#cb4-25"></a>                 <span class="dt">TrialType=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;C1b&quot;</span>,studyName,<span class="dt">ignore.case =</span> T),<span class="st">&quot;C1b&quot;</span>,TrialType),</span>
<span id="cb4-26"><a href="#cb4-26"></a>                 <span class="dt">TrialType=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;C2a&quot;</span>,studyName,<span class="dt">ignore.case =</span> T),<span class="st">&quot;C2a&quot;</span>,TrialType),</span>
<span id="cb4-27"><a href="#cb4-27"></a>                 <span class="dt">TrialType=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;C2b&quot;</span>,studyName,<span class="dt">ignore.case =</span> T),<span class="st">&quot;C2b&quot;</span>,TrialType),</span>
<span id="cb4-28"><a href="#cb4-28"></a>                 <span class="dt">TrialType=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;NCRP&quot;</span>,studyName) <span class="op">&amp;</span><span class="st"> </span><span class="kw">is.na</span>(TrialType),<span class="st">&quot;NCRP&quot;</span>,TrialType),</span>
<span id="cb4-29"><a href="#cb4-29"></a>                 <span class="dt">TrialType=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;15nextgen60gs-cbUM|crossnblk|crossingblock&quot;</span>,studyName,<span class="dt">ignore.case =</span> T) <span class="op">&amp;</span></span>
<span id="cb4-30"><a href="#cb4-30"></a><span class="st">                                    </span><span class="op">!</span><span class="kw">grepl</span>(<span class="st">&quot;CET&quot;</span>,studyName),</span>
<span id="cb4-31"><a href="#cb4-31"></a>                                  <span class="st">&quot;CrossingBlock&quot;</span>,TrialType),</span>
<span id="cb4-32"><a href="#cb4-32"></a>                 <span class="dt">TrialType=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;seedling&quot;</span>,studyName,<span class="dt">ignore.case =</span> T),<span class="ot">NA</span>,TrialType)) }</span>
<span id="cb4-33"><a href="#cb4-33"></a>  <span class="kw">return</span>(outdata) }</span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>dbdata&lt;-<span class="kw">makeTrialTypeVar</span>(dbdata) </span>
<span id="cb5-2"><a href="#cb5-2"></a>dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb5-3"><a href="#cb5-3"></a><span class="st">  </span><span class="kw">count</span>(TrialType)</span></code></pre></div>
<pre><code># A tibble: 9 x 2
  TrialType         n
  &lt;chr&gt;         &lt;int&gt;
1 C1a            3382
2 C1b            4053
3 C2a            1941
4 C2b             279
5 CrossingBlock   141
6 NCRP            270
7 TP1           13228
8 TP2            8993
9 &lt;NA&gt;          20506</code></pre>
<div id="question-for-nrcri" class="section level2">
<h2>Question for NRCRI</h2>
<p>Looking at the <strong>studyName</strong>’s of trials getting NA for TrialType, meaning I can’t classify them at present.</p>
<p>Here is the list of trials I am <em>not</em> including.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a>dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb7-2"><a href="#cb7-2"></a><span class="st">  </span><span class="kw">filter</span>(<span class="kw">is.na</span>(TrialType)) <span class="op">%$%</span><span class="st"> </span><span class="kw">unique</span>(studyName)</span></code></pre></div>
<pre><code> [1] &quot;06cetEarlyBulkingUM&quot;        &quot;06SeedlingsEarlyBulkingUM&quot; 
 [3] &quot;07EarlyBulkingUM&quot;           &quot;07pytEarlyBulkingUM&quot;       
 [5] &quot;08EarlyBulkingIB&quot;           &quot;08EarlyBulkingOT&quot;          
 [7] &quot;08EarlyBulkingUM&quot;           &quot;09pytIgariam&quot;              
 [9] &quot;09pytOtobi&quot;                 &quot;09pytUmudike&quot;              
[11] &quot;10cetNR10BseriesUM&quot;         &quot;10cetNR10seriesUM&quot;         
[13] &quot;10uytCIATsetAUM&quot;            &quot;10uytCIATsetBKA&quot;           
[15] &quot;10uytUM&quot;                    &quot;11aytNR09seriesUM&quot;         
[17] &quot;11cetNR11seriesUM&quot;          &quot;11clonal63cpUM&quot;            
[19] &quot;11cmd48gpUM&quot;                &quot;11EarlyBulkingDrtKA&quot;       
[21] &quot;11UMUyellowRTuyt&quot;           &quot;11uyt10RepeatUM&quot;           
[23] &quot;12CETearlyBulkingDrtkano&quot;   &quot;12PYT36genpyramOtobi&quot;      
[25] &quot;12PYT75genpyramUM&quot;          &quot;13AYT12yrtOtobi&quot;           
[27] &quot;13AYT259mpopdOtobi&quot;         &quot;13AYT25umu&quot;                
[29] &quot;13AYT35gpOtobii&quot;            &quot;13AYT60gpumu&quot;              
[31] &quot;13AYT70mpopfOtobi&quot;          &quot;13CET261umu&quot;               
[33] &quot;13mlt12yrtIG&quot;               &quot;13PYT144mpopumu&quot;           
[35] &quot;13PYT17mpopfumu&quot;            &quot;13pyt20nr12UM&quot;             
[37] &quot;13UYT12yrtumu&quot;              &quot;13uyt15umu&quot;                
[39] &quot;14cCET150umu&quot;               &quot;14CET12Highbetacaroumu&quot;    
[41] &quot;14CET178mpopumu&quot;            &quot;14clonal35ctasetbUM&quot;       
[43] &quot;14clonal35ctaUM&quot;            &quot;14highbetacaro12umu&quot;       
[45] &quot;14iHighbetacaroIghariam&quot;    &quot;14PYT21highbetacaroumu&quot;    
[47] &quot;14PYT25wrtumu&quot;              &quot;14tHighBetaCaroOtobi&quot;      
[49] &quot;14UYT12umu&quot;                 &quot;15ayt11nr13UM&quot;             
[51] &quot;15ayt12wrtUM&quot;               &quot;15nextgen351cgm-amUM&quot;      
[53] &quot;15uyt10hbcUM&quot;               &quot;16uyt10hcbUM&quot;              
[55] &quot;17C2aSeedling Nursery&quot;      &quot;17cet200yellowrt_umu&quot;      
[57] &quot;17ppdyieldexperimentUM&quot;     &quot;17pyt30yellowrtumu&quot;        
[59] &quot;17uyt10yellowrt_umu&quot;        &quot;17uyt12yellowrt_umu&quot;       
[61] &quot;17uyt13yellowrt_umu&quot;        &quot;18AYT13yrtOtobi&quot;           
[63] &quot;18AYT13yrtumu&quot;              &quot;18C2bSeedlingNurseryumu&quot;   
[65] &quot;18introLatAmeumu&quot;           &quot;18pyt_lgbariam&quot;            
[67] &quot;18PYTmealiness_umu&quot;         &quot;18pyt_otobi&quot;               
[69] &quot;18UYT12yrtIgbariam&quot;         &quot;18UYT12yrtOtobi&quot;           
[71] &quot;18UYT12yrtumu_1&quot;            &quot;18UYT12yrtumu_2&quot;           
[73] &quot;19C3SeedlingNursery_umu&quot;    &quot;19cetcftumu_1&quot;             
[75] &quot;19cetcftumu_2&quot;              &quot;19cetcftumu_3&quot;             
[77] &quot;19crossingblockCETubiaja&quot;   &quot;19introLatAmeumu&quot;          
[79] &quot;19S1CETumu&quot;                 &quot;19UYTumudike&quot;              
[81] &quot;20pytNUEigbariam_1&quot;         &quot;20pytNUEigbariam_2&quot;        
[83] &quot;20pytNUEotobi_1&quot;            &quot;20pytNUEotobi_2&quot;           
[85] &quot;20pytNUEumu_1&quot;              &quot;20pytNUEumu_2&quot;             
[87] &quot;CETCrossingblock19_ubiaja&quot;  &quot;Ikenne2019RootPhenotyping&quot; 
[89] &quot;PYT 2010&quot;                   &quot;Umudike2013set1CGM&quot;        
[91] &quot;Umudike2013set2CGM&quot;         &quot;Umudike2019RootPhenotyping&quot;</code></pre>
<p>Should any of these trials have been included?</p>
<p>Especially the following new trials (post 2018)</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb9-2"><a href="#cb9-2"></a><span class="st">  </span><span class="kw">filter</span>(<span class="kw">is.na</span>(TrialType),</span>
<span id="cb9-3"><a href="#cb9-3"></a>         <span class="kw">as.numeric</span>(studyYear)<span class="op">&gt;</span><span class="dv">2018</span>) <span class="op">%$%</span><span class="st"> </span><span class="kw">unique</span>(studyName)</span></code></pre></div>
<pre><code> [1] &quot;19C3SeedlingNursery_umu&quot;    &quot;19cetcftumu_1&quot;             
 [3] &quot;19cetcftumu_2&quot;              &quot;19cetcftumu_3&quot;             
 [5] &quot;19crossingblockCETubiaja&quot;   &quot;19introLatAmeumu&quot;          
 [7] &quot;19S1CETumu&quot;                 &quot;19UYTumudike&quot;              
 [9] &quot;20pytNUEigbariam_1&quot;         &quot;20pytNUEigbariam_2&quot;        
[11] &quot;20pytNUEotobi_1&quot;            &quot;20pytNUEotobi_2&quot;           
[13] &quot;20pytNUEumu_1&quot;              &quot;20pytNUEumu_2&quot;             
[15] &quot;CETCrossingblock19_ubiaja&quot;  &quot;Ikenne2019RootPhenotyping&quot; 
[17] &quot;Umudike2019RootPhenotyping&quot;</code></pre>
</div>
<div id="remove-unclassified-trials" class="section level2">
<h2>Remove unclassified trials</h2>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>dbdata <span class="op">%&lt;&gt;%</span><span class="st"> </span></span>
<span id="cb11-2"><a href="#cb11-2"></a><span class="st">    </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(TrialType)) </span>
<span id="cb11-3"><a href="#cb11-3"></a>dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb11-4"><a href="#cb11-4"></a><span class="st">    </span><span class="kw">group_by</span>(programName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb11-5"><a href="#cb11-5"></a><span class="st">    </span><span class="kw">summarize</span>(<span class="dt">N=</span><span class="kw">n</span>())</span></code></pre></div>
<pre><code># A tibble: 1 x 2
  programName     N
  &lt;chr&gt;       &lt;int&gt;
1 NRCRI       32287</code></pre>
</div>
</div>
<div id="traits-and-traitabbreviations" class="section level1">
<h1>Traits and TraitAbbreviations</h1>
<p>Function to rename columns and remove everything unecessary</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="co">#&#39; @param  traitabbrevs data.frame with 2 cols (TraitAbbrev and TraitName). TraitName should match exactly to cassava ontology names</span></span>
<span id="cb13-2"><a href="#cb13-2"></a><span class="co">#&#39; @param  indata data.frame read from cassavabase download</span></span>
<span id="cb13-3"><a href="#cb13-3"></a><span class="co">#&#39; @param  customColsToKeep char. vec. of any custom cols you added and want to keep</span></span>
<span id="cb13-4"><a href="#cb13-4"></a>renameAndSelectCols&lt;-<span class="cf">function</span>(traitabbrevs,indata,</span>
<span id="cb13-5"><a href="#cb13-5"></a>                              <span class="dt">customColsToKeep=</span><span class="ot">NULL</span>){</span>
<span id="cb13-6"><a href="#cb13-6"></a>  outdata&lt;-indata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb13-7"><a href="#cb13-7"></a><span class="st">    </span><span class="kw">select</span>(studyYear,programName,locationName,studyName,studyDesign,plotWidth,plotLength,fieldSize,</span>
<span id="cb13-8"><a href="#cb13-8"></a>           plantingDate,harvestDate,locationName,germplasmName,</span>
<span id="cb13-9"><a href="#cb13-9"></a>           replicate,blockNumber,plotNumber,rowNumber,colNumber,entryType,</span>
<span id="cb13-10"><a href="#cb13-10"></a>           <span class="co"># trialType:numberReps,folderName, # these are columns that come from the metadata file</span></span>
<span id="cb13-11"><a href="#cb13-11"></a>           <span class="kw">any_of</span>(customColsToKeep), </span>
<span id="cb13-12"><a href="#cb13-12"></a>           <span class="kw">any_of</span>(traitabbrevs<span class="op">$</span>TraitName)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb13-13"><a href="#cb13-13"></a><span class="st">    </span><span class="kw">pivot_longer</span>(<span class="dt">cols =</span> traitabbrevs<span class="op">$</span>TraitName[traitabbrevs<span class="op">$</span>TraitName <span class="op">%in%</span><span class="st"> </span><span class="kw">colnames</span>(indata)],</span>
<span id="cb13-14"><a href="#cb13-14"></a>                 <span class="dt">names_to =</span> <span class="st">&quot;TraitName&quot;</span>,</span>
<span id="cb13-15"><a href="#cb13-15"></a>                 <span class="dt">values_to =</span> <span class="st">&quot;Value&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb13-16"><a href="#cb13-16"></a><span class="st">    </span><span class="kw">left_join</span>(.,traitabbrevs) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb13-17"><a href="#cb13-17"></a><span class="st">    </span><span class="kw">select</span>(<span class="op">-</span>TraitName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb13-18"><a href="#cb13-18"></a><span class="st">    </span><span class="kw">pivot_wider</span>(<span class="dt">names_from =</span> TraitAbbrev,</span>
<span id="cb13-19"><a href="#cb13-19"></a>                <span class="dt">values_from =</span> <span class="st">&quot;Value&quot;</span>)</span>
<span id="cb13-20"><a href="#cb13-20"></a>  <span class="kw">return</span>(outdata) }</span></code></pre></div>
<p>Making a table of abbreviations for renaming</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a>traitabbrevs&lt;-<span class="kw">tribble</span>(<span class="op">~</span>TraitAbbrev,<span class="op">~</span>TraitName,</span>
<span id="cb14-2"><a href="#cb14-2"></a>        <span class="st">&quot;CMD1S&quot;</span>,<span class="st">&quot;cassava.mosaic.disease.severity.1.month.evaluation.CO_334.0000191&quot;</span>,</span>
<span id="cb14-3"><a href="#cb14-3"></a>        <span class="st">&quot;CMD3S&quot;</span>,<span class="st">&quot;cassava.mosaic.disease.severity.3.month.evaluation.CO_334.0000192&quot;</span>,</span>
<span id="cb14-4"><a href="#cb14-4"></a>        <span class="st">&quot;CMD6S&quot;</span>,<span class="st">&quot;cassava.mosaic.disease.severity.6.month.evaluation.CO_334.0000194&quot;</span>,</span>
<span id="cb14-5"><a href="#cb14-5"></a>        <span class="st">&quot;CMD9S&quot;</span>,<span class="st">&quot;cassava.mosaic.disease.severity.9.month.evaluation.CO_334.0000193&quot;</span>,</span>
<span id="cb14-6"><a href="#cb14-6"></a>        <span class="st">&quot;CGM&quot;</span>,<span class="st">&quot;Cassava.green.mite.severity.CO_334.0000033&quot;</span>,</span>
<span id="cb14-7"><a href="#cb14-7"></a>        <span class="st">&quot;CGMS1&quot;</span>,<span class="st">&quot;cassava.green.mite.severity.first.evaluation.CO_334.0000189&quot;</span>,</span>
<span id="cb14-8"><a href="#cb14-8"></a>        <span class="st">&quot;CGMS2&quot;</span>,<span class="st">&quot;cassava.green.mite.severity.second.evaluation.CO_334.0000190&quot;</span>,</span>
<span id="cb14-9"><a href="#cb14-9"></a>        <span class="st">&quot;DM&quot;</span>,<span class="st">&quot;dry.matter.content.percentage.CO_334.0000092&quot;</span>,</span>
<span id="cb14-10"><a href="#cb14-10"></a>        <span class="st">&quot;PLTHT&quot;</span>,<span class="st">&quot;plant.height.measurement.in.cm.CO_334.0000018&quot;</span>,</span>
<span id="cb14-11"><a href="#cb14-11"></a>        <span class="st">&quot;BRNHT1&quot;</span>,<span class="st">&quot;first.apical.branch.height.measurement.in.cm.CO_334.0000106&quot;</span>,</span>
<span id="cb14-12"><a href="#cb14-12"></a>        <span class="st">&quot;SHTWT&quot;</span>,<span class="st">&quot;fresh.shoot.weight.measurement.in.kg.per.plot.CO_334.0000016&quot;</span>,</span>
<span id="cb14-13"><a href="#cb14-13"></a>        <span class="st">&quot;RTWT&quot;</span>,<span class="st">&quot;fresh.storage.root.weight.per.plot.CO_334.0000012&quot;</span>,</span>
<span id="cb14-14"><a href="#cb14-14"></a>        <span class="st">&quot;RTNO&quot;</span>,<span class="st">&quot;root.number.counting.CO_334.0000011&quot;</span>,</span>
<span id="cb14-15"><a href="#cb14-15"></a>        <span class="st">&quot;TCHART&quot;</span>,<span class="st">&quot;total.carotenoid.by.chart.1.8.CO_334.0000161&quot;</span>,</span>
<span id="cb14-16"><a href="#cb14-16"></a>        <span class="st">&quot;NOHAV&quot;</span>,<span class="st">&quot;plant.stands.harvested.counting.CO_334.0000010&quot;</span>)</span>
<span id="cb14-17"><a href="#cb14-17"></a>traitabbrevs</span></code></pre></div>
<pre><code># A tibble: 15 x 2
   TraitAbbrev TraitName                                                        
   &lt;chr&gt;       &lt;chr&gt;                                                            
 1 CMD1S       cassava.mosaic.disease.severity.1.month.evaluation.CO_334.0000191
 2 CMD3S       cassava.mosaic.disease.severity.3.month.evaluation.CO_334.0000192
 3 CMD6S       cassava.mosaic.disease.severity.6.month.evaluation.CO_334.0000194
 4 CMD9S       cassava.mosaic.disease.severity.9.month.evaluation.CO_334.0000193
 5 CGM         Cassava.green.mite.severity.CO_334.0000033                       
 6 CGMS1       cassava.green.mite.severity.first.evaluation.CO_334.0000189      
 7 CGMS2       cassava.green.mite.severity.second.evaluation.CO_334.0000190     
 8 DM          dry.matter.content.percentage.CO_334.0000092                     
 9 PLTHT       plant.height.measurement.in.cm.CO_334.0000018                    
10 BRNHT1      first.apical.branch.height.measurement.in.cm.CO_334.0000106      
11 SHTWT       fresh.shoot.weight.measurement.in.kg.per.plot.CO_334.0000016     
12 RTWT        fresh.storage.root.weight.per.plot.CO_334.0000012                
13 RTNO        root.number.counting.CO_334.0000011                              
14 TCHART      total.carotenoid.by.chart.1.8.CO_334.0000161                     
15 NOHAV       plant.stands.harvested.counting.CO_334.0000010                   </code></pre>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a>dbdata&lt;-<span class="kw">renameAndSelectCols</span>(traitabbrevs,<span class="dt">indata=</span>dbdata,<span class="dt">customColsToKeep =</span> <span class="st">&quot;TrialType&quot;</span>)</span></code></pre></div>
</div>
<div id="qc-trait-values" class="section level1">
<h1>QC Trait values</h1>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a>dbdata&lt;-dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb17-2"><a href="#cb17-2"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">CMD1S=</span><span class="kw">ifelse</span>(CMD1S<span class="op">&lt;</span><span class="dv">1</span> <span class="op">|</span><span class="st"> </span>CMD1S<span class="op">&gt;</span><span class="dv">5</span>,<span class="ot">NA</span>,CMD1S),</span>
<span id="cb17-3"><a href="#cb17-3"></a>         <span class="dt">CMD3S=</span><span class="kw">ifelse</span>(CMD3S<span class="op">&lt;</span><span class="dv">1</span> <span class="op">|</span><span class="st"> </span>CMD3S<span class="op">&gt;</span><span class="dv">5</span>,<span class="ot">NA</span>,CMD3S),</span>
<span id="cb17-4"><a href="#cb17-4"></a>         <span class="dt">CMD6S=</span><span class="kw">ifelse</span>(CMD6S<span class="op">&lt;</span><span class="dv">1</span> <span class="op">|</span><span class="st"> </span>CMD1S<span class="op">&gt;</span><span class="dv">5</span>,<span class="ot">NA</span>,CMD6S),</span>
<span id="cb17-5"><a href="#cb17-5"></a>         <span class="dt">CMD9S=</span><span class="kw">ifelse</span>(CMD9S<span class="op">&lt;</span><span class="dv">1</span> <span class="op">|</span><span class="st"> </span>CMD1S<span class="op">&gt;</span><span class="dv">5</span>,<span class="ot">NA</span>,CMD9S),</span>
<span id="cb17-6"><a href="#cb17-6"></a>         <span class="dt">CGM=</span><span class="kw">ifelse</span>(CGM<span class="op">&lt;</span><span class="dv">1</span> <span class="op">|</span><span class="st"> </span>CGM<span class="op">&gt;</span><span class="dv">5</span>,<span class="ot">NA</span>,CGM),</span>
<span id="cb17-7"><a href="#cb17-7"></a>         <span class="dt">CGMS1=</span><span class="kw">ifelse</span>(CGMS1<span class="op">&lt;</span><span class="dv">1</span> <span class="op">|</span><span class="st"> </span>CGMS1<span class="op">&gt;</span><span class="dv">5</span>,<span class="ot">NA</span>,CGMS1),</span>
<span id="cb17-8"><a href="#cb17-8"></a>         <span class="dt">CGMS2=</span><span class="kw">ifelse</span>(CGMS2<span class="op">&lt;</span><span class="dv">1</span> <span class="op">|</span><span class="st"> </span>CGMS2<span class="op">&gt;</span><span class="dv">5</span>,<span class="ot">NA</span>,CGMS2),</span>
<span id="cb17-9"><a href="#cb17-9"></a>         <span class="dt">DM=</span><span class="kw">ifelse</span>(DM<span class="op">&gt;</span><span class="dv">100</span> <span class="op">|</span><span class="st"> </span>DM<span class="op">&lt;=</span><span class="dv">0</span>,<span class="ot">NA</span>,DM),</span>
<span id="cb17-10"><a href="#cb17-10"></a>         <span class="dt">RTWT=</span><span class="kw">ifelse</span>(RTWT<span class="op">==</span><span class="dv">0</span> <span class="op">|</span><span class="st"> </span>NOHAV<span class="op">==</span><span class="dv">0</span> <span class="op">|</span><span class="st"> </span><span class="kw">is.na</span>(NOHAV),<span class="ot">NA</span>,RTWT),</span>
<span id="cb17-11"><a href="#cb17-11"></a>         <span class="dt">SHTWT=</span><span class="kw">ifelse</span>(SHTWT<span class="op">==</span><span class="dv">0</span> <span class="op">|</span><span class="st"> </span>NOHAV<span class="op">==</span><span class="dv">0</span> <span class="op">|</span><span class="st"> </span><span class="kw">is.na</span>(NOHAV),<span class="ot">NA</span>,SHTWT),</span>
<span id="cb17-12"><a href="#cb17-12"></a>         <span class="dt">RTNO=</span><span class="kw">ifelse</span>(RTNO<span class="op">==</span><span class="dv">0</span> <span class="op">|</span><span class="st"> </span>NOHAV<span class="op">==</span><span class="dv">0</span> <span class="op">|</span><span class="st"> </span><span class="kw">is.na</span>(NOHAV),<span class="ot">NA</span>,RTNO),</span>
<span id="cb17-13"><a href="#cb17-13"></a>         <span class="dt">NOHAV=</span><span class="kw">ifelse</span>(NOHAV<span class="op">==</span><span class="dv">0</span>,<span class="ot">NA</span>,NOHAV),</span>
<span id="cb17-14"><a href="#cb17-14"></a>         <span class="dt">NOHAV=</span><span class="kw">ifelse</span>(NOHAV<span class="op">&gt;</span><span class="dv">42</span>,<span class="ot">NA</span>,NOHAV),</span>
<span id="cb17-15"><a href="#cb17-15"></a>         <span class="dt">RTNO=</span><span class="kw">ifelse</span>(<span class="op">!</span>RTNO <span class="op">%in%</span><span class="st"> </span><span class="dv">1</span><span class="op">:</span><span class="dv">10000</span>,<span class="ot">NA</span>,RTNO))</span></code></pre></div>
</div>
<div id="post-qc-traits" class="section level1">
<h1>Post-QC traits</h1>
<div id="harvest-index" class="section level2">
<h2>Harvest index</h2>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a>dbdata&lt;-dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb18-2"><a href="#cb18-2"></a><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">HI=</span>RTWT<span class="op">/</span>(RTWT<span class="op">+</span>SHTWT))</span></code></pre></div>
</div>
<div id="unit-area-traits" class="section level2">
<h2>Unit area traits</h2>
<p>I anticipate this will not be necessary as it will be computed before or during data upload.</p>
<p>For calculating fresh root yield:</p>
<ol style="list-style-type: decimal">
<li><strong>PlotSpacing:</strong> Area in <span class="math inline">\(m^2\)</span> per plant. plotWidth and plotLength metadata would hypothetically provide this info, but is missing for vast majority of trials. Therefore, use info from Fola.</li>
<li><strong>maxNOHAV:</strong> Instead of ExpectedNOHAV. Need to know the max number of plants in the area harvested. For some trials, only the inner (or “net”) plot is harvested, therefore the PlantsPerPlot meta-variable will not suffice. Besides, the PlantsPerPlot information is missing for the vast majority of trials. Instead, use observed max(NOHAV) for each trial. We use this plus the PlotSpacing to calc. the area over which the RTWT was measured. During analysis, variation in the actual number of plants harvested will be accounted for.</li>
</ol>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a>dbdata&lt;-dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb19-2"><a href="#cb19-2"></a><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">PlotSpacing=</span><span class="kw">ifelse</span>(programName<span class="op">!=</span><span class="st">&quot;IITA&quot;</span>,<span class="dv">1</span>,</span>
<span id="cb19-3"><a href="#cb19-3"></a>                              <span class="kw">ifelse</span>(studyYear<span class="op">&lt;</span><span class="dv">2013</span>,<span class="dv">1</span>,</span>
<span id="cb19-4"><a href="#cb19-4"></a>                              <span class="kw">ifelse</span>(TrialType <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;CET&quot;</span>,<span class="st">&quot;GeneticGain&quot;</span>,<span class="st">&quot;ExpCET&quot;</span>),<span class="dv">1</span>,<span class="fl">0.8</span>))))</span>
<span id="cb19-5"><a href="#cb19-5"></a>maxNOHAV_byStudy&lt;-dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb19-6"><a href="#cb19-6"></a><span class="st">  </span><span class="kw">group_by</span>(programName,locationName,studyYear,studyName,studyDesign) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb19-7"><a href="#cb19-7"></a><span class="st">  </span><span class="kw">summarize</span>(<span class="dt">MaxNOHAV=</span><span class="kw">max</span>(NOHAV, <span class="dt">na.rm=</span>T)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb19-8"><a href="#cb19-8"></a><span class="st">  </span><span class="kw">ungroup</span>() <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb19-9"><a href="#cb19-9"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">MaxNOHAV=</span><span class="kw">ifelse</span>(MaxNOHAV<span class="op">==</span><span class="st">&quot;-Inf&quot;</span>,<span class="ot">NA</span>,MaxNOHAV))</span>
<span id="cb19-10"><a href="#cb19-10"></a></span>
<span id="cb19-11"><a href="#cb19-11"></a><span class="kw">write.csv</span>(maxNOHAV_byStudy <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(studyYear),<span class="dt">file=</span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;maxNOHAV_byStudy_NRCRI_2020April27.csv&quot;</span>), <span class="dt">row.names =</span> F)</span></code></pre></div>
<p>Previously, I took these values <em>as is</em>. I am unsatisfied with that. The trial number is small enough I’m going to curate manually below. I hope this gives better yield results.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a>maxNOHAV_byStudy <span class="op">%&lt;&gt;%</span></span>
<span id="cb20-2"><a href="#cb20-2"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">MaxNOHAV=</span><span class="kw">ifelse</span>(studyName<span class="op">==</span><span class="st">&quot;18C2acrossingblockCETubiaja&quot;</span>,<span class="dv">8</span>,MaxNOHAV),</span>
<span id="cb20-3"><a href="#cb20-3"></a>         <span class="dt">MaxNOHAV=</span><span class="kw">ifelse</span>(studyName<span class="op">==</span><span class="st">&quot;13TP1CET518kano&quot;</span>,<span class="dv">5</span>,MaxNOHAV),</span>
<span id="cb20-4"><a href="#cb20-4"></a>         <span class="dt">MaxNOHAV=</span><span class="kw">ifelse</span>(studyName<span class="op">==</span><span class="st">&quot;17C1aAYTGSkano&quot;</span>,<span class="dv">10</span>,MaxNOHAV),</span>
<span id="cb20-5"><a href="#cb20-5"></a>         <span class="dt">MaxNOHAV=</span><span class="kw">ifelse</span>(studyName<span class="op">==</span><span class="st">&quot;18C1bAYTGSOtobi&quot;</span>,<span class="dv">10</span>,MaxNOHAV),</span>
<span id="cb20-6"><a href="#cb20-6"></a>         <span class="dt">MaxNOHAV=</span><span class="kw">ifelse</span>(studyName<span class="op">==</span><span class="st">&quot;16C1aCETnonGSOtobi&quot;</span>,<span class="dv">5</span>,MaxNOHAV),</span>
<span id="cb20-7"><a href="#cb20-7"></a>         <span class="dt">MaxNOHAV=</span><span class="kw">ifelse</span>(studyName<span class="op">==</span><span class="st">&quot;17C1bCETkano&quot;</span>,<span class="dv">5</span>,MaxNOHAV),</span>
<span id="cb20-8"><a href="#cb20-8"></a>         <span class="dt">MaxNOHAV=</span><span class="kw">ifelse</span>(studyName<span class="op">==</span><span class="st">&quot;16C1aCETnonGSOtobi&quot;</span>,<span class="dv">5</span>,MaxNOHAV),</span>
<span id="cb20-9"><a href="#cb20-9"></a>         <span class="dt">MaxNOHAV=</span><span class="kw">ifelse</span>(studyName<span class="op">==</span><span class="st">&quot;18C1bAYTGSset2umu&quot;</span>,<span class="dv">10</span>,MaxNOHAV))</span>
<span id="cb20-10"><a href="#cb20-10"></a></span>
<span id="cb20-11"><a href="#cb20-11"></a><span class="co"># maxNOHAV_byStudy %&gt;%</span></span>
<span id="cb20-12"><a href="#cb20-12"></a><span class="co">#   filter(!is.na(MaxNOHAV),</span></span>
<span id="cb20-13"><a href="#cb20-13"></a><span class="co">#          MaxNOHAV&gt;=10)</span></span>
<span id="cb20-14"><a href="#cb20-14"></a><span class="co"># maxNOHAV_byStudy %&gt;%</span></span>
<span id="cb20-15"><a href="#cb20-15"></a><span class="co">#   filter(grepl(&quot;CET&quot;,studyName))</span></span>
<span id="cb20-16"><a href="#cb20-16"></a><span class="co"># dbdata %&gt;% filter(studyName==&quot;13TP1CET518kano&quot;) %$% table(NOHAV)</span></span>
<span id="cb20-17"><a href="#cb20-17"></a><span class="co"># dbdata %&gt;% filter(studyName==&quot;18C2acrossingblockCETubiaja&quot;) %$% table(NOHAV)</span></span>
<span id="cb20-18"><a href="#cb20-18"></a><span class="co"># dbdata %&gt;% filter(studyName==&quot;17C1aAYTGSkano&quot;) %$% table(NOHAV)</span></span>
<span id="cb20-19"><a href="#cb20-19"></a><span class="co"># dbdata %&gt;% filter(studyName==&quot;18C1bAYTGSOtobi&quot;) %$% table(NOHAV)</span></span>
<span id="cb20-20"><a href="#cb20-20"></a><span class="co"># dbdata %&gt;% filter(studyName==&quot;16C1aCETnonGSOtobi&quot;) %$% table(NOHAV)</span></span>
<span id="cb20-21"><a href="#cb20-21"></a><span class="co"># dbdata %&gt;% filter(studyName==&quot;17C1aAYTGSumu&quot;) %$% table(NOHAV)</span></span>
<span id="cb20-22"><a href="#cb20-22"></a><span class="co"># dbdata %&gt;% filter(studyName==&quot;17C1bCETkano&quot;) %$% table(NOHAV)</span></span>
<span id="cb20-23"><a href="#cb20-23"></a><span class="co"># dbdata %&gt;% filter(studyName==&quot;18C1bAYTGSset2umu&quot;) %$% table(NOHAV)</span></span></code></pre></div>
<p>18C2acrossingblockCETubiaja …10… Lydia says 8. 13TP1CET518kano… 5<br />
18C1bAYTGSOtobi… looks like it should be 5, but DB says 4 x 4 m plots.<br />
16C1aCETnonGSOtobi… 10 (says plot length 8)<br />
18C1bAYTGSset2umu… 10</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a><span class="co"># I log transform yield traits </span></span>
<span id="cb21-2"><a href="#cb21-2"></a><span class="co"># to satisfy homoskedastic residuals assumption </span></span>
<span id="cb21-3"><a href="#cb21-3"></a><span class="co"># of linear mixed models</span></span>
<span id="cb21-4"><a href="#cb21-4"></a>dbdata&lt;-<span class="kw">left_join</span>(dbdata,maxNOHAV_byStudy) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb21-5"><a href="#cb21-5"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">RTWT=</span><span class="kw">ifelse</span>(NOHAV<span class="op">&gt;</span>MaxNOHAV,<span class="ot">NA</span>,RTWT),</span>
<span id="cb21-6"><a href="#cb21-6"></a>         <span class="dt">SHTWT=</span><span class="kw">ifelse</span>(NOHAV<span class="op">&gt;</span>MaxNOHAV,<span class="ot">NA</span>,SHTWT),</span>
<span id="cb21-7"><a href="#cb21-7"></a>         <span class="dt">RTNO=</span><span class="kw">ifelse</span>(NOHAV<span class="op">&gt;</span>MaxNOHAV,<span class="ot">NA</span>,RTNO),</span>
<span id="cb21-8"><a href="#cb21-8"></a>         <span class="dt">HI=</span><span class="kw">ifelse</span>(NOHAV<span class="op">&gt;</span>MaxNOHAV,<span class="ot">NA</span>,HI),</span>
<span id="cb21-9"><a href="#cb21-9"></a>         <span class="dt">logFYLD=</span><span class="kw">log</span>(RTWT<span class="op">/</span>(MaxNOHAV<span class="op">*</span>PlotSpacing)<span class="op">*</span><span class="dv">10</span>),</span>
<span id="cb21-10"><a href="#cb21-10"></a>         <span class="dt">logTOPYLD=</span><span class="kw">log</span>(SHTWT<span class="op">/</span>(MaxNOHAV<span class="op">*</span>PlotSpacing)<span class="op">*</span><span class="dv">10</span>),</span>
<span id="cb21-11"><a href="#cb21-11"></a>         <span class="dt">logRTNO=</span><span class="kw">log</span>(RTNO),</span>
<span id="cb21-12"><a href="#cb21-12"></a>         <span class="dt">PropNOHAV=</span>NOHAV<span class="op">/</span>MaxNOHAV) </span>
<span id="cb21-13"><a href="#cb21-13"></a></span>
<span id="cb21-14"><a href="#cb21-14"></a><span class="co"># remove non transformed / per-plot (instead of per area) traits</span></span>
<span id="cb21-15"><a href="#cb21-15"></a>dbdata <span class="op">%&lt;&gt;%</span><span class="st"> </span><span class="kw">select</span>(<span class="op">-</span>RTWT,<span class="op">-</span>SHTWT,<span class="op">-</span>RTNO)</span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1"></a><span class="co"># dbdata %&gt;% </span></span>
<span id="cb22-2"><a href="#cb22-2"></a><span class="co">#   distinct(studyName,MaxNOHAV) %&gt;% </span></span>
<span id="cb22-3"><a href="#cb22-3"></a><span class="co">#   filter(!is.na(MaxNOHAV)) %&gt;% arrange(MaxNOHAV)</span></span></code></pre></div>
</div>
<div id="season-wide-mean-cmds" class="section level2">
<h2>Season-wide mean CMDS</h2>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a>dbdata&lt;-dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb23-2"><a href="#cb23-2"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">MCMDS=</span><span class="kw">rowMeans</span>(.[,<span class="kw">c</span>(<span class="st">&quot;CMD1S&quot;</span>,<span class="st">&quot;CMD3S&quot;</span>,<span class="st">&quot;CMD6S&quot;</span>,<span class="st">&quot;CMD9S&quot;</span>)], <span class="dt">na.rm =</span> T)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb23-3"><a href="#cb23-3"></a><span class="st">  </span><span class="kw">select</span>(<span class="op">-</span>CMD1S,<span class="op">-</span>CMD3S,<span class="op">-</span>CMD6S,<span class="op">-</span>CMD9S)</span></code></pre></div>
</div>
</div>
<div id="assign-genos-to-phenos" class="section level1">
<h1>Assign genos to phenos</h1>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1"></a><span class="kw">library</span>(tidyverse); <span class="kw">library</span>(magrittr)</span>
<span id="cb24-2"><a href="#cb24-2"></a>gbs2phenoMaster&lt;-dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-3"><a href="#cb24-3"></a><span class="st">  </span><span class="kw">select</span>(germplasmName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-4"><a href="#cb24-4"></a><span class="st">  </span>distinct <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-5"><a href="#cb24-5"></a><span class="st">  </span><span class="kw">left_join</span>(<span class="kw">read.csv</span>(here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;NRCRI_GBStoPhenoMaster_40318.csv&quot;</span>), </span>
<span id="cb24-6"><a href="#cb24-6"></a>                     <span class="dt">stringsAsFactors =</span> F)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-7"><a href="#cb24-7"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">FullSampleName=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;C2a&quot;</span>,germplasmName,<span class="dt">ignore.case =</span> T) <span class="op">&amp;</span><span class="st"> </span></span>
<span id="cb24-8"><a href="#cb24-8"></a><span class="st">                                 </span><span class="kw">is.na</span>(FullSampleName),germplasmName,FullSampleName)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-9"><a href="#cb24-9"></a><span class="st">  </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(FullSampleName)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-10"><a href="#cb24-10"></a><span class="st">  </span><span class="kw">select</span>(germplasmName,FullSampleName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-11"><a href="#cb24-11"></a><span class="st">  </span><span class="kw">bind_rows</span>(dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-12"><a href="#cb24-12"></a><span class="st">              </span><span class="kw">select</span>(germplasmName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-13"><a href="#cb24-13"></a><span class="st">              </span>distinct <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-14"><a href="#cb24-14"></a><span class="st">              </span><span class="kw">left_join</span>(<span class="kw">read.csv</span>(here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;IITA_GBStoPhenoMaster_33018.csv&quot;</span>), </span>
<span id="cb24-15"><a href="#cb24-15"></a>                                 <span class="dt">stringsAsFactors =</span> F)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-16"><a href="#cb24-16"></a><span class="st">              </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(FullSampleName)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-17"><a href="#cb24-17"></a><span class="st">              </span><span class="kw">select</span>(germplasmName,FullSampleName)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-18"><a href="#cb24-18"></a><span class="st">  </span><span class="kw">bind_rows</span>(dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-19"><a href="#cb24-19"></a><span class="st">              </span><span class="kw">select</span>(germplasmName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-20"><a href="#cb24-20"></a><span class="st">              </span>distinct <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-21"><a href="#cb24-21"></a><span class="st">              </span><span class="kw">left_join</span>(<span class="kw">read.csv</span>(here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;GBSdataMasterList_31818.csv&quot;</span>), </span>
<span id="cb24-22"><a href="#cb24-22"></a>                                 <span class="dt">stringsAsFactors =</span> F) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-23"><a href="#cb24-23"></a><span class="st">                          </span><span class="kw">select</span>(DNASample,FullSampleName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-24"><a href="#cb24-24"></a><span class="st">                          </span><span class="kw">rename</span>(<span class="dt">germplasmName=</span>DNASample)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-25"><a href="#cb24-25"></a><span class="st">              </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(FullSampleName)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-26"><a href="#cb24-26"></a><span class="st">              </span><span class="kw">select</span>(germplasmName,FullSampleName)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-27"><a href="#cb24-27"></a><span class="st">  </span><span class="kw">bind_rows</span>(dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-28"><a href="#cb24-28"></a><span class="st">              </span><span class="kw">select</span>(germplasmName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-29"><a href="#cb24-29"></a><span class="st">              </span>distinct <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-30"><a href="#cb24-30"></a><span class="st">              </span><span class="kw">mutate</span>(<span class="dt">germplasmSynonyms=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;^UG&quot;</span>,germplasmName,<span class="dt">ignore.case =</span> T),</span>
<span id="cb24-31"><a href="#cb24-31"></a>                                              <span class="kw">gsub</span>(<span class="st">&quot;UG&quot;</span>,<span class="st">&quot;Ug&quot;</span>,germplasmName),germplasmName)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-32"><a href="#cb24-32"></a><span class="st">              </span><span class="kw">left_join</span>(<span class="kw">read.csv</span>(here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;GBSdataMasterList_31818.csv&quot;</span>), </span>
<span id="cb24-33"><a href="#cb24-33"></a>                                 <span class="dt">stringsAsFactors =</span> F) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-34"><a href="#cb24-34"></a><span class="st">                          </span><span class="kw">select</span>(DNASample,FullSampleName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-35"><a href="#cb24-35"></a><span class="st">                          </span><span class="kw">rename</span>(<span class="dt">germplasmSynonyms=</span>DNASample)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-36"><a href="#cb24-36"></a><span class="st">              </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(FullSampleName)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-37"><a href="#cb24-37"></a><span class="st">              </span><span class="kw">select</span>(germplasmName,FullSampleName)) <span class="op">%&gt;%</span><span class="st">  </span></span>
<span id="cb24-38"><a href="#cb24-38"></a><span class="st">  </span><span class="kw">bind_rows</span>(dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-39"><a href="#cb24-39"></a><span class="st">              </span><span class="kw">select</span>(germplasmName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-40"><a href="#cb24-40"></a><span class="st">              </span>distinct <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-41"><a href="#cb24-41"></a><span class="st">              </span><span class="kw">mutate</span>(<span class="dt">germplasmSynonyms=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;^TZ&quot;</span>,germplasmName,</span>
<span id="cb24-42"><a href="#cb24-42"></a>                                                    <span class="dt">ignore.case =</span> T),</span>
<span id="cb24-43"><a href="#cb24-43"></a>                                              <span class="kw">gsub</span>(<span class="st">&quot;TZ&quot;</span>,<span class="st">&quot;&quot;</span>,germplasmName),germplasmName)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-44"><a href="#cb24-44"></a><span class="st">              </span><span class="kw">left_join</span>(<span class="kw">read.csv</span>(here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;GBSdataMasterList_31818.csv&quot;</span>), </span>
<span id="cb24-45"><a href="#cb24-45"></a>                                 <span class="dt">stringsAsFactors =</span> F) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-46"><a href="#cb24-46"></a><span class="st">                          </span><span class="kw">select</span>(DNASample,FullSampleName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-47"><a href="#cb24-47"></a><span class="st">                          </span><span class="kw">rename</span>(<span class="dt">germplasmSynonyms=</span>DNASample)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-48"><a href="#cb24-48"></a><span class="st">              </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(FullSampleName)) <span class="op">%&gt;%</span></span>
<span id="cb24-49"><a href="#cb24-49"></a><span class="st">              </span><span class="kw">select</span>(germplasmName,FullSampleName)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-50"><a href="#cb24-50"></a><span class="st">  </span>distinct <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-51"><a href="#cb24-51"></a><span class="st">  </span><span class="kw">left_join</span>(<span class="kw">read.csv</span>(here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;GBSdataMasterList_31818.csv&quot;</span>), </span>
<span id="cb24-52"><a href="#cb24-52"></a>                     <span class="dt">stringsAsFactors =</span> F) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-53"><a href="#cb24-53"></a><span class="st">              </span><span class="kw">select</span>(FullSampleName,OrigKeyFile,Institute) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-54"><a href="#cb24-54"></a><span class="st">              </span><span class="kw">rename</span>(<span class="dt">OriginOfSample=</span>Institute)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-55"><a href="#cb24-55"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">OrigKeyFile=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;C2a&quot;</span>,germplasmName,<span class="dt">ignore.case =</span> T),</span>
<span id="cb24-56"><a href="#cb24-56"></a>                            <span class="kw">ifelse</span>(<span class="kw">is.na</span>(OrigKeyFile),<span class="st">&quot;LavalGBS&quot;</span>,OrigKeyFile),</span>
<span id="cb24-57"><a href="#cb24-57"></a>                            OrigKeyFile),</span>
<span id="cb24-58"><a href="#cb24-58"></a>         <span class="dt">OriginOfSample=</span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="st">&quot;C2a&quot;</span>,germplasmName,<span class="dt">ignore.case =</span> T),</span>
<span id="cb24-59"><a href="#cb24-59"></a>                               <span class="kw">ifelse</span>(<span class="kw">is.na</span>(OriginOfSample),<span class="st">&quot;NRCRI&quot;</span>,OriginOfSample),</span>
<span id="cb24-60"><a href="#cb24-60"></a>                               OriginOfSample))</span>
<span id="cb24-61"><a href="#cb24-61"></a><span class="kw">nrow</span>(gbs2phenoMaster)</span></code></pre></div>
<pre><code>[1] 3219</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a>gbs2phenoMaster <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">count</span>(OriginOfSample)</span></code></pre></div>
<pre><code># A tibble: 4 x 2
  OriginOfSample     n
  &lt;chr&gt;          &lt;int&gt;
1 IITA             419
2 NRCRI           2769
3 TARI               4
4 &lt;NA&gt;              27</code></pre>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1"></a><span class="co"># gbs2phenoMaster %&gt;% filter(grepl(&quot;C2a&quot;,germplasmName,ignore.case = T))</span></span>
<span id="cb28-2"><a href="#cb28-2"></a></span>
<span id="cb28-3"><a href="#cb28-3"></a><span class="co"># first, filter to just program-DNAorigin matches</span></span>
<span id="cb28-4"><a href="#cb28-4"></a>germNamesWithGenos&lt;-dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb28-5"><a href="#cb28-5"></a><span class="st">  </span><span class="kw">select</span>(programName,germplasmName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb28-6"><a href="#cb28-6"></a><span class="st">  </span>distinct <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb28-7"><a href="#cb28-7"></a><span class="st">  </span><span class="kw">left_join</span>(gbs2phenoMaster) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb28-8"><a href="#cb28-8"></a><span class="st">  </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(FullSampleName))</span>
<span id="cb28-9"><a href="#cb28-9"></a><span class="kw">nrow</span>(germNamesWithGenos) <span class="co"># 3077</span></span></code></pre></div>
<pre><code>[1] 3219</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1"></a><span class="co"># program-germNames with locally sourced GBS samples</span></span>
<span id="cb30-2"><a href="#cb30-2"></a>germNamesWithGenos_HasLocalSourcedGBS&lt;-germNamesWithGenos <span class="op">%&gt;%</span><span class="st"> </span><span class="co">#count(OriginOfSample)</span></span>
<span id="cb30-3"><a href="#cb30-3"></a><span class="st">  </span><span class="kw">filter</span>(programName<span class="op">==</span>OriginOfSample) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb30-4"><a href="#cb30-4"></a><span class="st">  </span><span class="kw">select</span>(programName,germplasmName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb30-5"><a href="#cb30-5"></a><span class="st">  </span><span class="kw">semi_join</span>(germNamesWithGenos,.) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb30-6"><a href="#cb30-6"></a><span class="st">  </span><span class="kw">group_by</span>(programName,germplasmName) <span class="op">%&gt;%</span><span class="st"> </span><span class="co"># select one DNA per germplasmName per program</span></span>
<span id="cb30-7"><a href="#cb30-7"></a><span class="st">  </span><span class="kw">slice</span>(<span class="dv">1</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() </span>
<span id="cb30-8"><a href="#cb30-8"></a><span class="kw">nrow</span>(germNamesWithGenos_HasLocalSourcedGBS) <span class="co"># 2592</span></span></code></pre></div>
<pre><code>[1] 2696</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1"></a><span class="co"># the rest (program-germNames) with GBS but coming from a different breeding program</span></span>
<span id="cb32-2"><a href="#cb32-2"></a>germNamesWithGenos_NoLocalSourcedGBS&lt;-germNamesWithGenos <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb32-3"><a href="#cb32-3"></a><span class="st">  </span><span class="kw">filter</span>(programName<span class="op">==</span>OriginOfSample) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb32-4"><a href="#cb32-4"></a><span class="st">  </span><span class="kw">select</span>(programName,germplasmName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb32-5"><a href="#cb32-5"></a><span class="st">  </span><span class="kw">anti_join</span>(germNamesWithGenos,.) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb32-6"><a href="#cb32-6"></a><span class="st">  </span><span class="co"># select one DNA per germplasmName per program</span></span>
<span id="cb32-7"><a href="#cb32-7"></a><span class="st">  </span><span class="kw">group_by</span>(programName,germplasmName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb32-8"><a href="#cb32-8"></a><span class="st">  </span><span class="kw">slice</span>(<span class="dv">1</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ungroup</span>() </span>
<span id="cb32-9"><a href="#cb32-9"></a><span class="kw">nrow</span>(germNamesWithGenos_NoLocalSourcedGBS) <span class="co"># 202</span></span></code></pre></div>
<pre><code>[1] 212</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1"></a>gbsForPhenos&lt;-<span class="kw">bind_rows</span>(germNamesWithGenos_HasLocalSourcedGBS,</span>
<span id="cb34-2"><a href="#cb34-2"></a>                        germNamesWithGenos_NoLocalSourcedGBS) </span>
<span id="cb34-3"><a href="#cb34-3"></a><span class="kw">nrow</span>(gbsForPhenos) <span class="co"># 2794</span></span></code></pre></div>
<pre><code>[1] 2908</code></pre>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1"></a>dbdata <span class="op">%&lt;&gt;%</span><span class="st"> </span></span>
<span id="cb36-2"><a href="#cb36-2"></a><span class="st">    </span><span class="kw">left_join</span>(gbsForPhenos) </span>
<span id="cb36-3"><a href="#cb36-3"></a></span>
<span id="cb36-4"><a href="#cb36-4"></a><span class="co"># Create a new identifier, GID</span></span>
<span id="cb36-5"><a href="#cb36-5"></a><span class="co">## Equals the value SNP data name (FullSampleName) </span></span>
<span id="cb36-6"><a href="#cb36-6"></a><span class="co">## else germplasmName if no SNP data</span></span>
<span id="cb36-7"><a href="#cb36-7"></a>dbdata <span class="op">%&lt;&gt;%</span><span class="st"> </span></span>
<span id="cb36-8"><a href="#cb36-8"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">GID=</span><span class="kw">ifelse</span>(<span class="kw">is.na</span>(FullSampleName),germplasmName,FullSampleName))</span></code></pre></div>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1"></a><span class="co">#  going to check against SNP data</span></span>
<span id="cb37-2"><a href="#cb37-2"></a>snps&lt;-<span class="kw">readRDS</span>(<span class="dt">file=</span><span class="kw">url</span>(<span class="kw">paste0</span>(<span class="st">&quot;ftp://ftp.cassavabase.org/marnin_datasets/NGC_BigData/&quot;</span>,</span>
<span id="cb37-3"><a href="#cb37-3"></a>                              <span class="st">&quot;DosageMatrix_RefPanelAndGSprogeny_ReadyForGP_73019.rds&quot;</span>)))</span>
<span id="cb37-4"><a href="#cb37-4"></a>rownames_snps&lt;-<span class="kw">rownames</span>(snps); <span class="kw">rm</span>(snps); <span class="kw">gc</span>()</span></code></pre></div>
<pre><code>          used (Mb) gc trigger   (Mb) limit (Mb)  max used   (Mb)
Ncells 1089998 58.3    2343228  125.2         NA   2103047  112.4
Vcells 3231727 24.7  726581331 5543.4     102400 755469121 5763.8</code></pre>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1"></a><span class="co"># current matches to SNP data</span></span>
<span id="cb39-2"><a href="#cb39-2"></a>dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb39-3"><a href="#cb39-3"></a><span class="st">  </span><span class="kw">distinct</span>(GID,germplasmName,FullSampleName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb39-4"><a href="#cb39-4"></a><span class="st">  </span><span class="kw">semi_join</span>(<span class="kw">tibble</span>(<span class="dt">GID=</span>rownames_snps)) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">nrow</span>() <span class="co">#1340</span></span></code></pre></div>
<pre><code>[1] 1340</code></pre>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1"></a>dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb41-2"><a href="#cb41-2"></a><span class="st">  </span><span class="kw">distinct</span>(GID,germplasmName,FullSampleName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb41-3"><a href="#cb41-3"></a><span class="st">  </span><span class="kw">semi_join</span>(<span class="kw">tibble</span>(<span class="dt">GID=</span>rownames_snps)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb41-4"><a href="#cb41-4"></a><span class="st">  </span><span class="kw">filter</span>(<span class="kw">grepl</span>(<span class="st">&quot;c1&quot;</span>,GID,<span class="dt">ignore.case =</span> F)) <span class="co"># no C1 clones currently match</span></span></code></pre></div>
<pre><code># A tibble: 0 x 3
# … with 3 variables: GID &lt;chr&gt;, germplasmName &lt;chr&gt;, FullSampleName &lt;chr&gt;</code></pre>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1"></a>dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb43-2"><a href="#cb43-2"></a><span class="st">  </span><span class="kw">distinct</span>(GID,germplasmName,FullSampleName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb43-3"><a href="#cb43-3"></a><span class="st">  </span><span class="kw">semi_join</span>(<span class="kw">tibble</span>(<span class="dt">GID=</span>rownames_snps)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb43-4"><a href="#cb43-4"></a><span class="st">  </span><span class="kw">filter</span>(<span class="kw">grepl</span>(<span class="st">&quot;c2&quot;</span>,GID,<span class="dt">ignore.case =</span> F)) <span class="co"># no C2 clones either</span></span></code></pre></div>
<pre><code># A tibble: 0 x 3
# … with 3 variables: GID &lt;chr&gt;, germplasmName &lt;chr&gt;, FullSampleName &lt;chr&gt;</code></pre>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1"></a>dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb45-2"><a href="#cb45-2"></a><span class="st">  </span><span class="kw">distinct</span>(GID,germplasmName,FullSampleName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb45-3"><a href="#cb45-3"></a><span class="st">  </span><span class="kw">anti_join</span>(<span class="kw">tibble</span>(<span class="dt">GID=</span>rownames_snps)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb45-4"><a href="#cb45-4"></a><span class="st">  </span><span class="kw">filter</span>(<span class="kw">grepl</span>(<span class="st">&quot;c1|c2&quot;</span>,GID,<span class="dt">ignore.case =</span> T)) <span class="co"># definitely there are both C1 and C2 phenotypes</span></span></code></pre></div>
<pre><code># A tibble: 2,404 x 3
   GID             germplasmName   FullSampleName
   &lt;chr&gt;           &lt;chr&gt;           &lt;chr&gt;         
 1 NR16F100C1bP001 NR16F100C1bP001 &lt;NA&gt;          
 2 NR16F100C1bP002 NR16F100C1bP002 &lt;NA&gt;          
 3 NR16F104C1bP001 NR16F104C1bP001 &lt;NA&gt;          
 4 NR16F105C1bP001 NR16F105C1bP001 &lt;NA&gt;          
 5 NR16F105C1bP002 NR16F105C1bP002 &lt;NA&gt;          
 6 NR16F106C1bP001 NR16F106C1bP001 &lt;NA&gt;          
 7 NR16F107C1bP002 NR16F107C1bP002 &lt;NA&gt;          
 8 NR16F107C1bP004 NR16F107C1bP004 &lt;NA&gt;          
 9 NR16F108C1bP001 NR16F108C1bP001 &lt;NA&gt;          
10 NR16F109C1bP001 NR16F109C1bP001 &lt;NA&gt;          
# … with 2,394 more rows</code></pre>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1"></a><span class="co"># and there are C1 and C2 genotypes</span></span>
<span id="cb47-2"><a href="#cb47-2"></a>rownames_snps <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">grep</span>(<span class="st">&quot;c1&quot;</span>,.,<span class="dt">value =</span> T,<span class="dt">ignore.case =</span> T) <span class="op">%&gt;%</span><span class="st"> </span>length <span class="co"># [1] 1762</span></span></code></pre></div>
<pre><code>[1] 1762</code></pre>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1"></a>rownames_snps <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">grep</span>(<span class="st">&quot;c2&quot;</span>,.,<span class="dt">value =</span> T,<span class="dt">ignore.case =</span> T) <span class="op">%&gt;%</span><span class="st"> </span>length <span class="co"># [1] 4291 </span></span></code></pre></div>
<pre><code>[1] 4291</code></pre>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1"></a>germ2snps&lt;-dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-2"><a href="#cb51-2"></a><span class="st">  </span><span class="kw">distinct</span>(germplasmName,FullSampleName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-3"><a href="#cb51-3"></a><span class="st">  </span><span class="kw">semi_join</span>(<span class="kw">tibble</span>(<span class="dt">FullSampleName=</span>rownames_snps)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-4"><a href="#cb51-4"></a><span class="st">  </span><span class="kw">bind_rows</span>(dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-5"><a href="#cb51-5"></a><span class="st">              </span><span class="kw">distinct</span>(germplasmName,FullSampleName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-6"><a href="#cb51-6"></a><span class="st">              </span><span class="kw">anti_join</span>(<span class="kw">tibble</span>(<span class="dt">FullSampleName=</span>rownames_snps)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-7"><a href="#cb51-7"></a><span class="st">              </span><span class="kw">filter</span>(<span class="kw">grepl</span>(<span class="st">&quot;c1a&quot;</span>,germplasmName,<span class="dt">ignore.case =</span> T)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-8"><a href="#cb51-8"></a><span class="st">              </span><span class="kw">select</span>(<span class="op">-</span>FullSampleName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-9"><a href="#cb51-9"></a><span class="st">              </span><span class="kw">left_join</span>(<span class="kw">tibble</span>(<span class="dt">FullSampleName=</span>rownames_snps) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-10"><a href="#cb51-10"></a><span class="st">                          </span><span class="kw">filter</span>(<span class="kw">grepl</span>(<span class="st">&quot;c1a&quot;</span>,FullSampleName,<span class="dt">ignore.case =</span> T)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-11"><a href="#cb51-11"></a><span class="st">                          </span><span class="kw">separate</span>(FullSampleName,<span class="kw">c</span>(<span class="st">&quot;dartID&quot;</span>,<span class="st">&quot;germplasmName&quot;</span>),<span class="st">&quot;</span><span class="ch">\\</span><span class="st">.</span><span class="ch">\\</span><span class="st">.</span><span class="ch">\\</span><span class="st">.&quot;</span>,<span class="dt">extra =</span> <span class="st">&#39;merge&#39;</span>,<span class="dt">remove =</span> F) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-12"><a href="#cb51-12"></a><span class="st">                          </span><span class="kw">select</span>(<span class="op">-</span>dartID))) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-13"><a href="#cb51-13"></a><span class="st">  </span><span class="kw">bind_rows</span>(dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-14"><a href="#cb51-14"></a><span class="st">              </span><span class="kw">distinct</span>(germplasmName,FullSampleName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-15"><a href="#cb51-15"></a><span class="st">              </span><span class="kw">anti_join</span>(<span class="kw">tibble</span>(<span class="dt">FullSampleName=</span>rownames_snps)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-16"><a href="#cb51-16"></a><span class="st">              </span><span class="kw">filter</span>(<span class="kw">grepl</span>(<span class="st">&quot;C1b&quot;</span>,germplasmName,<span class="dt">ignore.case =</span> T)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-17"><a href="#cb51-17"></a><span class="st">              </span><span class="kw">filter</span>(<span class="kw">grepl</span>(<span class="st">&quot;NR16C1b&quot;</span>,germplasmName,<span class="dt">ignore.case =</span> T)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-18"><a href="#cb51-18"></a><span class="st">              </span><span class="kw">select</span>(<span class="op">-</span>FullSampleName) <span class="op">%&gt;%</span></span>
<span id="cb51-19"><a href="#cb51-19"></a><span class="st">              </span><span class="kw">left_join</span>(<span class="kw">tibble</span>(<span class="dt">FullSampleName=</span>rownames_snps) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-20"><a href="#cb51-20"></a><span class="st">                          </span><span class="kw">filter</span>(<span class="kw">grepl</span>(<span class="st">&quot;c1b&quot;</span>,FullSampleName,<span class="dt">ignore.case =</span> T)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-21"><a href="#cb51-21"></a><span class="st">                          </span><span class="kw">separate</span>(FullSampleName,<span class="kw">c</span>(<span class="st">&quot;germplasmName&quot;</span>,<span class="st">&quot;GBS_ID&quot;</span>),<span class="st">&quot;:&quot;</span>,<span class="dt">extra =</span> <span class="st">&#39;merge&#39;</span>,<span class="dt">remove =</span> F) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-22"><a href="#cb51-22"></a><span class="st">                          </span><span class="kw">select</span>(<span class="op">-</span>GBS_ID) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-23"><a href="#cb51-23"></a><span class="st">                          </span><span class="kw">mutate</span>(<span class="dt">germplasmName=</span><span class="kw">gsub</span>(<span class="st">&quot;C1b&quot;</span>,<span class="st">&quot;&quot;</span>,germplasmName),</span>
<span id="cb51-24"><a href="#cb51-24"></a>                                 <span class="dt">germplasmName=</span><span class="kw">paste0</span>(<span class="st">&quot;NR16C1b&quot;</span>,germplasmName)))) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-25"><a href="#cb51-25"></a><span class="st">  </span><span class="kw">bind_rows</span>(dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-26"><a href="#cb51-26"></a><span class="st">              </span><span class="kw">distinct</span>(germplasmName,FullSampleName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-27"><a href="#cb51-27"></a><span class="st">              </span><span class="kw">anti_join</span>(<span class="kw">tibble</span>(<span class="dt">FullSampleName=</span>rownames_snps)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-28"><a href="#cb51-28"></a><span class="st">              </span><span class="kw">filter</span>(<span class="kw">grepl</span>(<span class="st">&quot;C1b&quot;</span>,germplasmName,<span class="dt">ignore.case =</span> T)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-29"><a href="#cb51-29"></a><span class="st">              </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">grepl</span>(<span class="st">&quot;NR16C1b&quot;</span>,germplasmName,<span class="dt">ignore.case =</span> T)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-30"><a href="#cb51-30"></a><span class="st">              </span><span class="kw">select</span>(<span class="op">-</span>FullSampleName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-31"><a href="#cb51-31"></a><span class="st">              </span><span class="kw">left_join</span>(<span class="kw">tibble</span>(<span class="dt">FullSampleName=</span>rownames_snps) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-32"><a href="#cb51-32"></a><span class="st">                          </span><span class="kw">filter</span>(<span class="kw">grepl</span>(<span class="st">&quot;c1b&quot;</span>,FullSampleName,<span class="dt">ignore.case =</span> T)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-33"><a href="#cb51-33"></a><span class="st">                          </span><span class="kw">separate</span>(FullSampleName,<span class="kw">c</span>(<span class="st">&quot;germplasmName&quot;</span>,<span class="st">&quot;GBS_ID&quot;</span>),<span class="st">&quot;:&quot;</span>,<span class="dt">extra =</span> <span class="st">&#39;merge&#39;</span>,<span class="dt">remove =</span> F) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-34"><a href="#cb51-34"></a><span class="st">                          </span><span class="kw">select</span>(<span class="op">-</span>GBS_ID) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-35"><a href="#cb51-35"></a><span class="st">                          </span><span class="kw">mutate</span>(<span class="dt">germplasmName=</span><span class="kw">paste0</span>(<span class="st">&quot;NR16&quot;</span>,germplasmName)))) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-36"><a href="#cb51-36"></a><span class="st">  </span><span class="kw">bind_rows</span>(dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-37"><a href="#cb51-37"></a><span class="st">              </span><span class="kw">distinct</span>(germplasmName,FullSampleName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-38"><a href="#cb51-38"></a><span class="st">              </span><span class="kw">anti_join</span>(<span class="kw">tibble</span>(<span class="dt">FullSampleName=</span>rownames_snps)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-39"><a href="#cb51-39"></a><span class="st">              </span><span class="kw">filter</span>(<span class="kw">grepl</span>(<span class="st">&quot;c2&quot;</span>,germplasmName,<span class="dt">ignore.case =</span> T)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-40"><a href="#cb51-40"></a><span class="st">              </span><span class="kw">select</span>(<span class="op">-</span>FullSampleName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-41"><a href="#cb51-41"></a><span class="st">              </span><span class="kw">left_join</span>(<span class="kw">tibble</span>(<span class="dt">FullSampleName=</span>rownames_snps) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-42"><a href="#cb51-42"></a><span class="st">                          </span><span class="kw">filter</span>(<span class="kw">grepl</span>(<span class="st">&quot;c2&quot;</span>,FullSampleName,<span class="dt">ignore.case =</span> T),</span>
<span id="cb51-43"><a href="#cb51-43"></a>                                 <span class="kw">grepl</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">.</span><span class="ch">\\</span><span class="st">.</span><span class="ch">\\</span><span class="st">.&quot;</span>,FullSampleName)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-44"><a href="#cb51-44"></a><span class="st">                          </span><span class="kw">separate</span>(FullSampleName,<span class="kw">c</span>(<span class="st">&quot;dartID&quot;</span>,<span class="st">&quot;germplasmName&quot;</span>),<span class="st">&quot;</span><span class="ch">\\</span><span class="st">.</span><span class="ch">\\</span><span class="st">.</span><span class="ch">\\</span><span class="st">.&quot;</span>,<span class="dt">extra =</span> <span class="st">&#39;merge&#39;</span>,<span class="dt">remove =</span> F) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-45"><a href="#cb51-45"></a><span class="st">                          </span><span class="kw">select</span>(<span class="op">-</span>dartID))) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-46"><a href="#cb51-46"></a><span class="st">  </span>distinct</span>
<span id="cb51-47"><a href="#cb51-47"></a>germ2snps <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb51-48"><a href="#cb51-48"></a><span class="st">  </span><span class="kw">count</span>(germplasmName) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(<span class="kw">desc</span>(n))</span></code></pre></div>
<pre><code># A tibble: 3,744 x 2
   germplasmName       n
   &lt;chr&gt;           &lt;int&gt;
 1 NR16C1bF185P001     4
 2 NR16F185C1bP001     4
 3 NR16C1bF170P019     3
 4 NR16C1bF170P020     3
 5 NR16C1bF180P002     3
 6 NR16F180C1bP002     3
 7 NR16F182C1bP001     3
 8 NR16C1bF171P002     2
 9 NR16C1bF179P001     2
10 NR16C1bF179P002     2
# … with 3,734 more rows</code></pre>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1"></a>germ2snps <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb53-2"><a href="#cb53-2"></a><span class="st">  </span><span class="kw">count</span>(FullSampleName) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">arrange</span>(<span class="kw">desc</span>(n))</span></code></pre></div>
<pre><code># A tibble: 3,284 x 2
   FullSampleName                     n
   &lt;chr&gt;                          &lt;int&gt;
 1 &lt;NA&gt;                             341
 2 F116C1bP006:CA7RRANXX:3:499841     3
 3 F123C1bP002:CA7RRANXX:4:499895     3
 4 F123C1bP006:CA7RRANXX:4:499899     3
 5 F12C1bP018:CABV7ANXX:5:503932      3
 6 F135C1bP001:CABJYANXX:5:505238     3
 7 F136C1bP002:CABJYANXX:5:505240     3
 8 F140C1bP001:CABJYANXX:5:505270     3
 9 F152C1bP010:CA7RRANXX:5:499961     3
10 F152C1bP014:CA7RRANXX:5:499965     3
# … with 3,274 more rows</code></pre>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1"></a><span class="kw">length</span>(<span class="kw">unique</span>(dbdata<span class="op">$</span>FullSampleName)) <span class="co"># [1] 3234</span></span></code></pre></div>
<pre><code>[1] 2895</code></pre>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1"></a><span class="kw">table</span>(<span class="kw">unique</span>(dbdata<span class="op">$</span>FullSampleName) <span class="op">%in%</span><span class="st"> </span>rownames_snps)</span></code></pre></div>
<pre><code>
FALSE  TRUE 
 1568  1327 </code></pre>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1"></a><span class="co"># FALSE  TRUE </span></span>
<span id="cb59-2"><a href="#cb59-2"></a><span class="co">#  1907  1327 </span></span>
<span id="cb59-3"><a href="#cb59-3"></a>dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb59-4"><a href="#cb59-4"></a><span class="st">  </span><span class="kw">select</span>(<span class="op">-</span>GID,<span class="op">-</span>FullSampleName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb59-5"><a href="#cb59-5"></a><span class="st">  </span><span class="kw">left_join</span>(germ2snps) <span class="op">%$%</span></span>
<span id="cb59-6"><a href="#cb59-6"></a><span class="st">  </span><span class="kw">length</span>(<span class="kw">unique</span>(FullSampleName)) <span class="co"># [1] 6270</span></span></code></pre></div>
<pre><code>[1] 3284</code></pre>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1"></a>dbdata <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb61-2"><a href="#cb61-2"></a><span class="st">  </span><span class="kw">select</span>(<span class="op">-</span>GID,<span class="op">-</span>FullSampleName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb61-3"><a href="#cb61-3"></a><span class="st">  </span><span class="kw">left_join</span>(germ2snps) <span class="op">%$%</span></span>
<span id="cb61-4"><a href="#cb61-4"></a><span class="st">  </span><span class="kw">table</span>(<span class="kw">unique</span>(FullSampleName) <span class="op">%in%</span><span class="st"> </span>rownames_snps) </span></code></pre></div>
<pre><code>
FALSE  TRUE 
    1  3283 </code></pre>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1"></a><span class="co"># FALSE  TRUE </span></span>
<span id="cb63-2"><a href="#cb63-2"></a><span class="co">#     1  6269 </span></span>
<span id="cb63-3"><a href="#cb63-3"></a></span>
<span id="cb63-4"><a href="#cb63-4"></a><span class="co"># Merge updated pheno-to-SNP matches to raw pheno DF</span></span>
<span id="cb63-5"><a href="#cb63-5"></a>dbdata <span class="op">%&lt;&gt;%</span><span class="st"> </span></span>
<span id="cb63-6"><a href="#cb63-6"></a><span class="st">  </span><span class="kw">select</span>(<span class="op">-</span>GID,<span class="op">-</span>FullSampleName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb63-7"><a href="#cb63-7"></a><span class="st">  </span><span class="kw">left_join</span>(germ2snps) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb63-8"><a href="#cb63-8"></a><span class="st">  </span><span class="co"># Re-create the GID identifier</span></span>
<span id="cb63-9"><a href="#cb63-9"></a><span class="st">  </span><span class="co">## Equals the value SNP data name (FullSampleName) </span></span>
<span id="cb63-10"><a href="#cb63-10"></a><span class="st">  </span><span class="co">## else germplasmName if no SNP data</span></span>
<span id="cb63-11"><a href="#cb63-11"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">GID=</span><span class="kw">ifelse</span>(<span class="kw">is.na</span>(FullSampleName),germplasmName,FullSampleName))</span></code></pre></div>
</div>
<div id="output-cleaned-file" class="section level1">
<h1>Output “cleaned” file</h1>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1"></a><span class="kw">saveRDS</span>(dbdata,<span class="dt">file=</span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;NRCRI_CleanedTrialData_2020April21.rds&quot;</span>))</span></code></pre></div>
</div>
<div id="next-step" class="section level1">
<h1>Next step</h1>
<ol start="2" style="list-style-type: decimal">
<li><a href="02-curateByTrial.html">Curate by trait-trial</a>: Model each trait-trial separately, remove outliers, get BLUPs</li>
</ol>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1"></a><span class="kw">sessionInfo</span>()</span></code></pre></div>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] magrittr_1.5    forcats_0.5.0   stringr_1.4.0   dplyr_0.8.5    
 [5] purrr_0.3.4     readr_1.3.1     tidyr_1.0.2     tibble_3.0.1   
 [9] ggplot2_3.3.0   tidyverse_1.3.0 workflowr_1.6.1

loaded via a namespace (and not attached):
 [1] tidyselect_1.0.0 xfun_0.13        haven_2.2.0      lattice_0.20-41 
 [5] colorspace_1.4-1 vctrs_0.2.4      generics_0.0.2   htmltools_0.4.0 
 [9] yaml_2.2.1       utf8_1.1.4       rlang_0.4.5      later_1.0.0     
[13] pillar_1.4.3     withr_2.2.0      glue_1.4.0       DBI_1.1.0       
[17] dbplyr_1.4.3     modelr_0.1.6     readxl_1.3.1     lifecycle_0.2.0 
[21] munsell_0.5.0    gtable_0.3.0     cellranger_1.1.0 rvest_0.3.5     
[25] evaluate_0.14    knitr_1.28       httpuv_1.5.2     fansi_0.4.1     
[29] broom_0.5.6      Rcpp_1.0.4.6     promises_1.1.0   backports_1.1.6 
[33] scales_1.1.0     jsonlite_1.6.1   fs_1.4.1         hms_0.5.3       
[37] digest_0.6.25    stringi_1.4.6    grid_3.6.1       rprojroot_1.3-2 
[41] here_0.1         cli_2.0.2        tools_3.6.1      crayon_1.3.4    
[45] whisker_0.4      pkgconfig_2.0.3  ellipsis_0.3.0   xml2_1.3.2      
[49] reprex_0.3.0     lubridate_1.7.8  rstudioapi_0.11  assertthat_0.2.1
[53] rmarkdown_2.1    httr_1.4.1       R6_2.4.1         nlme_3.1-147    
[57] git2r_0.26.1     compiler_3.6.1  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
