<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="wolfemd" />


<title>Check prediction accuracy</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/darkly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>


<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">NRCRI_2020GS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Synopsis</a>
</li>
<li>
  <a href="01-cleanTPdata.html">Data Cleaning</a>
</li>
<li>
  <a href="02-curateByTrial.html">Curate Data</a>
</li>
<li>
  <a href="03-GetBLUPs.html">Get BLUPs</a>
</li>
<li>
  <a href="04-CrossValidation.html">Check prediction accuracy</a>
</li>
<li>
  <a href="05-GetGEBVs.html">Get GEBV for GS C2</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/wolfemd/NRCRI_2020GS">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Check prediction accuracy</h1>
<h4 class="author">wolfemd</h4>
<h4 class="date">2019-7-27</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-04-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>NRCRI_2020GS/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.1). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200421code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200421)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200421code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200421)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong8c45991"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> 8c45991 </a>
</p>
</div>
<div id="strongRepositoryversionstrong8c45991" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version 8c45991. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  NRCRI 2020 GS.gslides
    Untracked:  data/DatabaseDownload_42120/
    Untracked:  data/GBSdataMasterList_31818.csv
    Untracked:  data/IITA_GBStoPhenoMaster_33018.csv
    Untracked:  data/NRCRI_CleanedTrialData_2020April21.rds
    Untracked:  data/NRCRI_ExptDesignsDetected_2020April21.rds
    Untracked:  data/NRCRI_GBStoPhenoMaster_40318.csv
    Untracked:  data/iita_blupsForCrossVal_72619.rds
    Untracked:  data/iita_blupsForCrossVal_outliersRemoved_73019.rds
    Untracked:  output/DosageMatrix_NRCRI_SamplesForGP_2020April27.rds
    Untracked:  output/GEBV_NRCRI_IITAaugmented_ModelADE_2020April27.rds
    Untracked:  output/GEBV_NRCRI_IITAaugmented_ModelA_2020April27.rds
    Untracked:  output/GEBV_NRCRI_NRCRIalone_ModelADE_2020April27.rds
    Untracked:  output/GEBV_NRCRI_NRCRIalone_ModelA_2020April27.rds
    Untracked:  output/GETGV_NRCRI_IITAaugmented_ModelADE_2020April27.rds
    Untracked:  output/GETGV_NRCRI_NRCRIalone_ModelADE_2020April27.rds
    Untracked:  output/Kinship_AA_NRCRI_2020April27.rds
    Untracked:  output/Kinship_AD_NRCRI_2020April27.rds
    Untracked:  output/Kinship_A_NRCRI_2020April27.rds
    Untracked:  output/Kinship_DD_NRCRI_2020April27.rds
    Untracked:  output/Kinship_D_NRCRI_2020April27.rds
    Untracked:  output/NRCRI_CuratedTrials_2020April21.rds
    Untracked:  output/NRCRI_CuratedTrials_2020April27.rds
    Untracked:  output/cvresults_AD_nrOnly.rds
    Untracked:  output/cvresults_A_iitaAugmented.rds
    Untracked:  output/cvresults_A_nrOnly.rds
    Untracked:  output/genomicPredictions_ModelADE_NRCRI_2020April27.rds
    Untracked:  output/genomicPredictions_ModelA_NRCRI_2020April27.rds
    Untracked:  output/maxNOHAV_byStudy_NRCRI_2020April27.csv
    Untracked:  output/nrcri_blupsForModelTraining_2020April21.rds
    Untracked:  output/nrcri_blupsForModelTraining_2020April27.rds
    Untracked:  workflowr_log.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/04-CrossValidation.Rmd</code>) and HTML (<code>docs/04-CrossValidation.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
8c45991
</td>
<td>
wolfemd
</td>
<td>
2020-04-28
</td>
<td>
Publish the first set of analyses and files for NRCRI 2020 GS.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="previous-step" class="section level1">
<h1>Previous step</h1>
<ol start="3" style="list-style-type: decimal">
<li><a href="03-GetBLUPs.html">Get BLUPs combining all trial data</a>: Combine data from all trait-trials to get BLUPs for downstream genomic prediction.</li>
</ol>
</div>
<div id="objective" class="section level1">
<h1>Objective</h1>
<p><strong>Current Step:</strong></p>
<ol start="4" style="list-style-type: decimal">
<li><a href="04-CrossValidation.html">Check prediction accuracy</a>: Evaluate prediction accuracy with cross-validation.</li>
</ol>
<p>5-fold cross-validation. Replicate 5-times.</p>
<p>3 genomic models:</p>
<ol style="list-style-type: decimal">
<li>Additive-only (<strong>A</strong>)</li>
<li>Additive plus dominance (<strong>AD</strong>)</li>
<li>Addtitive plus dominance plus epistasis (<strong>ADE</strong>)</li>
</ol>
</div>
<div id="prep.-genomic-data" class="section level1">
<h1>Prep. genomic data</h1>
<div id="get-snp-data-from-ftp" class="section level2">
<h2>Get SNP data from FTP</h2>
<p>The data for the next step can be found on the cassavabase FTP server <a href="ftp://ftp.cassavabase.org/marnin_datasets/NGC_BigData/">here</a>.</p>
<p>Can be loaded directly to R from FTP.</p>
<p><strong>NOTICE:</strong> You need enough RAM and a stable network connection. I do the next steps, including cross-validation on a server with plenty of RAM and a good, stable network connection, rather than on my personal computer (a laptop with 16 GB RAM).</p>
<p>The outputs (kinship matrices and filtered snp dosages) of the steps below, which are too large for GitHub, can be found on the cassavabase FTP server <a href="ftp://ftp.cassavabase.org/marnin_datasets/NRCRI_2020GS/">here</a>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1"></a><span class="co"># activate multithread OpenBLAS for fast compute of SigmaM (genotypic var-covar matrix)</span></span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="bu">export</span> <span class="va">OMP_NUM_THREADS=</span>56</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="kw">library</span>(tidyverse); <span class="kw">library</span>(magrittr); </span>
<span id="cb2-2"><a href="#cb2-2"></a>snps&lt;-<span class="kw">readRDS</span>(<span class="dt">file=</span><span class="kw">url</span>(<span class="kw">paste0</span>(<span class="st">&quot;ftp://ftp.cassavabase.org/marnin_datasets/NGC_BigData/&quot;</span>,</span>
<span id="cb2-3"><a href="#cb2-3"></a>                              <span class="st">&quot;DosageMatrix_RefPanelAndGSprogeny_ReadyForGP_73019.rds&quot;</span>)))</span>
<span id="cb2-4"><a href="#cb2-4"></a><span class="co">#rm(list=(ls() %&gt;% grep(&quot;snps&quot;,.,value = T, invert = T)))</span></span></code></pre></div>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>blups_nrcri&lt;-<span class="kw">readRDS</span>(<span class="dt">file=</span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;nrcri_blupsForModelTraining_2020April27.rds&quot;</span>))</span>
<span id="cb3-2"><a href="#cb3-2"></a>blups_nrcri&lt;-blups_nrcri <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="st">  </span><span class="kw">select</span>(Trait,modelOutput) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb3-4"><a href="#cb3-4"></a><span class="st">  </span><span class="kw">unnest</span>(modelOutput) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb3-5"><a href="#cb3-5"></a><span class="st">  </span><span class="kw">select</span>(Trait,BLUPs) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb3-6"><a href="#cb3-6"></a><span class="st">  </span><span class="kw">unnest</span>(BLUPs) <span class="op">%&gt;%</span><span class="st">  </span></span>
<span id="cb3-7"><a href="#cb3-7"></a><span class="st">  </span><span class="kw">filter</span>(GID <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(snps))</span>
<span id="cb3-8"><a href="#cb3-8"></a><span class="kw">table</span>(<span class="kw">unique</span>(blups_nrcri<span class="op">$</span>GID) <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(snps)) <span class="co"># 2879!</span></span>
<span id="cb3-9"><a href="#cb3-9"></a></span>
<span id="cb3-10"><a href="#cb3-10"></a>blups_iita&lt;-<span class="kw">readRDS</span>(<span class="dt">file=</span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;iita_blupsForCrossVal_outliersRemoved_73019.rds&quot;</span>))</span>
<span id="cb3-11"><a href="#cb3-11"></a>blups_iita&lt;-blups_iita <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb3-12"><a href="#cb3-12"></a><span class="st">  </span><span class="kw">select</span>(Trait,blups) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb3-13"><a href="#cb3-13"></a><span class="st">  </span><span class="kw">unnest</span>(blups) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb3-14"><a href="#cb3-14"></a><span class="st">  </span><span class="kw">select</span>(<span class="op">-</span><span class="st">`</span><span class="dt">std error</span><span class="st">`</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb3-15"><a href="#cb3-15"></a><span class="st">  </span><span class="kw">filter</span>(GID <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(snps),</span>
<span id="cb3-16"><a href="#cb3-16"></a>         <span class="op">!</span><span class="kw">grepl</span>(<span class="st">&quot;TMS13F|TMS14F|TMS15F|2013_&quot;</span>,GID)) <span class="co"># don&#39;t want IITA GS progenies</span></span>
<span id="cb3-17"><a href="#cb3-17"></a><span class="kw">table</span>(<span class="kw">unique</span>(blups_iita<span class="op">$</span>GID) <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(snps)) <span class="co"># 1228</span></span>
<span id="cb3-18"><a href="#cb3-18"></a><span class="kw">union</span>(blups_nrcri<span class="op">$</span>GID,blups_iita<span class="op">$</span>GID) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">grep</span>(<span class="st">&quot;c2&quot;</span>,.,<span class="dt">value =</span> T,<span class="dt">ignore.case =</span> T)</span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>samples2Keep&lt;-<span class="kw">union</span>(blups_nrcri<span class="op">$</span>GID,blups_iita<span class="op">$</span>GID) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb4-2"><a href="#cb4-2"></a><span class="st">  </span><span class="kw">union</span>(.,<span class="kw">grep</span>(<span class="st">&quot;c2&quot;</span>,<span class="kw">rownames</span>(snps),<span class="dt">value =</span> T, <span class="dt">ignore.case =</span> T))</span>
<span id="cb4-3"><a href="#cb4-3"></a><span class="kw">table</span>(<span class="kw">rownames</span>(snps) <span class="op">%in%</span><span class="st"> </span><span class="kw">union</span>(blups_nrcri<span class="op">$</span>GID,blups_iita<span class="op">$</span>GID)) <span class="co"># 3740</span></span>
<span id="cb4-4"><a href="#cb4-4"></a><span class="kw">length</span>(samples2Keep) <span class="co"># 7062</span></span>
<span id="cb4-5"><a href="#cb4-5"></a>snps&lt;-snps[samples2Keep,]</span></code></pre></div>
</div>
<div id="maf1-filter" class="section level2">
<h2>MAF&gt;1% filter</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>maf_filter&lt;-<span class="cf">function</span>(snps,thresh){</span>
<span id="cb5-2"><a href="#cb5-2"></a>    freq&lt;-<span class="kw">colMeans</span>(snps, <span class="dt">na.rm=</span>T)<span class="op">/</span><span class="dv">2</span>; maf&lt;-freq;</span>
<span id="cb5-3"><a href="#cb5-3"></a>    maf[<span class="kw">which</span>(maf <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>)]&lt;-<span class="dv">1</span><span class="op">-</span>maf[<span class="kw">which</span>(maf <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.5</span>)]</span>
<span id="cb5-4"><a href="#cb5-4"></a>    snps1&lt;-snps[,<span class="kw">which</span>(maf<span class="op">&gt;</span>thresh)];</span>
<span id="cb5-5"><a href="#cb5-5"></a>    <span class="kw">return</span>(snps1) }</span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>snps <span class="op">%&lt;&gt;%</span><span class="st"> </span><span class="kw">maf_filter</span>(.,<span class="fl">0.01</span>)</span>
<span id="cb6-2"><a href="#cb6-2"></a><span class="kw">dim</span>(snps) <span class="co"># [1]  7062 68587</span></span></code></pre></div>
</div>
<div id="make-add-dom-and-epi-kinships" class="section level2">
<h2>Make Add, Dom and Epi kinships</h2>
<p>Going to use my own kinship function b/c I trust it’s dominance matrix calculation.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="co">#&#39; kinship function</span></span>
<span id="cb7-2"><a href="#cb7-2"></a><span class="co">#&#39;</span></span>
<span id="cb7-3"><a href="#cb7-3"></a><span class="co">#&#39; Function to create additive and dominance genomic relationship matrices from biallelic dosages.</span></span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="co">#&#39;</span></span>
<span id="cb7-5"><a href="#cb7-5"></a><span class="co">#&#39; @param M dosage matrix. Assumes SNPs in M coded 0, 1, 2 (requires rounding dosages to integers). M is Nind x Mrow, numeric matrix, with row/columanes to indicate SNP/ind ID.</span></span>
<span id="cb7-6"><a href="#cb7-6"></a><span class="co">#&#39; @param type string, &quot;add&quot; or &quot;dom&quot;. type=&quot;add&quot; gives same as rrBLUP::A.mat(), i.e. Van Raden, Method 1. type=&quot;dom&quot; gives classical parameterization according to Vitezica et al. 2013.</span></span>
<span id="cb7-7"><a href="#cb7-7"></a><span class="co">#&#39;</span></span>
<span id="cb7-8"><a href="#cb7-8"></a><span class="co">#&#39; @return square symmetic genomic relationship matrix</span></span>
<span id="cb7-9"><a href="#cb7-9"></a><span class="co">#&#39; @export</span></span>
<span id="cb7-10"><a href="#cb7-10"></a><span class="co">#&#39;</span></span>
<span id="cb7-11"><a href="#cb7-11"></a><span class="co">#&#39; @examples</span></span>
<span id="cb7-12"><a href="#cb7-12"></a><span class="co">#&#39; K&lt;-kinship(M,&quot;add&quot;)</span></span>
<span id="cb7-13"><a href="#cb7-13"></a>kinship&lt;-<span class="cf">function</span>(M,type){</span>
<span id="cb7-14"><a href="#cb7-14"></a>      M&lt;-<span class="kw">round</span>(M)</span>
<span id="cb7-15"><a href="#cb7-15"></a>      freq &lt;-<span class="st"> </span><span class="kw">colMeans</span>(M,<span class="dt">na.rm=</span>T)<span class="op">/</span><span class="dv">2</span></span>
<span id="cb7-16"><a href="#cb7-16"></a>      P &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="kw">rep</span>(freq,<span class="kw">nrow</span>(M)),<span class="dt">byrow=</span>T,<span class="dt">ncol=</span><span class="kw">ncol</span>(M))</span>
<span id="cb7-17"><a href="#cb7-17"></a>      <span class="cf">if</span>(type<span class="op">==</span><span class="st">&quot;add&quot;</span>){</span>
<span id="cb7-18"><a href="#cb7-18"></a>            Z &lt;-<span class="st"> </span>M<span class="dv">-2</span><span class="op">*</span>P</span>
<span id="cb7-19"><a href="#cb7-19"></a>            varD&lt;-<span class="kw">sum</span>(<span class="dv">2</span><span class="op">*</span>freq<span class="op">*</span>(<span class="dv">1</span><span class="op">-</span>freq))</span>
<span id="cb7-20"><a href="#cb7-20"></a>            K &lt;-<span class="st"> </span><span class="kw">tcrossprod</span>(Z)<span class="op">/</span><span class="st"> </span>varD</span>
<span id="cb7-21"><a href="#cb7-21"></a>            <span class="kw">return</span>(K)</span>
<span id="cb7-22"><a href="#cb7-22"></a>      }</span>
<span id="cb7-23"><a href="#cb7-23"></a>      <span class="cf">if</span>(type<span class="op">==</span><span class="st">&quot;dom&quot;</span>){</span>
<span id="cb7-24"><a href="#cb7-24"></a>            W&lt;-M;</span>
<span id="cb7-25"><a href="#cb7-25"></a>            W[<span class="kw">which</span>(W<span class="op">==</span><span class="dv">1</span>)]&lt;-<span class="dv">2</span><span class="op">*</span>P[<span class="kw">which</span>(W<span class="op">==</span><span class="dv">1</span>)];</span>
<span id="cb7-26"><a href="#cb7-26"></a>            W[<span class="kw">which</span>(W<span class="op">==</span><span class="dv">2</span>)]&lt;-(<span class="dv">4</span><span class="op">*</span>P[<span class="kw">which</span>(W<span class="op">==</span><span class="dv">2</span>)]<span class="op">-</span><span class="dv">2</span>);</span>
<span id="cb7-27"><a href="#cb7-27"></a>            W &lt;-<span class="st"> </span>W<span class="dv">-2</span><span class="op">*</span>(P<span class="op">^</span><span class="dv">2</span>)</span>
<span id="cb7-28"><a href="#cb7-28"></a>            varD&lt;-<span class="kw">sum</span>((<span class="dv">2</span><span class="op">*</span>freq<span class="op">*</span>(<span class="dv">1</span><span class="op">-</span>freq))<span class="op">^</span><span class="dv">2</span>)</span>
<span id="cb7-29"><a href="#cb7-29"></a>            D &lt;-<span class="st"> </span><span class="kw">tcrossprod</span>(W) <span class="op">/</span><span class="st"> </span>varD</span>
<span id="cb7-30"><a href="#cb7-30"></a>            <span class="kw">return</span>(D)</span>
<span id="cb7-31"><a href="#cb7-31"></a>      }</span>
<span id="cb7-32"><a href="#cb7-32"></a>}</span></code></pre></div>
<p>Make the kinships.</p>
<p>Below e.g. <code>A*A</code> makes a matrix that approximates additive-by-additive epistasis relationships.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>A&lt;-<span class="kw">kinship</span>(snps,<span class="dt">type=</span><span class="st">&quot;add&quot;</span>)</span>
<span id="cb8-2"><a href="#cb8-2"></a>D&lt;-<span class="kw">kinship</span>(snps,<span class="dt">type=</span><span class="st">&quot;dom&quot;</span>)</span>
<span id="cb8-3"><a href="#cb8-3"></a>AA&lt;-A<span class="op">*</span>A</span>
<span id="cb8-4"><a href="#cb8-4"></a>AD&lt;-A<span class="op">*</span>D</span>
<span id="cb8-5"><a href="#cb8-5"></a>DD&lt;-D<span class="op">*</span>D</span>
<span id="cb8-6"><a href="#cb8-6"></a></span>
<span id="cb8-7"><a href="#cb8-7"></a><span class="kw">saveRDS</span>(snps,<span class="dt">file=</span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;DosageMatrix_NRCRI_SamplesForGP_2020April27.rds&quot;</span>))</span>
<span id="cb8-8"><a href="#cb8-8"></a><span class="kw">saveRDS</span>(A,<span class="dt">file=</span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;Kinship_A_NRCRI_2020April27.rds&quot;</span>))</span>
<span id="cb8-9"><a href="#cb8-9"></a><span class="kw">saveRDS</span>(D,<span class="dt">file=</span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;Kinship_D_NRCRI_2020April27.rds&quot;</span>))</span>
<span id="cb8-10"><a href="#cb8-10"></a><span class="kw">saveRDS</span>(AA,<span class="dt">file=</span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;Kinship_AA_NRCRI_2020April27.rds&quot;</span>))</span>
<span id="cb8-11"><a href="#cb8-11"></a><span class="kw">saveRDS</span>(AD,<span class="dt">file=</span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;Kinship_AD_NRCRI_2020April27.rds&quot;</span>))</span>
<span id="cb8-12"><a href="#cb8-12"></a><span class="kw">saveRDS</span>(DD,<span class="dt">file=</span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;Kinship_DD_NRCRI_2020April27.rds&quot;</span>))</span>
<span id="cb8-13"><a href="#cb8-13"></a><span class="co">#rm(snps); gc()</span></span></code></pre></div>
<p><strong>NOTICE:</strong> The outputs (kinship matrices and filtered snp dosages) of the steps below, which are too large for GitHub, can be found on the cassavabase FTP server <a href="ftp://ftp.cassavabase.org/marnin_datasets/NRCRI_2020GS/">here</a>.</p>
</div>
</div>
<div id="cross-validation" class="section level1">
<h1>Cross-validation</h1>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1"></a><span class="co"># activate multithread OpenBLAS </span></span>
<span id="cb9-2"><a href="#cb9-2"></a><span class="bu">export</span> <span class="va">OMP_NUM_THREADS=</span>48</span>
<span id="cb9-3"><a href="#cb9-3"></a><span class="co">#export OMP_NUM_THREADS=88</span></span>
<span id="cb9-4"><a href="#cb9-4"></a><span class="co">#export OMP_NUM_THREADS=88</span></span></code></pre></div>
<div id="set-up-training-testing-data" class="section level2">
<h2>Set-up training-testing data</h2>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="kw">rm</span>(<span class="dt">list=</span><span class="kw">ls</span>())</span>
<span id="cb10-2"><a href="#cb10-2"></a><span class="kw">library</span>(tidyverse); <span class="kw">library</span>(magrittr); </span>
<span id="cb10-3"><a href="#cb10-3"></a>A&lt;-<span class="kw">readRDS</span>(<span class="dt">file=</span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;Kinship_A_NRCRI_2020April27.rds&quot;</span>))</span>
<span id="cb10-4"><a href="#cb10-4"></a></span>
<span id="cb10-5"><a href="#cb10-5"></a>blups_iita&lt;-<span class="kw">readRDS</span>(<span class="dt">file=</span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;data&quot;</span>,<span class="st">&quot;iita_blupsForCrossVal_outliersRemoved_73019.rds&quot;</span>))</span>
<span id="cb10-6"><a href="#cb10-6"></a>blups_iita&lt;-blups_iita <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-7"><a href="#cb10-7"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(Trait,blups) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-8"><a href="#cb10-8"></a><span class="st">  </span><span class="kw">unnest</span>(blups) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-9"><a href="#cb10-9"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span><span class="st">`</span><span class="dt">std error</span><span class="st">`</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-10"><a href="#cb10-10"></a><span class="st">  </span><span class="kw">filter</span>(GID <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(A),</span>
<span id="cb10-11"><a href="#cb10-11"></a>         <span class="op">!</span><span class="kw">grepl</span>(<span class="st">&quot;TMS13F|TMS14F|TMS15F|2013_&quot;</span>,GID)) <span class="co"># don&#39;t want IITA GS progenies</span></span>
<span id="cb10-12"><a href="#cb10-12"></a>blups_nrcri&lt;-<span class="kw">readRDS</span>(<span class="dt">file=</span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;nrcri_blupsForModelTraining_2020April27.rds&quot;</span>))</span>
<span id="cb10-13"><a href="#cb10-13"></a>blups_nrcri&lt;-blups_nrcri <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-14"><a href="#cb10-14"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(Trait,modelOutput) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-15"><a href="#cb10-15"></a><span class="st">  </span><span class="kw">unnest</span>(modelOutput) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-16"><a href="#cb10-16"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(Trait,BLUPs) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-17"><a href="#cb10-17"></a><span class="st">  </span><span class="kw">unnest</span>(BLUPs) <span class="op">%&gt;%</span><span class="st">  </span></span>
<span id="cb10-18"><a href="#cb10-18"></a><span class="st">  </span><span class="kw">filter</span>(GID <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(A))</span>
<span id="cb10-19"><a href="#cb10-19"></a><span class="co"># Set-up a grouping variable for:</span></span>
<span id="cb10-20"><a href="#cb10-20"></a><span class="co">## nrTP, C1a, C1b and C2a.</span></span>
<span id="cb10-21"><a href="#cb10-21"></a><span class="co">## Nest by Trait.</span></span>
<span id="cb10-22"><a href="#cb10-22"></a>c1a&lt;-blups_nrcri<span class="op">$</span>GID <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-23"><a href="#cb10-23"></a><span class="st">  </span>unique <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-24"><a href="#cb10-24"></a><span class="st">  </span><span class="kw">grep</span>(<span class="st">&quot;c1a&quot;</span>,.,<span class="dt">value =</span> T,<span class="dt">ignore.case =</span> T) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-25"><a href="#cb10-25"></a><span class="st">  </span><span class="kw">union</span>(.,blups_nrcri<span class="op">$</span>GID <span class="op">%&gt;%</span><span class="st"> </span>unique <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-26"><a href="#cb10-26"></a><span class="st">          </span><span class="kw">grep</span>(<span class="st">&quot;^F&quot;</span>,.,<span class="dt">value =</span> T,<span class="dt">ignore.case =</span> T) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-27"><a href="#cb10-27"></a><span class="st">          </span><span class="kw">grep</span>(<span class="st">&quot;c1b&quot;</span>,.,<span class="dt">value =</span> T,<span class="dt">ignore.case =</span> T,<span class="dt">invert =</span> T))</span>
<span id="cb10-28"><a href="#cb10-28"></a>c1b&lt;-blups_nrcri<span class="op">$</span>GID <span class="op">%&gt;%</span><span class="st"> </span>unique <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">grep</span>(<span class="st">&quot;c1b&quot;</span>,.,<span class="dt">value =</span> T,<span class="dt">ignore.case =</span> T)</span>
<span id="cb10-29"><a href="#cb10-29"></a>c2a&lt;-blups_nrcri<span class="op">$</span>GID <span class="op">%&gt;%</span><span class="st"> </span>unique <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-30"><a href="#cb10-30"></a><span class="st">  </span><span class="kw">grep</span>(<span class="st">&quot;C2a|C2b&quot;</span>,.,<span class="dt">value =</span> T,<span class="dt">ignore.case =</span> T) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-31"><a href="#cb10-31"></a><span class="st">  </span><span class="kw">grep</span>(<span class="st">&quot;NR17&quot;</span>,.,<span class="dt">value =</span> T,<span class="dt">ignore.case =</span> T)</span>
<span id="cb10-32"><a href="#cb10-32"></a>nrTP&lt;-<span class="kw">setdiff</span>(<span class="kw">unique</span>(blups_nrcri<span class="op">$</span>GID),<span class="kw">unique</span>(<span class="kw">c</span>(c1a,c1b,c2a)))</span>
<span id="cb10-33"><a href="#cb10-33"></a></span>
<span id="cb10-34"><a href="#cb10-34"></a>cv2do&lt;-blups_nrcri <span class="op">%&gt;%</span></span>
<span id="cb10-35"><a href="#cb10-35"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">Group=</span><span class="kw">ifelse</span>(GID <span class="op">%in%</span><span class="st"> </span>nrTP,<span class="st">&quot;nrTP&quot;</span>,</span>
<span id="cb10-36"><a href="#cb10-36"></a>                      <span class="kw">ifelse</span>(GID <span class="op">%in%</span><span class="st"> </span>c1a,<span class="st">&quot;C1a&quot;</span>,</span>
<span id="cb10-37"><a href="#cb10-37"></a>                             <span class="kw">ifelse</span>(GID <span class="op">%in%</span><span class="st"> </span>c1b, <span class="st">&quot;C1b&quot;</span>,</span>
<span id="cb10-38"><a href="#cb10-38"></a>                                    <span class="kw">ifelse</span>(GID <span class="op">%in%</span><span class="st"> </span>c2a,<span class="st">&quot;C2a&quot;</span>,<span class="ot">NA</span>))))) <span class="op">%&gt;%</span></span>
<span id="cb10-39"><a href="#cb10-39"></a><span class="st">  </span><span class="kw">nest</span>(<span class="dt">TrainTestData=</span><span class="op">-</span>Trait) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-40"><a href="#cb10-40"></a><span class="st">  </span><span class="kw">left_join</span>(blups_iita <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-41"><a href="#cb10-41"></a><span class="st">               </span><span class="kw">nest</span>(<span class="dt">augmentTP=</span><span class="op">-</span>Trait))</span>
<span id="cb10-42"><a href="#cb10-42"></a>cv2do<span class="op">$</span>TrainTestData[[<span class="dv">6</span>]] <span class="op">%&gt;%</span></span>
<span id="cb10-43"><a href="#cb10-43"></a><span class="st">  </span><span class="kw">count</span>(Group)</span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a><span class="co"># test arguments to function</span></span>
<span id="cb11-2"><a href="#cb11-2"></a><span class="co"># ----------------------</span></span>
<span id="cb11-3"><a href="#cb11-3"></a><span class="co">## Test 1 (additive only, no augmentTP)</span></span>
<span id="cb11-4"><a href="#cb11-4"></a><span class="co"># TrainTestData&lt;-cv2do_nrAlone$TrainTestData[[1]]</span></span>
<span id="cb11-5"><a href="#cb11-5"></a><span class="co"># nrepeats&lt;-1</span></span>
<span id="cb11-6"><a href="#cb11-6"></a><span class="co"># nfolds&lt;-2</span></span>
<span id="cb11-7"><a href="#cb11-7"></a><span class="co"># ncores&lt;-1</span></span>
<span id="cb11-8"><a href="#cb11-8"></a><span class="co"># gid&lt;-&quot;GID&quot;</span></span>
<span id="cb11-9"><a href="#cb11-9"></a><span class="co"># byGroup&lt;-TRUE</span></span>
<span id="cb11-10"><a href="#cb11-10"></a><span class="co"># modelType&lt;-&quot;A&quot;</span></span>
<span id="cb11-11"><a href="#cb11-11"></a><span class="co"># grms&lt;-list(A=A)</span></span>
<span id="cb11-12"><a href="#cb11-12"></a><span class="co"># augmentTP&lt;-NULL</span></span>
<span id="cb11-13"><a href="#cb11-13"></a><span class="co"># </span></span>
<span id="cb11-14"><a href="#cb11-14"></a><span class="co"># ## Test 2 (additive + dominance , no augmentTP)</span></span>
<span id="cb11-15"><a href="#cb11-15"></a><span class="co"># TrainTestData&lt;-cv2do_nrAlone$TrainTestData[[10]]</span></span>
<span id="cb11-16"><a href="#cb11-16"></a><span class="co"># nrepeats&lt;-1</span></span>
<span id="cb11-17"><a href="#cb11-17"></a><span class="co"># nfolds&lt;-2</span></span>
<span id="cb11-18"><a href="#cb11-18"></a><span class="co"># ncores&lt;-1</span></span>
<span id="cb11-19"><a href="#cb11-19"></a><span class="co"># gid&lt;-&quot;GID&quot;</span></span>
<span id="cb11-20"><a href="#cb11-20"></a><span class="co"># byGroup&lt;-TRUE</span></span>
<span id="cb11-21"><a href="#cb11-21"></a><span class="co"># modelType&lt;-&quot;AD&quot;</span></span>
<span id="cb11-22"><a href="#cb11-22"></a><span class="co"># grms&lt;-list(A=A,D=D)</span></span>
<span id="cb11-23"><a href="#cb11-23"></a><span class="co"># augmentTP&lt;-NULL</span></span>
<span id="cb11-24"><a href="#cb11-24"></a><span class="co"># splits&lt;-cvsamples$splits[[1]]</span></span>
<span id="cb11-25"><a href="#cb11-25"></a><span class="co"># GroupName&lt;-cvsamples$GroupName[[1]]</span></span>
<span id="cb11-26"><a href="#cb11-26"></a><span class="co"># ----------------------</span></span></code></pre></div>
<p>The function below implements nfold cross-validation. Specifically, for each of <strong>nrepeats</strong> it splits the data into <strong>nfolds</strong> sets according to <strong>gid</strong>. So if <code>nfolds=5</code> then the the clones will be divided into 5 groups and 5 predictions will be made. In each prediction, 4/5 of the clones will be used to predict the remaining 1/5. Accuracy of the model is measured as the correlation between the BLUPs (adj. mean for each CLONE) in the <em>test set</em> and the GEBV (the prediction made of each clone when it was in the test set).</p>
</div>
<div id="the-ultimate-runcrossval-func" class="section level2">
<h2>The ultimate runCrossVal func</h2>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a><span class="co">#&#39; @param byGroup logical, if TRUE, assumes a column named &quot;Group&quot; is present which unique classifies each GID into some genetic grouping. </span></span>
<span id="cb12-2"><a href="#cb12-2"></a><span class="co">#&#39; @param modelType string, A, AD or ADE representing model with Additive-only, Add. plus Dominance, and Add. plus Dom. plus. Epistasis (AA+AD+DD), respectively.</span></span>
<span id="cb12-3"><a href="#cb12-3"></a><span class="co">#&#39; @param grms list of GRMs where each element is named either A, D, AA, AD, DD. Matrices supplied must match required by A, AD and ADE models. For ADE grms=list(A=A,D=D,AA=AA,AD=AD,DD=DD)...</span></span>
<span id="cb12-4"><a href="#cb12-4"></a><span class="co">#&#39; @param augmentTP option to supply an additional set of training data, which will be added to each training model but never included in the test set.</span></span>
<span id="cb12-5"><a href="#cb12-5"></a><span class="co">#&#39; @param TrainTestData data.frame with de-regressed BLUPs, BLUPs and weights (WT) for training and test. If byGroup==TRUE, a column with Group as the header uniquely classifying GIDs into genetic groups, is expected.</span></span>
<span id="cb12-6"><a href="#cb12-6"></a>runCrossVal&lt;-<span class="cf">function</span>(TrainTestData,modelType,grms,nrepeats,nfolds,<span class="dt">ncores=</span><span class="dv">1</span>,</span>
<span id="cb12-7"><a href="#cb12-7"></a>                      <span class="dt">byGroup=</span><span class="ot">FALSE</span>,<span class="dt">augmentTP=</span><span class="ot">NULL</span>,<span class="dt">gid=</span><span class="st">&quot;GID&quot;</span>,...){</span>
<span id="cb12-8"><a href="#cb12-8"></a>  <span class="kw">require</span>(sommer); <span class="kw">require</span>(rsample)</span>
<span id="cb12-9"><a href="#cb12-9"></a>  <span class="co"># Set-up replicated cross-validation folds </span></span>
<span id="cb12-10"><a href="#cb12-10"></a>  <span class="co"># splitting by clone (if clone in training dataset, it can&#39;t be in testing)</span></span>
<span id="cb12-11"><a href="#cb12-11"></a>  <span class="cf">if</span>(byGroup){ </span>
<span id="cb12-12"><a href="#cb12-12"></a>    cvsamples&lt;-<span class="kw">tibble</span>(<span class="dt">GroupName=</span><span class="kw">unique</span>(TrainTestData<span class="op">$</span>Group))</span>
<span id="cb12-13"><a href="#cb12-13"></a>  } <span class="cf">else</span> { cvsamples&lt;-<span class="kw">tibble</span>(<span class="dt">GroupName=</span><span class="st">&quot;None&quot;</span>) }</span>
<span id="cb12-14"><a href="#cb12-14"></a>  cvsamples&lt;-cvsamples <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb12-15"><a href="#cb12-15"></a><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">Splits=</span><span class="kw">map</span>(GroupName,<span class="cf">function</span>(GroupName){</span>
<span id="cb12-16"><a href="#cb12-16"></a>      <span class="cf">if</span>(GroupName<span class="op">!=</span><span class="st">&quot;None&quot;</span>){ </span>
<span id="cb12-17"><a href="#cb12-17"></a>        thisgroup&lt;-TrainTestData <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb12-18"><a href="#cb12-18"></a><span class="st">          </span><span class="kw">filter</span>(Group<span class="op">==</span>GroupName) } <span class="cf">else</span> { thisgroup&lt;-TrainTestData }</span>
<span id="cb12-19"><a href="#cb12-19"></a>      out&lt;-<span class="kw">tibble</span>(<span class="dt">repeats=</span><span class="dv">1</span><span class="op">:</span>nrepeats,</span>
<span id="cb12-20"><a href="#cb12-20"></a>                  <span class="dt">splits=</span><span class="kw">rerun</span>(nrepeats,<span class="kw">group_vfold_cv</span>(thisgroup, <span class="dt">group =</span> gid, <span class="dt">v =</span> nfolds))) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb12-21"><a href="#cb12-21"></a><span class="st">        </span><span class="kw">unnest</span>(splits)</span>
<span id="cb12-22"><a href="#cb12-22"></a>      <span class="kw">return</span>(out)</span>
<span id="cb12-23"><a href="#cb12-23"></a>    })) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb12-24"><a href="#cb12-24"></a><span class="st">    </span><span class="kw">unnest</span>(Splits)</span>
<span id="cb12-25"><a href="#cb12-25"></a>  </span>
<span id="cb12-26"><a href="#cb12-26"></a>  <span class="co">## Internal function</span></span>
<span id="cb12-27"><a href="#cb12-27"></a>  <span class="co">## fits prediction model and calcs. accuracy for each train-test split</span></span>
<span id="cb12-28"><a href="#cb12-28"></a>  </span>
<span id="cb12-29"><a href="#cb12-29"></a>  fitModel&lt;-<span class="kw">possibly</span>(<span class="cf">function</span>(splits,modelType,augmentTP,TrainTestData,GroupName,grms){</span>
<span id="cb12-30"><a href="#cb12-30"></a>    starttime&lt;-<span class="kw">proc.time</span>()[<span class="dv">3</span>]</span>
<span id="cb12-31"><a href="#cb12-31"></a>    <span class="co"># Set-up training set</span></span>
<span id="cb12-32"><a href="#cb12-32"></a>    trainingdata&lt;-<span class="kw">training</span>(splits)</span>
<span id="cb12-33"><a href="#cb12-33"></a>    <span class="co">## Make sure, if there is an augmentTP, no GIDs in test-sets</span></span>
<span id="cb12-34"><a href="#cb12-34"></a>    <span class="cf">if</span>(<span class="op">!</span><span class="kw">is.null</span>(augmentTP)){</span>
<span id="cb12-35"><a href="#cb12-35"></a>      <span class="co">## remove any test-set members from augment TP before adding to training data</span></span>
<span id="cb12-36"><a href="#cb12-36"></a>      training_augment&lt;-augmentTP <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(<span class="op">!</span>(<span class="op">!!</span><span class="kw">sym</span>(gid) <span class="op">%in%</span><span class="st"> </span><span class="kw">testing</span>(splits)[[gid]]))</span>
<span id="cb12-37"><a href="#cb12-37"></a>      trainingdata&lt;-<span class="kw">bind_rows</span>(trainingdata,training_augment) }</span>
<span id="cb12-38"><a href="#cb12-38"></a>    <span class="cf">if</span>(GroupName<span class="op">!=</span><span class="st">&quot;None&quot;</span>){ trainingdata&lt;-<span class="kw">bind_rows</span>(trainingdata,</span>
<span id="cb12-39"><a href="#cb12-39"></a>                                                   TrainTestData <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb12-40"><a href="#cb12-40"></a><span class="st">                                                     </span><span class="kw">filter</span>(Group<span class="op">!=</span>GroupName,</span>
<span id="cb12-41"><a href="#cb12-41"></a>                                                            <span class="op">!</span>(<span class="op">!!</span><span class="kw">sym</span>(gid) <span class="op">%in%</span><span class="st"> </span><span class="kw">testing</span>(splits)[[gid]]))) }</span>
<span id="cb12-42"><a href="#cb12-42"></a>    <span class="co"># Subset kinship matrices</span></span>
<span id="cb12-43"><a href="#cb12-43"></a>    traintestgids&lt;-<span class="kw">union</span>(trainingdata[[gid]],<span class="kw">testing</span>(splits)[[gid]])</span>
<span id="cb12-44"><a href="#cb12-44"></a>    A1&lt;-grms[[<span class="st">&quot;A&quot;</span>]][traintestgids,traintestgids]</span>
<span id="cb12-45"><a href="#cb12-45"></a>    trainingdata[[<span class="kw">paste0</span>(gid,<span class="st">&quot;a&quot;</span>)]]&lt;-<span class="kw">factor</span>(trainingdata[[gid]],<span class="dt">levels=</span><span class="kw">rownames</span>(A1))</span>
<span id="cb12-46"><a href="#cb12-46"></a>    <span class="cf">if</span>(modelType <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;AD&quot;</span>,<span class="st">&quot;ADE&quot;</span>)){</span>
<span id="cb12-47"><a href="#cb12-47"></a>      D1&lt;-grms[[<span class="st">&quot;D&quot;</span>]][traintestgids,traintestgids]</span>
<span id="cb12-48"><a href="#cb12-48"></a>      trainingdata[[<span class="kw">paste0</span>(gid,<span class="st">&quot;d&quot;</span>)]]&lt;-<span class="kw">factor</span>(trainingdata[[gid]],<span class="dt">levels=</span><span class="kw">rownames</span>(D1))</span>
<span id="cb12-49"><a href="#cb12-49"></a>      <span class="cf">if</span>(modelType<span class="op">==</span><span class="st">&quot;ADE&quot;</span>){</span>
<span id="cb12-50"><a href="#cb12-50"></a>        AA1&lt;-grms[[<span class="st">&quot;AA&quot;</span>]][traintestgids,traintestgids]</span>
<span id="cb12-51"><a href="#cb12-51"></a>        AD1&lt;-grms[[<span class="st">&quot;AD&quot;</span>]][traintestgids,traintestgids]</span>
<span id="cb12-52"><a href="#cb12-52"></a>        DD1&lt;-grms[[<span class="st">&quot;DD&quot;</span>]][traintestgids,traintestgids]</span>
<span id="cb12-53"><a href="#cb12-53"></a>        trainingdata[[<span class="kw">paste0</span>(gid,<span class="st">&quot;aa&quot;</span>)]]&lt;-<span class="kw">factor</span>(trainingdata[[gid]],<span class="dt">levels=</span><span class="kw">rownames</span>(AA1))</span>
<span id="cb12-54"><a href="#cb12-54"></a>        trainingdata[[<span class="kw">paste0</span>(gid,<span class="st">&quot;ad&quot;</span>)]]&lt;-<span class="kw">factor</span>(trainingdata[[gid]],<span class="dt">levels=</span><span class="kw">rownames</span>(AD1))</span>
<span id="cb12-55"><a href="#cb12-55"></a>        trainingdata[[<span class="kw">paste0</span>(gid,<span class="st">&quot;dd&quot;</span>)]]&lt;-<span class="kw">factor</span>(trainingdata[[gid]],<span class="dt">levels=</span><span class="kw">rownames</span>(DD1))</span>
<span id="cb12-56"><a href="#cb12-56"></a>      }</span>
<span id="cb12-57"><a href="#cb12-57"></a>    }</span>
<span id="cb12-58"><a href="#cb12-58"></a>    <span class="co"># Set-up random model statements</span></span>
<span id="cb12-59"><a href="#cb12-59"></a>    randFormula&lt;-<span class="kw">paste0</span>(<span class="st">&quot;~vs(&quot;</span>,gid,<span class="st">&quot;a,Gu=A1)&quot;</span>)</span>
<span id="cb12-60"><a href="#cb12-60"></a>    <span class="cf">if</span>(modelType <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;AD&quot;</span>,<span class="st">&quot;ADE&quot;</span>)){</span>
<span id="cb12-61"><a href="#cb12-61"></a>      randFormula&lt;-<span class="kw">paste0</span>(randFormula,<span class="st">&quot;+vs(&quot;</span>,gid,<span class="st">&quot;d,Gu=D1)&quot;</span>)</span>
<span id="cb12-62"><a href="#cb12-62"></a>      <span class="cf">if</span>(modelType<span class="op">==</span><span class="st">&quot;ADE&quot;</span>){</span>
<span id="cb12-63"><a href="#cb12-63"></a>        randFormula&lt;-<span class="kw">paste0</span>(randFormula,</span>
<span id="cb12-64"><a href="#cb12-64"></a>                            <span class="st">&quot;+vs(&quot;</span>,gid,<span class="st">&quot;aa,Gu=AA1)&quot;</span>,</span>
<span id="cb12-65"><a href="#cb12-65"></a>                            <span class="st">&quot;+vs(&quot;</span>,gid,<span class="st">&quot;ad,Gu=AD1)&quot;</span>,</span>
<span id="cb12-66"><a href="#cb12-66"></a>                            <span class="st">&quot;+vs(&quot;</span>,gid,<span class="st">&quot;dd,Gu=DD1)&quot;</span>)</span>
<span id="cb12-67"><a href="#cb12-67"></a>      }</span>
<span id="cb12-68"><a href="#cb12-68"></a>    }</span>
<span id="cb12-69"><a href="#cb12-69"></a>    <span class="co"># Fit genomic prediction model  </span></span>
<span id="cb12-70"><a href="#cb12-70"></a>    fit &lt;-<span class="st"> </span><span class="kw">mmer</span>(<span class="dt">fixed =</span> drgBLUP <span class="op">~</span><span class="dv">1</span>,</span>
<span id="cb12-71"><a href="#cb12-71"></a>                <span class="dt">random =</span> <span class="kw">as.formula</span>(randFormula),</span>
<span id="cb12-72"><a href="#cb12-72"></a>                <span class="dt">weights =</span> WT,</span>
<span id="cb12-73"><a href="#cb12-73"></a>                <span class="dt">data=</span>trainingdata) </span>
<span id="cb12-74"><a href="#cb12-74"></a>    <span class="co"># Gather the BLUPs</span></span>
<span id="cb12-75"><a href="#cb12-75"></a>    gblups&lt;-<span class="kw">tibble</span>(<span class="dt">GID=</span><span class="kw">as.character</span>(<span class="kw">names</span>(fit<span class="op">$</span>U[[<span class="kw">paste0</span>(<span class="st">&quot;u:&quot;</span>,gid,<span class="st">&quot;a&quot;</span>)]]<span class="op">$</span>drgBLUP)),</span>
<span id="cb12-76"><a href="#cb12-76"></a>                   <span class="dt">GEBV=</span><span class="kw">as.numeric</span>(fit<span class="op">$</span>U[[<span class="kw">paste0</span>(<span class="st">&quot;u:&quot;</span>,gid,<span class="st">&quot;a&quot;</span>)]]<span class="op">$</span>drgBLUP))</span>
<span id="cb12-77"><a href="#cb12-77"></a>    <span class="cf">if</span>(modelType <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;AD&quot;</span>,<span class="st">&quot;ADE&quot;</span>)){</span>
<span id="cb12-78"><a href="#cb12-78"></a>      gblups <span class="op">%&lt;&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">GEDD=</span><span class="kw">as.numeric</span>(fit<span class="op">$</span>U[[<span class="kw">paste0</span>(<span class="st">&quot;u:&quot;</span>,gid,<span class="st">&quot;d&quot;</span>)]]<span class="op">$</span>drgBLUP))</span>
<span id="cb12-79"><a href="#cb12-79"></a>      <span class="cf">if</span>(modelType<span class="op">==</span><span class="st">&quot;ADE&quot;</span>){</span>
<span id="cb12-80"><a href="#cb12-80"></a>        gblups <span class="op">%&lt;&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">GEEDaa=</span><span class="kw">as.numeric</span>(fit<span class="op">$</span>U[[<span class="kw">paste0</span>(<span class="st">&quot;u:&quot;</span>,gid,<span class="st">&quot;aa&quot;</span>)]]<span class="op">$</span>drgBLUP),</span>
<span id="cb12-81"><a href="#cb12-81"></a>                           <span class="dt">GEEDad=</span><span class="kw">as.numeric</span>(fit<span class="op">$</span>U[[<span class="kw">paste0</span>(<span class="st">&quot;u:&quot;</span>,gid,<span class="st">&quot;ad&quot;</span>)]]<span class="op">$</span>drgBLUP),</span>
<span id="cb12-82"><a href="#cb12-82"></a>                           <span class="dt">GEEDdd=</span><span class="kw">as.numeric</span>(fit<span class="op">$</span>U[[<span class="kw">paste0</span>(<span class="st">&quot;u:&quot;</span>,gid,<span class="st">&quot;dd&quot;</span>)]]<span class="op">$</span>drgBLUP)) </span>
<span id="cb12-83"><a href="#cb12-83"></a>      }</span>
<span id="cb12-84"><a href="#cb12-84"></a>    }</span>
<span id="cb12-85"><a href="#cb12-85"></a>    <span class="co"># Calc GETGVs </span></span>
<span id="cb12-86"><a href="#cb12-86"></a>    <span class="co">## Note that for modelType==&quot;A&quot;, GEBV==GETGV</span></span>
<span id="cb12-87"><a href="#cb12-87"></a>    gblups <span class="op">%&lt;&gt;%</span><span class="st"> </span></span>
<span id="cb12-88"><a href="#cb12-88"></a><span class="st">      </span><span class="kw">mutate</span>(<span class="dt">GETGV=</span><span class="kw">rowSums</span>(.[,<span class="kw">grepl</span>(<span class="st">&quot;GE&quot;</span>,<span class="kw">colnames</span>(.))])) </span>
<span id="cb12-89"><a href="#cb12-89"></a>    <span class="co"># Test set validation data</span></span>
<span id="cb12-90"><a href="#cb12-90"></a>    validationData&lt;-TrainTestData <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb12-91"><a href="#cb12-91"></a><span class="st">      </span>dplyr<span class="op">::</span><span class="kw">select</span>(gid,BLUP) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb12-92"><a href="#cb12-92"></a><span class="st">      </span><span class="kw">filter</span>(GID <span class="op">%in%</span><span class="st"> </span><span class="kw">testing</span>(splits)[[gid]])</span>
<span id="cb12-93"><a href="#cb12-93"></a>    <span class="co"># Measure accuracy in test set</span></span>
<span id="cb12-94"><a href="#cb12-94"></a>    <span class="co">## cor(GEBV,BLUP)</span></span>
<span id="cb12-95"><a href="#cb12-95"></a>    <span class="co">## cor(GETGV,BLUP)</span></span>
<span id="cb12-96"><a href="#cb12-96"></a>    accuracy&lt;-gblups <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb12-97"><a href="#cb12-97"></a><span class="st">      </span><span class="kw">mutate</span>(<span class="dt">GETGV=</span><span class="kw">rowSums</span>(.[,<span class="kw">grepl</span>(<span class="st">&quot;GE&quot;</span>,<span class="kw">colnames</span>(.))])) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb12-98"><a href="#cb12-98"></a><span class="st">      </span><span class="kw">filter</span>(GID <span class="op">%in%</span><span class="st"> </span><span class="kw">testing</span>(splits)[[gid]]) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb12-99"><a href="#cb12-99"></a><span class="st">      </span><span class="kw">left_join</span>(validationData) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb12-100"><a href="#cb12-100"></a><span class="st">      </span><span class="kw">summarize</span>(<span class="dt">accGEBV=</span><span class="kw">cor</span>(GEBV,BLUP, <span class="dt">use =</span> <span class="st">&#39;complete.obs&#39;</span>),</span>
<span id="cb12-101"><a href="#cb12-101"></a>                <span class="dt">accGETGV=</span><span class="kw">cor</span>(GETGV,BLUP, <span class="dt">use =</span> <span class="st">&#39;complete.obs&#39;</span>))</span>
<span id="cb12-102"><a href="#cb12-102"></a>    computeTime&lt;-<span class="kw">proc.time</span>()[<span class="dv">3</span>]<span class="op">-</span>starttime</span>
<span id="cb12-103"><a href="#cb12-103"></a>    accuracy <span class="op">%&lt;&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">computeTime=</span>computeTime)</span>
<span id="cb12-104"><a href="#cb12-104"></a>    <span class="kw">return</span>(accuracy)</span>
<span id="cb12-105"><a href="#cb12-105"></a>  },<span class="dt">otherwise =</span> <span class="ot">NA</span>)</span>
<span id="cb12-106"><a href="#cb12-106"></a>  <span class="co">## Run models across all train-test splits</span></span>
<span id="cb12-107"><a href="#cb12-107"></a>  <span class="co">## Parallelize </span></span>
<span id="cb12-108"><a href="#cb12-108"></a>  <span class="kw">require</span>(furrr); <span class="kw">plan</span>(multiprocess); <span class="kw">options</span>(<span class="dt">mc.cores=</span>ncores);</span>
<span id="cb12-109"><a href="#cb12-109"></a>  cvsamples&lt;-cvsamples <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb12-110"><a href="#cb12-110"></a><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">accuracy=</span><span class="kw">future_map2</span>(splits,GroupName,</span>
<span id="cb12-111"><a href="#cb12-111"></a>                                <span class="op">~</span><span class="kw">fitModel</span>(<span class="dt">splits=</span>.x,<span class="dt">GroupName=</span>.y,</span>
<span id="cb12-112"><a href="#cb12-112"></a>                                          <span class="dt">modelType=</span>modelType,<span class="dt">augmentTP=</span><span class="ot">NULL</span>,<span class="dt">TrainTestData=</span>TrainTestData,<span class="dt">grms=</span>grms),</span>
<span id="cb12-113"><a href="#cb12-113"></a>                                <span class="dt">.progress =</span> <span class="ot">FALSE</span>)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb12-114"><a href="#cb12-114"></a><span class="st">    </span><span class="kw">unnest</span>(accuracy)</span>
<span id="cb12-115"><a href="#cb12-115"></a>  <span class="kw">return</span>(cvsamples) </span>
<span id="cb12-116"><a href="#cb12-116"></a>}</span></code></pre></div>
<p>Run some tests of the function…</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="co"># options(future.globals.maxSize= 1500*1024^2)</span></span>
<span id="cb13-2"><a href="#cb13-2"></a><span class="co"># test_cv_ad_yield&lt;-runCrossVal(TrainTestData=cv2do$TrainTestData[[8]],</span></span>
<span id="cb13-3"><a href="#cb13-3"></a><span class="co">#                               modelType=&quot;AD&quot;,</span></span>
<span id="cb13-4"><a href="#cb13-4"></a><span class="co">#                               grms=list(A=A,D=D),</span></span>
<span id="cb13-5"><a href="#cb13-5"></a><span class="co">#                               byGroup=TRUE,augmentTP=NULL,</span></span>
<span id="cb13-6"><a href="#cb13-6"></a><span class="co">#                               nrepeats=1,nfolds=2,ncores=2,gid=&quot;GID&quot;)</span></span>
<span id="cb13-7"><a href="#cb13-7"></a><span class="co"># </span></span>
<span id="cb13-8"><a href="#cb13-8"></a><span class="co"># TrainTestData&lt;-cv2do %&gt;% filter(Trait==&quot;logFYLD&quot;) %$% TrainTestData[[1]]</span></span>
<span id="cb13-9"><a href="#cb13-9"></a><span class="co"># augmentTP&lt;-cv2do %&gt;% filter(Trait==&quot;logFYLD&quot;) %$% augmentTP[[1]]</span></span>
<span id="cb13-10"><a href="#cb13-10"></a><span class="co"># test_cv_a_augment&lt;-runCrossVal(TrainTestData=TrainTestData,</span></span>
<span id="cb13-11"><a href="#cb13-11"></a><span class="co">#                                modelType=&quot;A&quot;,</span></span>
<span id="cb13-12"><a href="#cb13-12"></a><span class="co">#                                grms=list(A=A),</span></span>
<span id="cb13-13"><a href="#cb13-13"></a><span class="co">#                                byGroup=TRUE,augmentTP=augmentTP,</span></span>
<span id="cb13-14"><a href="#cb13-14"></a><span class="co">#                                nrepeats=1,nfolds=2,ncores=2,gid=&quot;GID&quot;)</span></span>
<span id="cb13-15"><a href="#cb13-15"></a><span class="co"># test_cv_a_noaug&lt;-runCrossVal(TrainTestData=TrainTestData,</span></span>
<span id="cb13-16"><a href="#cb13-16"></a><span class="co">#                              modelType=&quot;A&quot;,</span></span>
<span id="cb13-17"><a href="#cb13-17"></a><span class="co">#                              grms=list(A=A),</span></span>
<span id="cb13-18"><a href="#cb13-18"></a><span class="co">#                              byGroup=TRUE,augmentTP=NULL,</span></span>
<span id="cb13-19"><a href="#cb13-19"></a><span class="co">#                              nrepeats=1,nfolds=2,ncores=2,gid=&quot;GID&quot;)</span></span></code></pre></div>
</div>
<div id="cv---modeltypea" class="section level2">
<h2>CV - modelType=“A”:</h2>
<div id="nrcri-alone" class="section level3">
<h3>NRCRI alone</h3>
<p>cbsulm13 (96 cores; 512GB RAM)</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a>cv_A_nrOnly&lt;-cv2do <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb14-2"><a href="#cb14-2"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">CVresults=</span><span class="kw">map</span>(TrainTestData,<span class="op">~</span><span class="kw">runCrossVal</span>(<span class="dt">TrainTestData=</span>.,</span>
<span id="cb14-3"><a href="#cb14-3"></a>                                                  <span class="dt">modelType=</span><span class="st">&quot;A&quot;</span>,</span>
<span id="cb14-4"><a href="#cb14-4"></a>                                                  <span class="dt">grms=</span><span class="kw">list</span>(<span class="dt">A=</span>A),</span>
<span id="cb14-5"><a href="#cb14-5"></a>                                                  <span class="dt">byGroup=</span><span class="ot">TRUE</span>,<span class="dt">augmentTP=</span><span class="ot">NULL</span>,</span>
<span id="cb14-6"><a href="#cb14-6"></a>                                                  <span class="dt">nrepeats=</span><span class="dv">5</span>,<span class="dt">nfolds=</span><span class="dv">5</span>,<span class="dt">ncores=</span><span class="dv">10</span>,<span class="dt">gid=</span><span class="st">&quot;GID&quot;</span>)))</span>
<span id="cb14-7"><a href="#cb14-7"></a>cv_A_nrOnly <span class="op">%&lt;&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">Dataset=</span><span class="st">&quot;NRalone&quot;</span>,<span class="dt">modelType=</span><span class="st">&quot;A&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>TrainTestData,<span class="op">-</span>augmentTP)</span>
<span id="cb14-8"><a href="#cb14-8"></a><span class="kw">saveRDS</span>(cv_A_nrOnly,<span class="dt">file=</span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;cvresults_A_nrOnly.rds&quot;</span>))</span></code></pre></div>
</div>
<div id="iita-augmented" class="section level3">
<h3>IITA augmented</h3>
<p>cbsulm18 (88 cores; 512GB)</p>
<p>For this one, try with <code>ncores=1</code> instead of <code>ncores=10</code>.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a>cv_A_iitaAugmented&lt;-cv2do <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb15-2"><a href="#cb15-2"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">isnullAugment=</span><span class="kw">map_lgl</span>(augmentTP,<span class="op">~</span><span class="kw">is.null</span>(.))) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb15-3"><a href="#cb15-3"></a><span class="st">  </span><span class="kw">filter</span>(<span class="op">!</span>isnullAugment) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb15-4"><a href="#cb15-4"></a><span class="st">  </span><span class="kw">select</span>(<span class="op">-</span>isnullAugment) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb15-5"><a href="#cb15-5"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">CVresults=</span><span class="kw">map2</span>(TrainTestData,augmentTP,<span class="op">~</span><span class="kw">runCrossVal</span>(<span class="dt">TrainTestData=</span>.x,</span>
<span id="cb15-6"><a href="#cb15-6"></a>                                                  <span class="dt">modelType=</span><span class="st">&quot;A&quot;</span>,</span>
<span id="cb15-7"><a href="#cb15-7"></a>                                                  <span class="dt">grms=</span><span class="kw">list</span>(<span class="dt">A=</span>A),</span>
<span id="cb15-8"><a href="#cb15-8"></a>                                                  <span class="dt">byGroup=</span><span class="ot">TRUE</span>,<span class="dt">augmentTP=</span>.y,</span>
<span id="cb15-9"><a href="#cb15-9"></a>                                                  <span class="dt">nrepeats=</span><span class="dv">5</span>,<span class="dt">nfolds=</span><span class="dv">5</span>,<span class="dt">ncores=</span><span class="dv">1</span>,<span class="dt">gid=</span><span class="st">&quot;GID&quot;</span>)))</span>
<span id="cb15-10"><a href="#cb15-10"></a>cv_A_iitaAugmented <span class="op">%&lt;&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">Dataset=</span><span class="st">&quot;IITAaugmented&quot;</span>,<span class="dt">modelType=</span><span class="st">&quot;A&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>TrainTestData,<span class="op">-</span>augmentTP)</span>
<span id="cb15-11"><a href="#cb15-11"></a><span class="kw">saveRDS</span>(cv_A_iitaAugmented,<span class="dt">file=</span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;cvresults_A_iitaAugmented.rds&quot;</span>))</span></code></pre></div>
</div>
</div>
<div id="cv---modeltypead" class="section level2">
<h2>CV - modelType=“AD”:</h2>
<div id="nrcri-alone-1" class="section level3">
<h3>NRCRI alone</h3>
<p>cbsulm15 (96 cores; 512GB RAM)</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1"></a><span class="kw">options</span>(<span class="dt">future.globals.maxSize=</span> <span class="dv">1500</span><span class="op">*</span><span class="dv">1024</span><span class="op">^</span><span class="dv">2</span>)</span>
<span id="cb16-2"><a href="#cb16-2"></a>D&lt;-<span class="kw">readRDS</span>(<span class="dt">file=</span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;Kinship_D_NRCRI_2020April27.rds&quot;</span>))</span>
<span id="cb16-3"><a href="#cb16-3"></a>cv_AD_nrOnly&lt;-cv2do <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb16-4"><a href="#cb16-4"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">CVresults=</span><span class="kw">map</span>(TrainTestData,<span class="op">~</span><span class="kw">runCrossVal</span>(<span class="dt">TrainTestData=</span>.,</span>
<span id="cb16-5"><a href="#cb16-5"></a>                                                  <span class="dt">modelType=</span><span class="st">&quot;AD&quot;</span>,</span>
<span id="cb16-6"><a href="#cb16-6"></a>                                                  <span class="dt">grms=</span><span class="kw">list</span>(<span class="dt">A=</span>A,<span class="dt">D=</span>D),</span>
<span id="cb16-7"><a href="#cb16-7"></a>                                                  <span class="dt">byGroup=</span><span class="ot">TRUE</span>,<span class="dt">augmentTP=</span><span class="ot">NULL</span>,</span>
<span id="cb16-8"><a href="#cb16-8"></a>                                                  <span class="dt">nrepeats=</span><span class="dv">5</span>,<span class="dt">nfolds=</span><span class="dv">5</span>,<span class="dt">ncores=</span><span class="dv">4</span>,<span class="dt">gid=</span><span class="st">&quot;GID&quot;</span>)))</span>
<span id="cb16-9"><a href="#cb16-9"></a>cv_AD_nrOnly <span class="op">%&lt;&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">Dataset=</span><span class="st">&quot;NRalone&quot;</span>,<span class="dt">modelType=</span><span class="st">&quot;AD&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>TrainTestData,<span class="op">-</span>augmentTP)</span>
<span id="cb16-10"><a href="#cb16-10"></a><span class="kw">saveRDS</span>(cv_AD_nrOnly,<span class="dt">file=</span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;cvresults_AD_nrOnly.rds&quot;</span>))</span></code></pre></div>
</div>
<div id="iita-augmented-1" class="section level3">
<h3>IITA augmented</h3>
<p>cbsulm13 (96 cores; 512GB RAM)</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a><span class="kw">options</span>(<span class="dt">future.globals.maxSize=</span> <span class="dv">1500</span><span class="op">*</span><span class="dv">1024</span><span class="op">^</span><span class="dv">2</span>)</span>
<span id="cb17-2"><a href="#cb17-2"></a>D&lt;-<span class="kw">readRDS</span>(<span class="dt">file=</span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;Kinship_D_NRCRI_2020April27.rds&quot;</span>))</span>
<span id="cb17-3"><a href="#cb17-3"></a>cv_AD_iitaAugmented&lt;-cv2do <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb17-4"><a href="#cb17-4"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">isnullAugment=</span><span class="kw">map_lgl</span>(augmentTP,<span class="op">~</span><span class="kw">is.null</span>(.))) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb17-5"><a href="#cb17-5"></a><span class="st">  </span><span class="kw">filter</span>(<span class="op">!</span>isnullAugment) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb17-6"><a href="#cb17-6"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>isnullAugment) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb17-7"><a href="#cb17-7"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">CVresults=</span><span class="kw">map2</span>(TrainTestData,augmentTP,<span class="op">~</span><span class="kw">runCrossVal</span>(<span class="dt">TrainTestData=</span>.x,</span>
<span id="cb17-8"><a href="#cb17-8"></a>                                                             <span class="dt">modelType=</span><span class="st">&quot;AD&quot;</span>,</span>
<span id="cb17-9"><a href="#cb17-9"></a>                                                             <span class="dt">grms=</span><span class="kw">list</span>(<span class="dt">A=</span>A,<span class="dt">D=</span>D),</span>
<span id="cb17-10"><a href="#cb17-10"></a>                                                             <span class="dt">byGroup=</span><span class="ot">TRUE</span>,<span class="dt">augmentTP=</span>.y,</span>
<span id="cb17-11"><a href="#cb17-11"></a>                                                             <span class="dt">nrepeats=</span><span class="dv">5</span>,<span class="dt">nfolds=</span><span class="dv">5</span>,<span class="dt">ncores=</span><span class="dv">10</span>,<span class="dt">gid=</span><span class="st">&quot;GID&quot;</span>)))</span>
<span id="cb17-12"><a href="#cb17-12"></a>cv_AD_iitaAugmented <span class="op">%&lt;&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">Dataset=</span><span class="st">&quot;IITAaugmented&quot;</span>,<span class="dt">modelType=</span><span class="st">&quot;AD&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>TrainTestData,<span class="op">-</span>augmentTP)</span>
<span id="cb17-13"><a href="#cb17-13"></a><span class="kw">saveRDS</span>(cv_AD_iitaAugmented,<span class="dt">file=</span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;cvresults_AD_iitaAugmented.rds&quot;</span>))</span></code></pre></div>
</div>
</div>
<div id="cv---modeltypeade" class="section level2">
<h2>CV - modelType=“ADE”:</h2>
<div id="nrcri-alone-2" class="section level3">
<h3>NRCRI alone</h3>
<p>cbsulm15 (96 cores; 512GB RAM)</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1"></a>runCrossVal_dev&lt;-<span class="cf">function</span>(TrainTestData,modelType,grms,nrepeats,nfolds,<span class="dt">ncores=</span><span class="dv">1</span>,</span>
<span id="cb18-2"><a href="#cb18-2"></a>                      <span class="dt">byGroup=</span><span class="ot">FALSE</span>,<span class="dt">augmentTP=</span><span class="ot">NULL</span>,<span class="dt">gid=</span><span class="st">&quot;GID&quot;</span>,...){</span>
<span id="cb18-3"><a href="#cb18-3"></a>  <span class="kw">require</span>(sommer); <span class="kw">require</span>(rsample)</span>
<span id="cb18-4"><a href="#cb18-4"></a>  <span class="co"># Set-up replicated cross-validation folds </span></span>
<span id="cb18-5"><a href="#cb18-5"></a>  <span class="co"># splitting by clone (if clone in training dataset, it can&#39;t be in testing)</span></span>
<span id="cb18-6"><a href="#cb18-6"></a>  <span class="cf">if</span>(byGroup){ </span>
<span id="cb18-7"><a href="#cb18-7"></a>    cvsamples&lt;-<span class="kw">tibble</span>(<span class="dt">GroupName=</span><span class="kw">unique</span>(TrainTestData<span class="op">$</span>Group))</span>
<span id="cb18-8"><a href="#cb18-8"></a>  } <span class="cf">else</span> { cvsamples&lt;-<span class="kw">tibble</span>(<span class="dt">GroupName=</span><span class="st">&quot;None&quot;</span>) }</span>
<span id="cb18-9"><a href="#cb18-9"></a>  cvsamples&lt;-cvsamples <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb18-10"><a href="#cb18-10"></a><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">Splits=</span><span class="kw">map</span>(GroupName,<span class="cf">function</span>(GroupName){</span>
<span id="cb18-11"><a href="#cb18-11"></a>      <span class="cf">if</span>(GroupName<span class="op">!=</span><span class="st">&quot;None&quot;</span>){ </span>
<span id="cb18-12"><a href="#cb18-12"></a>        thisgroup&lt;-TrainTestData <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb18-13"><a href="#cb18-13"></a><span class="st">          </span><span class="kw">filter</span>(Group<span class="op">==</span>GroupName) } <span class="cf">else</span> { thisgroup&lt;-TrainTestData }</span>
<span id="cb18-14"><a href="#cb18-14"></a>      out&lt;-<span class="kw">tibble</span>(<span class="dt">repeats=</span><span class="dv">1</span><span class="op">:</span>nrepeats,</span>
<span id="cb18-15"><a href="#cb18-15"></a>                  <span class="dt">splits=</span><span class="kw">rerun</span>(nrepeats,<span class="kw">group_vfold_cv</span>(thisgroup, <span class="dt">group =</span> gid, <span class="dt">v =</span> nfolds))) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb18-16"><a href="#cb18-16"></a><span class="st">        </span><span class="kw">unnest</span>(splits)</span>
<span id="cb18-17"><a href="#cb18-17"></a>      <span class="kw">return</span>(out)</span>
<span id="cb18-18"><a href="#cb18-18"></a>    })) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb18-19"><a href="#cb18-19"></a><span class="st">    </span><span class="kw">unnest</span>(Splits)</span>
<span id="cb18-20"><a href="#cb18-20"></a>  </span>
<span id="cb18-21"><a href="#cb18-21"></a>  <span class="co">## Internal function</span></span>
<span id="cb18-22"><a href="#cb18-22"></a>  <span class="co">## fits prediction model and calcs. accuracy for each train-test split</span></span>
<span id="cb18-23"><a href="#cb18-23"></a>  </span>
<span id="cb18-24"><a href="#cb18-24"></a>  fitModel&lt;-<span class="kw">possibly</span>(<span class="cf">function</span>(splits,modelType,augmentTP,TrainTestData,GroupName,grms){</span>
<span id="cb18-25"><a href="#cb18-25"></a>    starttime&lt;-<span class="kw">proc.time</span>()[<span class="dv">3</span>]</span>
<span id="cb18-26"><a href="#cb18-26"></a>    <span class="co"># Set-up training set</span></span>
<span id="cb18-27"><a href="#cb18-27"></a>    trainingdata&lt;-<span class="kw">training</span>(splits)</span>
<span id="cb18-28"><a href="#cb18-28"></a>    <span class="co">## Make sure, if there is an augmentTP, no GIDs in test-sets</span></span>
<span id="cb18-29"><a href="#cb18-29"></a>    <span class="cf">if</span>(<span class="op">!</span><span class="kw">is.null</span>(augmentTP)){</span>
<span id="cb18-30"><a href="#cb18-30"></a>      <span class="co">## remove any test-set members from augment TP before adding to training data</span></span>
<span id="cb18-31"><a href="#cb18-31"></a>      training_augment&lt;-augmentTP <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(<span class="op">!</span>(<span class="op">!!</span><span class="kw">sym</span>(gid) <span class="op">%in%</span><span class="st"> </span><span class="kw">testing</span>(splits)[[gid]]))</span>
<span id="cb18-32"><a href="#cb18-32"></a>      trainingdata&lt;-<span class="kw">bind_rows</span>(trainingdata,training_augment) }</span>
<span id="cb18-33"><a href="#cb18-33"></a>    <span class="cf">if</span>(GroupName<span class="op">!=</span><span class="st">&quot;None&quot;</span>){ trainingdata&lt;-<span class="kw">bind_rows</span>(trainingdata,</span>
<span id="cb18-34"><a href="#cb18-34"></a>                                                   TrainTestData <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb18-35"><a href="#cb18-35"></a><span class="st">                                                     </span><span class="kw">filter</span>(Group<span class="op">!=</span>GroupName,</span>
<span id="cb18-36"><a href="#cb18-36"></a>                                                            <span class="op">!</span>(<span class="op">!!</span><span class="kw">sym</span>(gid) <span class="op">%in%</span><span class="st"> </span><span class="kw">testing</span>(splits)[[gid]]))) }</span>
<span id="cb18-37"><a href="#cb18-37"></a>    <span class="co"># Subset kinship matrices</span></span>
<span id="cb18-38"><a href="#cb18-38"></a>    traintestgids&lt;-<span class="kw">union</span>(trainingdata[[gid]],<span class="kw">testing</span>(splits)[[gid]])</span>
<span id="cb18-39"><a href="#cb18-39"></a>    A1&lt;-grms[[<span class="st">&quot;A&quot;</span>]][traintestgids,traintestgids]</span>
<span id="cb18-40"><a href="#cb18-40"></a>    trainingdata[[<span class="kw">paste0</span>(gid,<span class="st">&quot;a&quot;</span>)]]&lt;-<span class="kw">factor</span>(trainingdata[[gid]],<span class="dt">levels=</span><span class="kw">rownames</span>(A1))</span>
<span id="cb18-41"><a href="#cb18-41"></a>    <span class="cf">if</span>(modelType <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;AD&quot;</span>,<span class="st">&quot;ADE&quot;</span>)){</span>
<span id="cb18-42"><a href="#cb18-42"></a>      D1&lt;-grms[[<span class="st">&quot;D&quot;</span>]][traintestgids,traintestgids]</span>
<span id="cb18-43"><a href="#cb18-43"></a>      trainingdata[[<span class="kw">paste0</span>(gid,<span class="st">&quot;d&quot;</span>)]]&lt;-<span class="kw">factor</span>(trainingdata[[gid]],<span class="dt">levels=</span><span class="kw">rownames</span>(D1))</span>
<span id="cb18-44"><a href="#cb18-44"></a>      <span class="cf">if</span>(modelType<span class="op">==</span><span class="st">&quot;ADE&quot;</span>){</span>
<span id="cb18-45"><a href="#cb18-45"></a>        <span class="co">#AA1&lt;-grms[[&quot;AA&quot;]][traintestgids,traintestgids]</span></span>
<span id="cb18-46"><a href="#cb18-46"></a>        AD1&lt;-grms[[<span class="st">&quot;AD&quot;</span>]][traintestgids,traintestgids]</span>
<span id="cb18-47"><a href="#cb18-47"></a>        <span class="kw">diag</span>(AD1)&lt;-<span class="kw">diag</span>(AD1)<span class="op">+</span><span class="fl">1e-06</span></span>
<span id="cb18-48"><a href="#cb18-48"></a>        <span class="co">#DD1&lt;-grms[[&quot;DD&quot;]][traintestgids,traintestgids]</span></span>
<span id="cb18-49"><a href="#cb18-49"></a>        <span class="co">#trainingdata[[paste0(gid,&quot;aa&quot;)]]&lt;-factor(trainingdata[[gid]],levels=rownames(AA1))</span></span>
<span id="cb18-50"><a href="#cb18-50"></a>        trainingdata[[<span class="kw">paste0</span>(gid,<span class="st">&quot;ad&quot;</span>)]]&lt;-<span class="kw">factor</span>(trainingdata[[gid]],<span class="dt">levels=</span><span class="kw">rownames</span>(AD1))</span>
<span id="cb18-51"><a href="#cb18-51"></a>        <span class="co">#trainingdata[[paste0(gid,&quot;dd&quot;)]]&lt;-factor(trainingdata[[gid]],levels=rownames(DD1))</span></span>
<span id="cb18-52"><a href="#cb18-52"></a>      }</span>
<span id="cb18-53"><a href="#cb18-53"></a>    }</span>
<span id="cb18-54"><a href="#cb18-54"></a>    <span class="co"># Set-up random model statements</span></span>
<span id="cb18-55"><a href="#cb18-55"></a>    randFormula&lt;-<span class="kw">paste0</span>(<span class="st">&quot;~vs(&quot;</span>,gid,<span class="st">&quot;a,Gu=A1)&quot;</span>)</span>
<span id="cb18-56"><a href="#cb18-56"></a>    <span class="cf">if</span>(modelType <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;AD&quot;</span>,<span class="st">&quot;ADE&quot;</span>)){</span>
<span id="cb18-57"><a href="#cb18-57"></a>      randFormula&lt;-<span class="kw">paste0</span>(randFormula,<span class="st">&quot;+vs(&quot;</span>,gid,<span class="st">&quot;d,Gu=D1)&quot;</span>)</span>
<span id="cb18-58"><a href="#cb18-58"></a>      <span class="cf">if</span>(modelType<span class="op">==</span><span class="st">&quot;ADE&quot;</span>){</span>
<span id="cb18-59"><a href="#cb18-59"></a>        randFormula&lt;-<span class="kw">paste0</span>(randFormula,<span class="st">&quot;+vs(&quot;</span>,gid,<span class="st">&quot;ad,Gu=AD1)&quot;</span>)</span>
<span id="cb18-60"><a href="#cb18-60"></a>                            <span class="co">#&quot;+vs(&quot;,gid,&quot;aa,Gu=AA1)&quot;,</span></span>
<span id="cb18-61"><a href="#cb18-61"></a>                            <span class="co">#&quot;+vs(&quot;,gid,&quot;ad,Gu=AD1)&quot;)</span></span>
<span id="cb18-62"><a href="#cb18-62"></a>                            <span class="co">#&quot;+vs(&quot;,gid,&quot;dd,Gu=DD1)&quot;)</span></span>
<span id="cb18-63"><a href="#cb18-63"></a>      }</span>
<span id="cb18-64"><a href="#cb18-64"></a>    }</span>
<span id="cb18-65"><a href="#cb18-65"></a>    <span class="co"># Fit genomic prediction model  </span></span>
<span id="cb18-66"><a href="#cb18-66"></a>    fit &lt;-<span class="st"> </span><span class="kw">mmer</span>(<span class="dt">fixed =</span> drgBLUP <span class="op">~</span><span class="dv">1</span>,</span>
<span id="cb18-67"><a href="#cb18-67"></a>                <span class="dt">random =</span> <span class="kw">as.formula</span>(randFormula),</span>
<span id="cb18-68"><a href="#cb18-68"></a>                <span class="dt">weights =</span> WT,</span>
<span id="cb18-69"><a href="#cb18-69"></a>                <span class="dt">data=</span>trainingdata) </span>
<span id="cb18-70"><a href="#cb18-70"></a>    <span class="co"># Gather the BLUPs</span></span>
<span id="cb18-71"><a href="#cb18-71"></a>    gblups&lt;-<span class="kw">tibble</span>(<span class="dt">GID=</span><span class="kw">as.character</span>(<span class="kw">names</span>(fit<span class="op">$</span>U[[<span class="kw">paste0</span>(<span class="st">&quot;u:&quot;</span>,gid,<span class="st">&quot;a&quot;</span>)]]<span class="op">$</span>drgBLUP)),</span>
<span id="cb18-72"><a href="#cb18-72"></a>                   <span class="dt">GEBV=</span><span class="kw">as.numeric</span>(fit<span class="op">$</span>U[[<span class="kw">paste0</span>(<span class="st">&quot;u:&quot;</span>,gid,<span class="st">&quot;a&quot;</span>)]]<span class="op">$</span>drgBLUP))</span>
<span id="cb18-73"><a href="#cb18-73"></a>    <span class="cf">if</span>(modelType <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;AD&quot;</span>,<span class="st">&quot;ADE&quot;</span>)){</span>
<span id="cb18-74"><a href="#cb18-74"></a>      gblups <span class="op">%&lt;&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">GEDD=</span><span class="kw">as.numeric</span>(fit<span class="op">$</span>U[[<span class="kw">paste0</span>(<span class="st">&quot;u:&quot;</span>,gid,<span class="st">&quot;d&quot;</span>)]]<span class="op">$</span>drgBLUP))</span>
<span id="cb18-75"><a href="#cb18-75"></a>      <span class="cf">if</span>(modelType<span class="op">==</span><span class="st">&quot;ADE&quot;</span>){</span>
<span id="cb18-76"><a href="#cb18-76"></a>        gblups <span class="op">%&lt;&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="co">#GEEDaa=as.numeric(fit$U[[paste0(&quot;u:&quot;,gid,&quot;aa&quot;)]]$drgBLUP),</span></span>
<span id="cb18-77"><a href="#cb18-77"></a>                           <span class="dt">GEEDad=</span><span class="kw">as.numeric</span>(fit<span class="op">$</span>U[[<span class="kw">paste0</span>(<span class="st">&quot;u:&quot;</span>,gid,<span class="st">&quot;ad&quot;</span>)]]<span class="op">$</span>drgBLUP))</span>
<span id="cb18-78"><a href="#cb18-78"></a>                           <span class="co">#GEEDdd=as.numeric(fit$U[[paste0(&quot;u:&quot;,gid,&quot;dd&quot;)]]$drgBLUP)) </span></span>
<span id="cb18-79"><a href="#cb18-79"></a>      }</span>
<span id="cb18-80"><a href="#cb18-80"></a>    }</span>
<span id="cb18-81"><a href="#cb18-81"></a>    <span class="co"># Calc GETGVs </span></span>
<span id="cb18-82"><a href="#cb18-82"></a>    <span class="co">## Note that for modelType==&quot;A&quot;, GEBV==GETGV</span></span>
<span id="cb18-83"><a href="#cb18-83"></a>    gblups <span class="op">%&lt;&gt;%</span><span class="st"> </span></span>
<span id="cb18-84"><a href="#cb18-84"></a><span class="st">      </span><span class="kw">mutate</span>(<span class="dt">GETGV=</span><span class="kw">rowSums</span>(.[,<span class="kw">grepl</span>(<span class="st">&quot;GE&quot;</span>,<span class="kw">colnames</span>(.))])) </span>
<span id="cb18-85"><a href="#cb18-85"></a>    <span class="co"># Test set validation data</span></span>
<span id="cb18-86"><a href="#cb18-86"></a>    validationData&lt;-TrainTestData <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb18-87"><a href="#cb18-87"></a><span class="st">      </span>dplyr<span class="op">::</span><span class="kw">select</span>(gid,BLUP) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb18-88"><a href="#cb18-88"></a><span class="st">      </span><span class="kw">filter</span>(GID <span class="op">%in%</span><span class="st"> </span><span class="kw">testing</span>(splits)[[gid]])</span>
<span id="cb18-89"><a href="#cb18-89"></a>    <span class="co"># Measure accuracy in test set</span></span>
<span id="cb18-90"><a href="#cb18-90"></a>    <span class="co">## cor(GEBV,BLUP)</span></span>
<span id="cb18-91"><a href="#cb18-91"></a>    <span class="co">## cor(GETGV,BLUP)</span></span>
<span id="cb18-92"><a href="#cb18-92"></a>    accuracy&lt;-gblups <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb18-93"><a href="#cb18-93"></a><span class="st">      </span><span class="kw">mutate</span>(<span class="dt">GETGV=</span><span class="kw">rowSums</span>(.[,<span class="kw">grepl</span>(<span class="st">&quot;GE&quot;</span>,<span class="kw">colnames</span>(.))])) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb18-94"><a href="#cb18-94"></a><span class="st">      </span><span class="kw">filter</span>(GID <span class="op">%in%</span><span class="st"> </span><span class="kw">testing</span>(splits)[[gid]]) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb18-95"><a href="#cb18-95"></a><span class="st">      </span><span class="kw">left_join</span>(validationData) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb18-96"><a href="#cb18-96"></a><span class="st">      </span><span class="kw">summarize</span>(<span class="dt">accGEBV=</span><span class="kw">cor</span>(GEBV,BLUP, <span class="dt">use =</span> <span class="st">&#39;complete.obs&#39;</span>),</span>
<span id="cb18-97"><a href="#cb18-97"></a>                <span class="dt">accGETGV=</span><span class="kw">cor</span>(GETGV,BLUP, <span class="dt">use =</span> <span class="st">&#39;complete.obs&#39;</span>))</span>
<span id="cb18-98"><a href="#cb18-98"></a>    computeTime&lt;-<span class="kw">proc.time</span>()[<span class="dv">3</span>]<span class="op">-</span>starttime</span>
<span id="cb18-99"><a href="#cb18-99"></a>    accuracy <span class="op">%&lt;&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">computeTime=</span>computeTime)</span>
<span id="cb18-100"><a href="#cb18-100"></a>    <span class="kw">return</span>(accuracy)</span>
<span id="cb18-101"><a href="#cb18-101"></a>  },<span class="dt">otherwise =</span> <span class="ot">NA</span>)</span>
<span id="cb18-102"><a href="#cb18-102"></a>  <span class="co">## Run models across all train-test splits</span></span>
<span id="cb18-103"><a href="#cb18-103"></a>  <span class="co">## Parallelize </span></span>
<span id="cb18-104"><a href="#cb18-104"></a>  <span class="kw">require</span>(furrr); <span class="kw">plan</span>(multiprocess); <span class="kw">options</span>(<span class="dt">mc.cores=</span>ncores);</span>
<span id="cb18-105"><a href="#cb18-105"></a>  cvsamples&lt;-cvsamples <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb18-106"><a href="#cb18-106"></a><span class="st">    </span><span class="kw">mutate</span>(<span class="dt">accuracy=</span><span class="kw">future_map2</span>(splits,GroupName,</span>
<span id="cb18-107"><a href="#cb18-107"></a>                                <span class="op">~</span><span class="kw">fitModel</span>(<span class="dt">splits=</span>.x,<span class="dt">GroupName=</span>.y,</span>
<span id="cb18-108"><a href="#cb18-108"></a>                                          <span class="dt">modelType=</span>modelType,<span class="dt">augmentTP=</span><span class="ot">NULL</span>,<span class="dt">TrainTestData=</span>TrainTestData,<span class="dt">grms=</span>grms),</span>
<span id="cb18-109"><a href="#cb18-109"></a>                                <span class="dt">.progress =</span> <span class="ot">FALSE</span>)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb18-110"><a href="#cb18-110"></a><span class="st">    </span><span class="kw">unnest</span>(accuracy)</span>
<span id="cb18-111"><a href="#cb18-111"></a>  <span class="kw">return</span>(cvsamples) </span>
<span id="cb18-112"><a href="#cb18-112"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a><span class="kw">options</span>(<span class="dt">future.globals.maxSize=</span> <span class="dv">3000</span><span class="op">*</span><span class="dv">1024</span><span class="op">^</span><span class="dv">2</span>)</span>
<span id="cb19-2"><a href="#cb19-2"></a>D&lt;-<span class="kw">readRDS</span>(<span class="dt">file=</span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;Kinship_D_NRCRI_2020April27.rds&quot;</span>))</span>
<span id="cb19-3"><a href="#cb19-3"></a><span class="co">#AA&lt;-readRDS(file=here::here(&quot;output&quot;,&quot;Kinship_AA_NRCRI_2020April27.rds&quot;))</span></span>
<span id="cb19-4"><a href="#cb19-4"></a>AD&lt;-<span class="kw">readRDS</span>(<span class="dt">file=</span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;Kinship_AD_NRCRI_2020April27.rds&quot;</span>))</span>
<span id="cb19-5"><a href="#cb19-5"></a><span class="co">#DD&lt;-readRDS(file=here::here(&quot;output&quot;,&quot;Kinship_DD_NRCRI_2020April27.rds&quot;))</span></span>
<span id="cb19-6"><a href="#cb19-6"></a></span>
<span id="cb19-7"><a href="#cb19-7"></a>cv_ADE_nrOnly&lt;-cv2do <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb19-8"><a href="#cb19-8"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">CVresults=</span><span class="kw">map</span>(TrainTestData,<span class="op">~</span><span class="kw">runCrossVal_dev</span>(<span class="dt">TrainTestData=</span>.,</span>
<span id="cb19-9"><a href="#cb19-9"></a>                                                  <span class="dt">modelType=</span><span class="st">&quot;ADE&quot;</span>,</span>
<span id="cb19-10"><a href="#cb19-10"></a>                                                  <span class="dt">grms=</span><span class="kw">list</span>(<span class="dt">A=</span>A,<span class="dt">D=</span>D,<span class="dt">AD=</span>AD),</span>
<span id="cb19-11"><a href="#cb19-11"></a>                                                  <span class="co">#grms=list(A=A,D=D,AA=AA,AD=AD,DD=DD), # test with all kernels failed</span></span>
<span id="cb19-12"><a href="#cb19-12"></a>                                                  <span class="dt">byGroup=</span><span class="ot">TRUE</span>,<span class="dt">augmentTP=</span><span class="ot">NULL</span>,</span>
<span id="cb19-13"><a href="#cb19-13"></a>                                                  <span class="dt">nrepeats=</span><span class="dv">5</span>,<span class="dt">nfolds=</span><span class="dv">5</span>,<span class="dt">ncores=</span><span class="dv">10</span>,<span class="dt">gid=</span><span class="st">&quot;GID&quot;</span>)))</span>
<span id="cb19-14"><a href="#cb19-14"></a></span>
<span id="cb19-15"><a href="#cb19-15"></a>cv_ADE_nrOnly <span class="op">%&lt;&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">Dataset=</span><span class="st">&quot;NRalone&quot;</span>,<span class="dt">modelType=</span><span class="st">&quot;ADE&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>TrainTestData,<span class="op">-</span>augmentTP)</span>
<span id="cb19-16"><a href="#cb19-16"></a><span class="kw">saveRDS</span>(cv_ADE_nrOnly,<span class="dt">file=</span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;cvresults_ADE_nrOnly.rds&quot;</span>))</span></code></pre></div>
</div>
<div id="iita-augmented-2" class="section level3">
<h3>IITA augmented</h3>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a><span class="kw">options</span>(<span class="dt">future.globals.maxSize=</span> <span class="dv">3000</span><span class="op">*</span><span class="dv">1024</span><span class="op">^</span><span class="dv">2</span>)</span>
<span id="cb20-2"><a href="#cb20-2"></a>D&lt;-<span class="kw">readRDS</span>(<span class="dt">file=</span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;Kinship_D_NRCRI_2020April27.rds&quot;</span>))</span>
<span id="cb20-3"><a href="#cb20-3"></a>AD&lt;-<span class="kw">readRDS</span>(<span class="dt">file=</span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;Kinship_AD_NRCRI_2020April27.rds&quot;</span>))</span>
<span id="cb20-4"><a href="#cb20-4"></a></span>
<span id="cb20-5"><a href="#cb20-5"></a>cv_ADE_iitaAugmented&lt;-cv2do <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb20-6"><a href="#cb20-6"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">isnullAugment=</span><span class="kw">map_lgl</span>(augmentTP,<span class="op">~</span><span class="kw">is.null</span>(.))) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb20-7"><a href="#cb20-7"></a><span class="st">  </span><span class="kw">filter</span>(<span class="op">!</span>isnullAugment) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb20-8"><a href="#cb20-8"></a><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>isnullAugment) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb20-9"><a href="#cb20-9"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">CVresults=</span><span class="kw">map2</span>(TrainTestData,augmentTP,<span class="op">~</span><span class="kw">runCrossVal_dev</span>(<span class="dt">TrainTestData=</span>.x,</span>
<span id="cb20-10"><a href="#cb20-10"></a>                                                             <span class="dt">modelType=</span><span class="st">&quot;ADE&quot;</span>,</span>
<span id="cb20-11"><a href="#cb20-11"></a>                                                             <span class="dt">grms=</span><span class="kw">list</span>(<span class="dt">A=</span>A,<span class="dt">D=</span>D,<span class="dt">AD=</span>AD),</span>
<span id="cb20-12"><a href="#cb20-12"></a>                                                             <span class="dt">byGroup=</span><span class="ot">TRUE</span>,<span class="dt">augmentTP=</span>.y,</span>
<span id="cb20-13"><a href="#cb20-13"></a>                                                             <span class="dt">nrepeats=</span><span class="dv">5</span>,<span class="dt">nfolds=</span><span class="dv">5</span>,<span class="dt">ncores=</span><span class="dv">10</span>,<span class="dt">gid=</span><span class="st">&quot;GID&quot;</span>)))</span>
<span id="cb20-14"><a href="#cb20-14"></a>cv_ADE_iitaAugmented <span class="op">%&lt;&gt;%</span><span class="st"> </span><span class="kw">mutate</span>(<span class="dt">Dataset=</span><span class="st">&quot;IITAaugmented&quot;</span>,<span class="dt">modelType=</span><span class="st">&quot;ADE&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(<span class="op">-</span>TrainTestData,<span class="op">-</span>augmentTP)</span>
<span id="cb20-15"><a href="#cb20-15"></a><span class="kw">saveRDS</span>(cv_ADE_iitaAugmented,<span class="dt">file=</span>here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;cvresults_ADE_iitaAugmented.rds&quot;</span>))</span></code></pre></div>
</div>
</div>
<div id="plot-results" class="section level2">
<h2>PLOT RESULTS</h2>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a><span class="kw">rm</span>(<span class="dt">list=</span><span class="kw">ls</span>());<span class="kw">gc</span>()</span></code></pre></div>
<pre><code>          used (Mb) gc trigger (Mb) limit (Mb) max used (Mb)
Ncells  629234 33.7    1284425 68.6         NA  1284425 68.6
Vcells 1188239  9.1    8388608 64.0     102400  2143512 16.4</code></pre>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a><span class="kw">library</span>(tidyverse); <span class="kw">library</span>(magrittr); </span>
<span id="cb23-2"><a href="#cb23-2"></a>cv&lt;-<span class="kw">readRDS</span>(here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;cvresults_A_iitaAugmented.rds&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb23-3"><a href="#cb23-3"></a><span class="st">  </span><span class="kw">bind_rows</span>(<span class="kw">readRDS</span>(here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;cvresults_A_nrOnly.rds&quot;</span>))) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb23-4"><a href="#cb23-4"></a><span class="st">  </span><span class="kw">bind_rows</span>(<span class="kw">readRDS</span>(here<span class="op">::</span><span class="kw">here</span>(<span class="st">&quot;output&quot;</span>,<span class="st">&quot;cvresults_AD_nrOnly.rds&quot;</span>))) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb23-5"><a href="#cb23-5"></a><span class="st">  </span><span class="kw">unnest</span>(CVresults) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb23-6"><a href="#cb23-6"></a><span class="st">  </span><span class="kw">select</span>(<span class="op">-</span>splits)</span></code></pre></div>
<div id="accuracy-gebv" class="section level3">
<h3>Accuracy GEBV</h3>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1"></a><span class="co">#library(viridis)</span></span>
<span id="cb24-2"><a href="#cb24-2"></a><span class="kw">library</span>(tidyverse); <span class="kw">library</span>(magrittr); </span>
<span id="cb24-3"><a href="#cb24-3"></a>cv <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-4"><a href="#cb24-4"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">GroupName=</span><span class="kw">factor</span>(GroupName,<span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;nrTP&quot;</span>,<span class="st">&quot;C1a&quot;</span>,<span class="st">&quot;C1b&quot;</span>,<span class="st">&quot;C2a&quot;</span>)),</span>
<span id="cb24-5"><a href="#cb24-5"></a>         <span class="dt">Dataset=</span><span class="kw">factor</span>(Dataset,<span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;NRalone&quot;</span>,<span class="st">&quot;IITAaugmented&quot;</span>)),</span>
<span id="cb24-6"><a href="#cb24-6"></a>         <span class="dt">modelType=</span><span class="kw">factor</span>(modelType,<span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;A&quot;</span>,<span class="st">&quot;AD&quot;</span>))) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb24-7"><a href="#cb24-7"></a><span class="st">  </span><span class="kw">ggplot</span>(.,<span class="kw">aes</span>(<span class="dt">x=</span>Dataset,<span class="dt">y=</span>accGEBV,<span class="dt">fill=</span>modelType,<span class="dt">linetype=</span>Dataset)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb24-8"><a href="#cb24-8"></a><span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">position =</span> <span class="kw">position_dodge</span>(<span class="dv">1</span>),<span class="dt">width=</span><span class="fl">0.75</span>,<span class="dt">color=</span><span class="st">&#39;gray&#39;</span>,<span class="dt">size=</span><span class="fl">0.75</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb24-9"><a href="#cb24-9"></a><span class="st">  </span><span class="kw">facet_grid</span>(GroupName<span class="op">~</span>Trait, <span class="dt">scales=</span><span class="st">&#39;free&#39;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb24-10"><a href="#cb24-10"></a><span class="st">  </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb24-11"><a href="#cb24-11"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">strip.text.x =</span> <span class="kw">element_text</span>(<span class="dt">face=</span><span class="st">&#39;bold&#39;</span>, <span class="dt">size=</span><span class="dv">12</span>),</span>
<span id="cb24-12"><a href="#cb24-12"></a>        <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">10</span>, <span class="dt">angle =</span> <span class="dv">90</span>),</span>
<span id="cb24-13"><a href="#cb24-13"></a>        <span class="dt">axis.title.y =</span> <span class="kw">element_text</span>(<span class="dt">face=</span><span class="st">&#39;bold&#39;</span>, <span class="dt">size=</span><span class="dv">12</span>)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb24-14"><a href="#cb24-14"></a><span class="st">  </span><span class="kw">scale_fill_viridis_d</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb24-15"><a href="#cb24-15"></a><span class="st">  </span><span class="co">#scale_color_manual(values = c(&quot;gray&quot;,&quot;gold&quot;)) + </span></span>
<span id="cb24-16"><a href="#cb24-16"></a><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="st">&quot;Cross-validated Prediction Accuracy (GEBVs)&quot;</span>) <span class="op">+</span></span>
<span id="cb24-17"><a href="#cb24-17"></a><span class="st">  </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">0</span>, <span class="dt">color=</span><span class="st">&#39;darkred&#39;</span>)</span></code></pre></div>
<p><img src="figure/04-CrossValidation.Rmd/unnamed-chunk-15-1.png" width="960" style="display: block; margin: auto;" /> ### Accuracy GETGV</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1"></a><span class="co">#library(viridis)</span></span>
<span id="cb25-2"><a href="#cb25-2"></a><span class="kw">library</span>(tidyverse); <span class="kw">library</span>(magrittr); </span>
<span id="cb25-3"><a href="#cb25-3"></a>cv <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb25-4"><a href="#cb25-4"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">GroupName=</span><span class="kw">factor</span>(GroupName,<span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;nrTP&quot;</span>,<span class="st">&quot;C1a&quot;</span>,<span class="st">&quot;C1b&quot;</span>,<span class="st">&quot;C2a&quot;</span>)),</span>
<span id="cb25-5"><a href="#cb25-5"></a>         <span class="dt">Dataset=</span><span class="kw">factor</span>(Dataset,<span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;NRalone&quot;</span>,<span class="st">&quot;IITAaugmented&quot;</span>)),</span>
<span id="cb25-6"><a href="#cb25-6"></a>         <span class="dt">modelType=</span><span class="kw">factor</span>(modelType,<span class="dt">levels=</span><span class="kw">c</span>(<span class="st">&quot;A&quot;</span>,<span class="st">&quot;AD&quot;</span>))) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb25-7"><a href="#cb25-7"></a><span class="st">  </span><span class="kw">ggplot</span>(.,<span class="kw">aes</span>(<span class="dt">x=</span>Dataset,<span class="dt">y=</span>accGETGV,<span class="dt">fill=</span>modelType,<span class="dt">linetype=</span>Dataset)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb25-8"><a href="#cb25-8"></a><span class="st">  </span><span class="kw">geom_boxplot</span>(<span class="dt">position =</span> <span class="kw">position_dodge</span>(<span class="dv">1</span>),<span class="dt">width=</span><span class="fl">0.75</span>,<span class="dt">color=</span><span class="st">&#39;gray&#39;</span>,<span class="dt">size=</span><span class="fl">0.75</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb25-9"><a href="#cb25-9"></a><span class="st">  </span><span class="kw">facet_grid</span>(GroupName<span class="op">~</span>Trait, <span class="dt">scales=</span><span class="st">&#39;free&#39;</span>) <span class="op">+</span><span class="st"> </span></span>
<span id="cb25-10"><a href="#cb25-10"></a><span class="st">  </span><span class="kw">theme_bw</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb25-11"><a href="#cb25-11"></a><span class="st">  </span><span class="kw">theme</span>(<span class="dt">strip.text.x =</span> <span class="kw">element_text</span>(<span class="dt">face=</span><span class="st">&#39;bold&#39;</span>, <span class="dt">size=</span><span class="dv">12</span>),</span>
<span id="cb25-12"><a href="#cb25-12"></a>        <span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">size=</span><span class="dv">10</span>, <span class="dt">angle =</span> <span class="dv">90</span>),</span>
<span id="cb25-13"><a href="#cb25-13"></a>        <span class="dt">axis.title.y =</span> <span class="kw">element_text</span>(<span class="dt">face=</span><span class="st">&#39;bold&#39;</span>, <span class="dt">size=</span><span class="dv">12</span>)) <span class="op">+</span><span class="st"> </span></span>
<span id="cb25-14"><a href="#cb25-14"></a><span class="st">  </span><span class="kw">scale_fill_viridis_d</span>() <span class="op">+</span><span class="st"> </span></span>
<span id="cb25-15"><a href="#cb25-15"></a><span class="st">  </span><span class="co">#scale_color_manual(values = c(&quot;gray&quot;,&quot;gold&quot;)) + </span></span>
<span id="cb25-16"><a href="#cb25-16"></a><span class="st">  </span><span class="kw">labs</span>(<span class="dt">title=</span><span class="st">&quot;Cross-validated Prediction Accuracy (GETGVs)&quot;</span>) <span class="op">+</span></span>
<span id="cb25-17"><a href="#cb25-17"></a><span class="st">  </span><span class="kw">geom_hline</span>(<span class="dt">yintercept =</span> <span class="dv">0</span>, <span class="dt">color=</span><span class="st">&#39;darkred&#39;</span>)</span></code></pre></div>
<p><img src="figure/04-CrossValidation.Rmd/unnamed-chunk-16-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="next-step" class="section level1">
<h1>Next step</h1>
<ol start="5" style="list-style-type: decimal">
<li><a href="05-GetGBLUPs.html">Genomic prediction of GS C2</a>: Predict <em>genomic</em> BLUPs (GEBV and GETGV) for all selection candidates using all available data.</li>
</ol>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1"></a><span class="kw">sessionInfo</span>()</span></code></pre></div>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] magrittr_1.5    forcats_0.5.0   stringr_1.4.0   dplyr_0.8.5    
 [5] purrr_0.3.4     readr_1.3.1     tidyr_1.0.2     tibble_3.0.1   
 [9] ggplot2_3.3.0   tidyverse_1.3.0 workflowr_1.6.1

loaded via a namespace (and not attached):
 [1] tidyselect_1.0.0  xfun_0.13         haven_2.2.0       lattice_0.20-41  
 [5] colorspace_1.4-1  vctrs_0.2.4       generics_0.0.2    viridisLite_0.3.0
 [9] htmltools_0.4.0   yaml_2.2.1        rlang_0.4.5       later_1.0.0      
[13] pillar_1.4.3      withr_2.2.0       glue_1.4.0        DBI_1.1.0        
[17] dbplyr_1.4.3      modelr_0.1.6      readxl_1.3.1      lifecycle_0.2.0  
[21] munsell_0.5.0     gtable_0.3.0      cellranger_1.1.0  rvest_0.3.5      
[25] evaluate_0.14     labeling_0.3      knitr_1.28        httpuv_1.5.2     
[29] fansi_0.4.1       broom_0.5.6       Rcpp_1.0.4.6      promises_1.1.0   
[33] backports_1.1.6   scales_1.1.0      jsonlite_1.6.1    farver_2.0.3     
[37] fs_1.4.1          hms_0.5.3         digest_0.6.25     stringi_1.4.6    
[41] grid_3.6.1        rprojroot_1.3-2   here_0.1          cli_2.0.2        
[45] tools_3.6.1       crayon_1.3.4      whisker_0.4       pkgconfig_2.0.3  
[49] ellipsis_0.3.0    xml2_1.3.2        reprex_0.3.0      lubridate_1.7.8  
[53] rstudioapi_0.11   assertthat_0.2.1  rmarkdown_2.1     httr_1.4.1       
[57] R6_2.4.1          nlme_3.1-147      git2r_0.26.1      compiler_3.6.1   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
