<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Marnin Wolfe" />


<title>NRCRI 2020 GS and related procedures</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
      </style>




<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">NRCRI_2020GS</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    April2020 Predictions
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01-cleanTPdata.html">Data Cleaning</a>
    </li>
    <li>
      <a href="02-curateByTrial.html">Curate Data</a>
    </li>
    <li>
      <a href="03-GetBLUPs.html">Get BLUPs</a>
    </li>
    <li>
      <a href="04-CrossValidation.html">Check prediction accuracy</a>
    </li>
    <li>
      <a href="05-GetGEBVs.html">Get GEBV for GS C2</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    October2020 Impute
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="convertDCas20_5440_ToVCF.html">Conversion DArT report DCas20_5440 to VCF</a>
    </li>
    <li>
      <a href="ImputeDCas20_5440.html">Impute DCas20_5440</a>
    </li>
    <li>
      <a href="convertDCas20_5510_ToVCF.html">Conversion DArT report DCas20_5510 to VCF</a>
    </li>
    <li>
      <a href="ImputeDCas20_5510.html">Impute DCas20_5510</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    October2020 Predictions
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="06-cleanTPdata.html">Data Cleaning</a>
    </li>
    <li>
      <a href="07-GetBLUPs.html">Get BLUPs</a>
    </li>
    <li>
      <a href="08-CrossValidation.html">Check prediction accuracy</a>
    </li>
    <li>
      <a href="09-GetGEBVs.html">Get GEBV for GS C2</a>
    </li>
    <li>
      <a href="10-Results.html">Results</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Other
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="about.html">About</a>
    </li>
    <li>
      <a href="license.html">License</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/wolfemd/NRCRI_2020GS">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">NRCRI 2020 GS and related procedures</h1>
<h4 class="author">Marnin Wolfe</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-10-16
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>NRCRI_2020GS/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomwolfemdNRCRI2020GStree977f389859d7b4659096714bee94507948610e9ftargetblank977f389a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/wolfemd/NRCRI_2020GS/tree/977f389859d7b4659096714bee94507948610e9f" target="_blank">977f389</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomwolfemdNRCRI2020GStree977f389859d7b4659096714bee94507948610e9ftargetblank977f389a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/wolfemd/NRCRI_2020GS/tree/977f389859d7b4659096714bee94507948610e9f" target="_blank">977f389</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  CrossValidationFunction_PlusNRCRI_2020GSnotes.gslides
    Untracked:  data/DatabaseDownload_2020Oct13/
    Untracked:  data/chr1_RefPanelAndGSprogeny_ReadyForGP_72719.fam
    Untracked:  output/Kinship_AA_NRCRI_2020April27.rds
    Untracked:  output/Kinship_AD_NRCRI_2020April27.rds
    Untracked:  output/Kinship_AD_NRCRI_2020Oct15.rds
    Untracked:  output/Kinship_A_NRCRI_2020April27.rds
    Untracked:  output/Kinship_A_NRCRI_2020Oct15.rds
    Untracked:  output/Kinship_DD_NRCRI_2020April27.rds
    Untracked:  output/Kinship_D_NRCRI_2020April27.rds
    Untracked:  output/Kinship_D_NRCRI_2020Oct15.rds
    Untracked:  workflowr_log.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML (<code>docs/index.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/NRCRI_2020GS/blob/ace4a862b6fade4533d3b864d6ca4a7c3ba39353/analysis/index.Rmd" target="_blank">ace4a86</a>
</td>
<td>
wolfemd
</td>
<td>
2020-10-16
</td>
<td>
Start workflowr project.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/NRCRI_2020GS/1a34bfa201a48a1fc5a113e578ea681a1904a3ae/docs/index.html" target="_blank">1a34bfa</a>
</td>
<td>
wolfemd
</td>
<td>
2020-10-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/NRCRI_2020GS/345a2fb1a96d1706f4206bd6d4cb2f8ed1d210ce/docs/index.html" target="_blank">345a2fb</a>
</td>
<td>
wolfemd
</td>
<td>
2020-10-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/NRCRI_2020GS/blob/8bff9f09297cd9e4836005531d6586cd91111b43/analysis/index.Rmd" target="_blank">8bff9f0</a>
</td>
<td>
wolfemd
</td>
<td>
2020-10-09
</td>
<td>
Start workflowr project.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/wolfemd/NRCRI_2020GS/f3f6163b319d8633e807c1c45ed2da79e2d747ed/docs/index.html" target="_blank">f3f6163</a>
</td>
<td>
wolfemd
</td>
<td>
2020-04-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/NRCRI_2020GS/blob/5389b1f7dff98c33f061e19639399f78d0e77671/analysis/index.Rmd" target="_blank">5389b1f</a>
</td>
<td>
wolfemd
</td>
<td>
2020-04-28
</td>
<td>
Start workflowr project.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/wolfemd/NRCRI_2020GS/blob/b7a6ef41737e4e3c989ec41e0e398bcbb2539469/analysis/index.Rmd" target="_blank">b7a6ef4</a>
</td>
<td>
wolfemd
</td>
<td>
2020-04-21
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>This repository and website documents all analyses, summary, tables and figures associated with NRCRI genomic prediction and related procedures (e.g. imputation).</p>
<div id="april-genomic-prediction" class="section level1">
<h1>April Genomic Prediction</h1>
<p>Re-prediction of NRCRI germplasm. Updating available training data as of April 2020. Produce GEBV and GETGV.</p>
<ol style="list-style-type: decimal">
<li><a href="01-cleanTPdata.html">Prepare a training dataset</a>: Download data from DB, “Clean” and format DB data.</li>
<li><a href="02-curateByTrial.html">Curate by trait-trial</a>: Model each trait-trial separately, remove outliers, get BLUPs.</li>
<li><a href="03-GetBLUPs.html">Get BLUPs combining all trial data</a>: Combine data from all trait-trials to get BLUPs for downstream genomic prediction.</li>
<li><a href="04-CrossValidation.html">Check prediction accuracy</a>: Evaluate prediction accuracy with cross-validation.</li>
<li><a href="05-GetGBLUPs.html">Genomic prediction of GS C2</a>: Predict <em>genomic</em> BLUPs (GEBV and GETGV) for all selection candidates using all available data.</li>
</ol>
</div>
<div id="october-imputations" class="section level1">
<h1>October Imputations</h1>
<div id="dcas20-5440" class="section level2">
<h2>DCas20-5440</h2>
<p>From Princess Onyyegbule on Sep 22, 2020: “These samples are from materials for inbreeding depression after one self-pollinated generation in two elite cassava varieties (TMS980581 and TMS070337). We want to assess the possibility of obtaining genetic gain by selecting transgressive individuals based on several productive traits, mostly high dry matter content.”</p>
<p><strong>Steps</strong>:</p>
<ol style="list-style-type: decimal">
<li><a href="convertDCas20_5440_ToVCF.html">Convert DCas20-5440 report to VCF for imputation</a>:</li>
<li><a href="ImputeDCas20_5440.html">Impute DCas20-5440</a>: with West Africa reference panel</li>
</ol>
<p><strong>Files</strong>:</p>
<ul>
<li><strong>RefPanel VCF filename:</strong> <code>chr*_ImputationReferencePanel_StageIIpartI_72219.vcf.gz</code></li>
<li><strong>Imputed filename:</strong> <code>chr*_DCas20_5440_WA_REFimputed.vcf.gz</code></li>
<li><strong>Post-impute filtered filename:</strong> <code>chr*_DCas20_5440_WA_REFimputedAndFiltered.vcf.gz</code></li>
<li><strong>Genome-wide dosage matrix format for use in R:</strong> <code>DosageMatrix_DCas20_5440_WA_REFimputedAndFiltered.rds</code></li>
</ul>
</div>
<div id="dcas20-5510" class="section level2">
<h2>DCas20-5510</h2>
<p>GS C3.</p>
<p><strong>Steps</strong>:</p>
<ol style="list-style-type: decimal">
<li><a href="convertDCas20_5510_ToVCF.html">Convert DCas20-5510 report to VCF for imputation</a>:</li>
<li><a href="ImputeDCas20_5510.html">Impute DCas20-5510</a>: with West Africa reference panel</li>
</ol>
<p><strong>Files</strong>:</p>
<ul>
<li><strong>RefPanel VCF filename:</strong> <code>chr*_ImputationReferencePanel_StageIIpartI_72219.vcf.gz</code></li>
<li><strong>Imputed filename:</strong> <code>chr*_DCas20_5510_WA_REFimputed.vcf.gz</code></li>
<li><strong>Post-impute filtered filename:</strong> <code>chr*_DCas20_5510_WA_REFimputedAndFiltered.vcf.gz</code></li>
<li><strong>Genome-wide dosage matrix format for use in R:</strong> <code>DosageMatrix_DCas20_5510_WA_REFimputedAndFiltered.rds</code></li>
</ul>
</div>
</div>
<div id="october-genomic-prediction" class="section level1">
<h1>October Genomic Prediction</h1>
<p>I will update the prediction done in April and predict GEBV/GETGV for all samples in the two new reports (<strong>DCas20-5440 and DCas20-5510</strong>). I learned some lessons doing a <a href="https://wolfemd.github.io/IITA_2020GS/">prediction for IITA in September</a>.</p>
<p>To fit the mixed-model that I want, I am again resorting to <code>asreml-R</code>. I fit random effects for rep and block only where complete and incomplete blocks, respectively are indicated in the trial design variables. <code>sommer</code> <em>should</em> be able to fit the same model via the <code>at()</code> function, <em>but</em> I am having trouble with it <em>and</em> <code>sommer</code> is much slower even without a dense covariance (i.e. a kinship), compared to <code>lme4::lmer()</code> or <code>asreml()</code>. <strong>Note:</strong> For genomic predictions I <em>do</em> use <code>sommer</code>.</p>
<ol style="list-style-type: decimal">
<li><a href="06-cleanTPdata.html">Prepare training dataset</a>: Download data from DB, “Clean” and format DB data.</li>
<li><a href="07-GetBLUPs.html">Get BLUPs combining all trial data</a>: Combine data from all trait-trials to get BLUPs for downstream genomic prediction.
<ul>
<li>Fit mixed-model to multi-trial dataset and extract BLUPs, de-regressed BLUPs and weights. Include two rounds of outlier removal.</li>
</ul></li>
<li><a href="08-CrossValidation.html">Check prediction accuracy</a>: Evaluate prediction accuracy with cross-validation.
<ul>
<li>Compare prediction accuracy with vs. without IITA’s training data to augment.</li>
</ul></li>
<li><a href="09-GetGBLUPs.html">Genomic prediction</a>: Predict <em>genomic</em> BLUPs (GEBV and GETGV) for all selection candidates using all available data.</li>
<li><a href="10-Results.html">Results</a>: New home for plots and other results.</li>
</ol>
</div>
<div id="data-availability-and-reproducibility" class="section level1">
<h1>Data availability and reproducibility</h1>
<p>The R package <strong>workflowr</strong> was used to document this study reproducibly.</p>
<p>Much of the supporting data <em>and</em> output from the analyses documented here are too large for GitHub.</p>
<p>The repository will be mirrored, here: <a href="ftp://ftp.cassavabase.org/marnin_datasets/NRCRI_2020GS/" class="uri">ftp://ftp.cassavabase.org/marnin_datasets/NRCRI_2020GS/</a> with all data.</p>
</div>
<div id="directory-structure-of-this-repository" class="section level1">
<h1>Directory structure of this repository</h1>
<p><strong>NOTICE:</strong> <code>data/</code> and <code>output/</code> are empty on GitHub. Please see <a href="ftp://ftp.cassavabase.org/marnin_datasets/NRCRI_2020GS/" class="uri">ftp://ftp.cassavabase.org/marnin_datasets/NRCRI_2020GS/</a> for access.</p>
<ol style="list-style-type: decimal">
<li><code>data/</code>: raw data (e.g. unimputed SNP data)</li>
<li><code>output/</code>: outputs (e.g. imputed SNP data)</li>
<li><code>analysis/</code>: most code and workflow documented in <strong>.Rmd</strong> files</li>
<li><code>docs/</code>: compiled <strong>.html</strong>, “knitted” from <strong>.Rmd</strong></li>
</ol>
<p>Supporting functions <code>code/</code></p>
<p>The analyses in the <strong>html</strong> / <strong>Rmd</strong> files referenced above often source R scripts in the <code>code/</code> sub-folder.</p>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
