---
title: "Convert NRCRI DCas20-5510 to VCF"
site: workflowr::wflow_site
date: "2020-October-13"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: inline
---

# Input Parameters
```{r, eval=F}
#' @dartvcfInput input name and path of "vcf" file from DArT
#' @dartcountsInput input name and path of counts file from DArT
#' @outName output path and name
#' @nskipvcf number of "VCF" rows to skip on read-in
#' @nskipcounts number of "counts file" rows to skip on read in
#' @ncores number of cores to use, could be VERY memory intensive
#' @dartVars chr vector, column names that _are not_ sample IDs in the read-counts file. I use this internally to assign the sampleIDs in the VCF file

library(tidyverse); library(magrittr)
dartvcfInput<-here::here("data/Report-DCas20-5510","Report_5510_VCF_Ref_Version6.txt")
dartcountsInput<-here::here("data/Report-DCas20-5510","Report_5510_Counts_Ref_Version6.csv")
outName<-here::here("data/Report-DCas20-5510","DCas20_5510")
nskipvcf<-2 
nskipcounts<-3
ncores<-10
```

# Prelim. check format

Start manual. Check that the files read in according to previous code. Adjust code if necessary. Make a function and apply it to the input files.

```{r, eval=F}
# vcf<-read.table(dartvcfInput,
#                 stringsAsFactors = F,skip = nskipvcf, header = T, sep = "\t", comment.char = "")
# readCounts<-read.csv(dartcountsInput, stringsAsFactors = F,header = T,skip=nskipcounts)
# 
# dim(vcf)
# # [1] 13603   404
# dim(readCounts)
# # [1] 27206   438
# 
# # # Initial look at names....
# # colnames(readCounts)[1:100]
# [44] "NR.NG.C3.Cass.0001_A01...NR19C3aF1P0002" 
#  [45] "NR.NG.C3.Cass.0001_B01...NR19C3aF6P0002" 
#  [46] "NR.NG.C3.Cass.0001_C01...NR19C3aF6P0003" 
#  [47] "NR.NG.C3.Cass.0001_D01...NR19C3aF6P0004" 
#  [48] "NR.NG.C3.Cass.0001_E01...NR19C3aF6P0005" 
#  [49] "NR.NG.C3.Cass.0001_F01...NR19C3aF6P0007" 
#  [50] "NR.NG.C3.Cass.0001_C02...NR19C3aF9P0002" 
#  [51] "NR.NG.C3.Cass.0001_E02...NR19C3aF10P0001"
#  [52] "NR.NG.C3.Cass.0001_F02...NR19C3aF10P0002"
#  [53] "NR.NG.C3.Cass.0001_G02...NR19C3aF10P0004"
#  [54] "NR.NG.C3.Cass.0001_H02...NR19C3aF10P0005"
# # colnames(vcf)[1:30]
# [10] "NR.NG.C3.Cass.0001_A01...NR19C3aF1P0002" 
# [11] "NR.NG.C3.Cass.0001_A10...NR19C3aF41P0004"
# [12] "NR.NG.C3.Cass.0001_A11...NR19C3aF52P0001"
# [13] "NR.NG.C3.Cass.0001_A12...NR19C3aF54P0002"
# [14] "NR.NG.C3.Cass.0001_A02...NR19C3aF7P0001" 
# [15] "NR.NG.C3.Cass.0001_A03...NR19C3aF10P0006"
# [16] "NR.NG.C3.Cass.0001_A04...NR19C3aF14P0003"
# [17] "NR.NG.C3.Cass.0001_A05...NR19C3aF18P0002" 
# rm(vcf,readCounts); gc()
```

# Conversion function

Available and sourced from `code/` subdirectory: `convertDart2vcf.R`.

# Run conversion function
```{r, eval=F}
source(here::here("code/","convertDart2vcf.R"))
convertDart2vcf(dartvcfInput,dartcountsInput,outName,
                nskipvcf=2,nskipcounts=3,ncores)
```

# Genomewide to per-chrom VCFs

Split the genome-wide VCF into per-chromosome VCFs for imputation.

```{r,eval = FALSE}
require(furrr); options(mc.cores=18); plan(multiprocess)
source(here::here("code","imputationFunctions.R"))

vcfIn<-here::here("data/Report-DCas20-5510","DCas20_5510.vcf.gz")
filters<-"--minDP 4 --maxDP 50" # because using GT not PL for impute (Beagle5)
outPath<-here::here("data/Report-DCas20-5510/")
outSuffix<-"DCas20_5510"

future_map(1:18,
           ~splitVCFbyChr(Chr=.,
                          vcfIn=vcfIn,filters=filters,
                          outPath=outPath,outSuffix=outSuffix))
```
